<!doctype html><html lang="en" class="no-js"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="lang:clipboard.copy" content="Copy to clipboard"><meta name="lang:clipboard.copied" content="Copied to clipboard"><meta name="lang:search.language" content="en"><meta name="lang:search.pipeline.stopwords" content="True"><meta name="lang:search.pipeline.trimmer" content="True"><meta name="lang:search.result.none" content="No matching documents"><meta name="lang:search.result.one" content="1 matching document"><meta name="lang:search.result.other" content="# matching documents"><meta name="lang:search.tokenizer" content="[\s\-]+"><link rel="shortcut icon" href="../../../assets/images/favicon.png"><meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.3.1"><title>Introduction to Galaxy - Bioinformatics Documentation</title><link rel="stylesheet" href="../../../assets/stylesheets/application.4031d38b.css"><script src="../../../assets/javascripts/modernizr.74668098.js"></script><link href="https://fonts.gstatic.com" rel="preconnect" crossorigin><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=swap"><style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style><link rel="stylesheet" href="../../../assets/fonts/material-icons.css"><link rel="stylesheet" href="../../../extra.css"><script>window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-21794901-5", "www.melbournebioinformatics.org.au/tutorials/")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })</script><script async src="https://www.google-analytics.com/analytics.js"></script></head><body dir="ltr"><svg class="md-svg"><defs><svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg></defs></svg> <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off"> <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off"> <label class="md-overlay" data-md-component="overlay" for="__drawer"></label><a href="#introduction-to-galaxy" tabindex="1" class="md-skip">Skip to content </a><header class="md-header" data-md-component="header"><nav class="md-header-nav md-grid"><div class="md-flex"><div class="md-flex__cell md-flex__cell--shrink"><a href="../../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo"><i class="md-icon"></i></a></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label></div><div class="md-flex__cell md-flex__cell--stretch"><div class="md-flex__ellipsis md-header-nav__title" data-md-component="title"><span class="md-header-nav__topic">Bioinformatics Documentation</span><span class="md-header-nav__topic">Introduction to Galaxy</span></div></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--search md-header-nav__button" for="__search"></label><div class="md-search" data-md-component="search" role="dialog"><label class="md-search__overlay" for="__search"></label><div class="md-search__inner" role="search"><form class="md-search__form" name="search"><input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active"> <label class="md-icon md-search__icon" for="__search"></label> <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">&#xE5CD;</button></form><div class="md-search__output"><div class="md-search__scrollwrap" data-md-scrollfix><div class="md-search-result" data-md-component="result"><div class="md-search-result__meta">Type to start searching</div><ol class="md-search-result__list"></ol></div></div></div></div></div></div><div class="md-flex__cell md-flex__cell--shrink"><div class="md-header-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div></div></div></nav></header><div class="md-container"><main class="md-main"><div class="md-main__inner md-grid" data-md-component="container"><div class="md-sidebar md-sidebar--primary" data-md-component="navigation"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--primary" data-md-level="0"><label class="md-nav__title md-nav__title--site" for="__drawer"><a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo"><i class="md-icon"></i></a>Bioinformatics Documentation</label><div class="md-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../.." title="Home" class="md-nav__link">Home</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2"><label class="md-nav__link" for="nav-2">Guides</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-2">Guides</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../../guides/bioinfo/" title="What is bioinformatics?" class="md-nav__link">What is bioinformatics?</a></li><li class="md-nav__item"><a href="../../../guides/galaxy/" title="The Galaxy platform" class="md-nav__link">The Galaxy platform</a></li></ul></nav></li><li class="md-nav__item md-nav__item--active md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked><label class="md-nav__link" for="nav-3">General Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-3">General Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../python_overview/python_overview/" title="Python Overview" class="md-nav__link">Python Overview</a></li><li class="md-nav__item"><a href="../../python_pandas_tidy_data/python_pandas_tidy_data/" title="Data tidying with Python and Pandas" class="md-nav__link">Data tidying with Python and Pandas</a></li><li class="md-nav__item"><a href="../../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">Data visualisation with Python</a></li><li class="md-nav__item"><a href="../../using_git/Using_Git/" title="Using Git and Github for revision control" class="md-nav__link">Using Git and Github for revision control</a></li><li class="md-nav__item"><a href="../../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">Introduction to HPC</a></li><li class="md-nav__item"><a href="../../unix/unix/" title="Introduction to Unix" class="md-nav__link">Introduction to Unix</a></li><li class="md-nav__item"><a href="../../gvl_launch/gvl_launch/" title="Launching a Personal GVL Server" class="md-nav__link">Launching a Personal GVL Server</a></li><li class="md-nav__item md-nav__item--active"><input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc"><label class="md-nav__link md-nav__link--active" for="__toc">Introduction to Galaxy</label><a href="./" title="Introduction to Galaxy" class="md-nav__link md-nav__link--active">Introduction to Galaxy</a><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#overview" title="Overview" class="md-nav__link">Overview</a></li><li class="md-nav__item"><a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">Learning Objectives</a></li><li class="md-nav__item"><a href="#requirements" title="Requirements" class="md-nav__link">Requirements</a></li><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a></li><li class="md-nav__item"><a href="#section-1-preparation" title="Section 1: Preparation." class="md-nav__link">Section 1: Preparation.</a></li><li class="md-nav__item"><a href="#section-2-getting-data-into-galaxy" title="Section 2: Getting data into Galaxy" class="md-nav__link">Section 2: Getting data into Galaxy</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#method-1-upload-a-file-from-your-own-computer" title="Method 1: Upload a file from your own computer" class="md-nav__link">Method 1: Upload a file from your own computer</a></li><li class="md-nav__item"><a href="#method-2-upload-a-file-from-a-url" title="Method 2: Upload a file from a URL" class="md-nav__link">Method 2: Upload a file from a URL</a></li><li class="md-nav__item"><a href="#method-2-again-get-data-from-a-data-library" title="Method 2 (again): Get data from a Data Library" class="md-nav__link">Method 2 (again): Get data from a Data Library</a></li><li class="md-nav__item"><a href="#the-data" title="The data" class="md-nav__link">The data</a></li></ul></nav></li><li class="md-nav__item"><a href="#section-3-play-with-the-tools" title="Section 3: Play with the tools" class="md-nav__link">Section 3: Play with the tools</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#example-1-histogram-and-summary-statistics" title="Example 1: Histogram and summary statistics" class="md-nav__link">Example 1: Histogram and summary statistics</a></li><li class="md-nav__item"><a href="#example-2-convert-fastq-to-fasta" title="Example 2: Convert Fastq to Fasta" class="md-nav__link">Example 2: Convert Fastq to Fasta</a></li><li class="md-nav__item"><a href="#example-3-find-ribosomal-rna-features-in-a-dna-sequence" title="Example 3: Find Ribosomal RNA Features in a DNA Sequence" class="md-nav__link">Example 3: Find Ribosomal RNA Features in a DNA Sequence</a></li></ul></nav></li><li class="md-nav__item"><a href="#what-now" title="What now?" class="md-nav__link">What now?</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-9" type="checkbox" id="nav-3-9"><label class="md-nav__link" for="nav-3-9">Galaxy Workflows</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-3-9">Galaxy Workflows</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../galaxy-workflows/galaxy-workflows/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../galaxy-workflows/history_creation/" title="History Creation" class="md-nav__link">History Creation</a></li></ul></nav></li><li class="md-nav__item"><a href="../../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">Containerized Bioinformatics</a></li><li class="md-nav__item"><a href="../../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">Common Workflow Language for Bioinformatics</a></li><li class="md-nav__item"><a href="../../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">Command-line software development</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4"><label class="md-nav__link" for="nav-4">Genomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-4">Genomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" title="Introduction to Variant Calling" class="md-nav__link">Introduction to Variant Calling</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2"><label class="md-nav__link" for="nav-4-2">Variant Calling Advanced</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-2">Variant Calling Advanced</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3"><label class="md-nav__link" for="nav-4-3">RNA-seq DGE Basic</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-3">RNA-seq DGE Basic</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="Tuxedo Protocol Tutorial" class="md-nav__link">Tuxedo Protocol Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNA-seq DGE Advanced" class="md-nav__link">RNA-seq DGE Advanced</a></li><li class="md-nav__item"><a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq DGE Experimental Design" class="md-nav__link">RNA-seq DGE Experimental Design</a></li><li class="md-nav__item"><a href="../../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">10X single-cell RNA-seq analysis in R</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-7" type="checkbox" id="nav-4-7"><label class="md-nav__link" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../assembly/assembly-background/" title="Method Background" class="md-nav__link">Method Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../assembly/spades/" title="Microbial assembly: Spades" class="md-nav__link">Microbial assembly: Spades</a></li><li class="md-nav__item"><a href="../../pacbio/" title="Long read assembly" class="md-nav__link">Long read assembly</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5"><label class="md-nav__link" for="nav-5">Proteomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-5">Proteomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-1" type="checkbox" id="nav-5-1"><label class="md-nav__link" for="nav-5-1">Identifying proteins from mass spectrometry data</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-5-1">Identifying proteins from mass spectrometry data</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../proteomics_basic/proteomics_basic/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_galaxy/" title="Galaxy Background" class="md-nav__link">Galaxy Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">Data Formats Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_findcountitems_workflow/" title="Create find count items workflow" class="md-nav__link">Create find count items workflow</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">Protein Databases</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">Search Engines</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">Protein Prophet</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6"><label class="md-nav__link" for="nav-6">Molecular Modelling Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-6">Molecular Modelling Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">Molecular Dynamics - Introduction to cluster computing</a></li><li class="md-nav__item"><a href="../../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">Molecular Dynamics - Building input files, visualising the trajectory</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7"><label class="md-nav__link" for="nav-7">Protocols</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-7">Protocols</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly-protocol/" title="De Novo Genome Assembly for Illumina Data" class="md-nav__link">De Novo Genome Assembly for Illumina Data</a></li></ul></nav></li></ul></nav></div></div></div><div class="md-sidebar md-sidebar--secondary" data-md-component="toc"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#overview" title="Overview" class="md-nav__link">Overview</a></li><li class="md-nav__item"><a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">Learning Objectives</a></li><li class="md-nav__item"><a href="#requirements" title="Requirements" class="md-nav__link">Requirements</a></li><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a></li><li class="md-nav__item"><a href="#section-1-preparation" title="Section 1: Preparation." class="md-nav__link">Section 1: Preparation.</a></li><li class="md-nav__item"><a href="#section-2-getting-data-into-galaxy" title="Section 2: Getting data into Galaxy" class="md-nav__link">Section 2: Getting data into Galaxy</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#method-1-upload-a-file-from-your-own-computer" title="Method 1: Upload a file from your own computer" class="md-nav__link">Method 1: Upload a file from your own computer</a></li><li class="md-nav__item"><a href="#method-2-upload-a-file-from-a-url" title="Method 2: Upload a file from a URL" class="md-nav__link">Method 2: Upload a file from a URL</a></li><li class="md-nav__item"><a href="#method-2-again-get-data-from-a-data-library" title="Method 2 (again): Get data from a Data Library" class="md-nav__link">Method 2 (again): Get data from a Data Library</a></li><li class="md-nav__item"><a href="#the-data" title="The data" class="md-nav__link">The data</a></li></ul></nav></li><li class="md-nav__item"><a href="#section-3-play-with-the-tools" title="Section 3: Play with the tools" class="md-nav__link">Section 3: Play with the tools</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#example-1-histogram-and-summary-statistics" title="Example 1: Histogram and summary statistics" class="md-nav__link">Example 1: Histogram and summary statistics</a></li><li class="md-nav__item"><a href="#example-2-convert-fastq-to-fasta" title="Example 2: Convert Fastq to Fasta" class="md-nav__link">Example 2: Convert Fastq to Fasta</a></li><li class="md-nav__item"><a href="#example-3-find-ribosomal-rna-features-in-a-dna-sequence" title="Example 3: Find Ribosomal RNA Features in a DNA Sequence" class="md-nav__link">Example 3: Find Ribosomal RNA Features in a DNA Sequence</a></li></ul></nav></li><li class="md-nav__item"><a href="#what-now" title="What now?" class="md-nav__link">What now?</a></li></ul></nav></div></div></div><div class="md-content"><article class="md-content__inner md-typeset"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/galaxy_101/galaxy_101.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a><style>
  .image-header-gvl {
    -webkit-column-count: 3;
    -moz-column-count: 3;
    column-count: 3;

    -webkit-column-gap: 140px;
    -moz-column-gap: 140px;
    column-gap: 140px;
  }
</style>

<div class="image-header-gvl">
    <a href=http://genome.edu.au><img src="../media/gvl_logo.jpg" alt="GVL logo" width="112"/></a>
    <a href="http://galaxyproject.org"><img src="../media/GalaxyLogoHighRes.png" width="164" /></a>
    <a href="https://www.melbournebioinformatics.org.au/"><img src="../../../img/melbioinf_logo.png" alt="Melbourne Bioinformatics logo" width="164"/></a>
</div>

<h1 id="introduction-to-galaxy">Introduction to Galaxy</h1>
<p>Written and maintained by <a href="mailto:simon.gladman@unimelb.edu.au">Simon Gladman</a> - Melbourne Bioinformatics (formerly VLSCI)</p>
<h2 id="overview">Overview</h2>
<p>This beginners tutorial will introduce Galaxy's interface, tool use, histories, and get new users of the Genomics Virtual Laboratory up and running. You can follow this tutorial with the <a href="../../galaxy-workflows/galaxy-workflows/">Galaxy Workflows</a> tutorial to learn about workflows.</p>
<p>Galaxy is an open source, web-based platform for accessible, reproducible, and transparent computational biomedical research. It allows users without programming experience to easily specify parameters and run individual tools as well as larger workflows. It also captures run information so that any user can repeat and understand a complete computational analysis. Finally, it allows users to share and publish analyses via the web.</p>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of the course, you will be able to:</p>
<ul>
<li>login to a Galaxy server.</li>
<li>upload data to a Galaxy server from:</li>
<li>A file on your local computer</li>
<li>A file on a remote datastore with an accessible URL.</li>
<li>use tools in Galaxy by:</li>
<li>Accessing the tool via the tool menu</li>
<li>Using the tool interface to run the particular tool</li>
<li>Viewing/accessing the tool output.</li>
</ul>
<h2 id="requirements">Requirements</h2>
<p>This is a hands-on workshop and attendees should bring their own laptops.</p>
<h2 id="background">Background</h2>
<p>Galaxy is a web based analysis and workflow platform designed for biologists to analyse their own data. It comes with most of the popular bioinformatics tools already installed and ready for use. There are many Galaxy servers around the world and some are tailored with specific toolsets and reference data for analysis of human genomics, microbial genomics, proteomics etc.</p>
<p>There are some introductory slides available <a href="https://docs.google.com/presentation/d/1dzHagGkswjH7MOZ7OACVXGU-riBs33K3J5lWpnCpPhs/pub?start=false&amp;loop=false&amp;slide=id.g537781ff1_2_18">here</a>.</p>
<p>Basically, the Galaxy interface is separated into 3 parts. The tool list on the left, the viewing pane in the middle and the analysis and data history on the right. We will be looking at all 3 parts in this tutorial.</p>
<p><img src="../media/Galaxy_components.png" alt="Galaxy interface components diagram" width="70%" /></p>
<p>This workshop/tutorial will familiarize you with the Galaxy interface. It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>Getting data into galaxy</li>
<li>How to access the tools</li>
<li>Using to use some common tools</li>
</ul>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.</h2>
<p>The purpose of this section is to get you to log in to the server.</p>
<ol>
<li>
<p>Open your browser. We recommend Firefox or Chrome (please don't use Internet Explorer or Safari).</p>
<ul>
<li>Go to the <a href="https://usegalaxy.org.au/">Galaxy Australia</a> server.</li>
<li>Alternatively, you can use a different Galaxy server - a list of available servers is <a href="https://galaxyproject.org/galaxy-services/">here</a>.</li>
</ul>
</li>
<li>
<p>If you have previously registered on this server just log in:</p>
<ul>
<li>On the top menu select: <strong>User -&gt; Login</strong></li>
<li>Enter your password</li>
<li>Click <strong>Submit</strong></li>
</ul>
</li>
<li>
<p>If you haven’t registered on this server, you’ll need to now.</p>
<ul>
<li>On the top menu select: <strong>User -&gt; Register</strong></li>
<li>Enter your email, choose a password, repeat it and add a (all lower case) one word name</li>
<li>Click <strong>Submit</strong></li>
</ul>
</li>
</ol>
<hr />
<h2 id="section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy</h2>
<p>There are 2 main ways to get your data into Galaxy. We will use each of these methods for 3 files and then use those 3 files for the rest of the workshop.</p>
<ul>
<li>
<p>Start a new history for this workshop. To do this:</p>
<ul>
<li>Click on the history menu button (the <img src="../media/Galaxy-menu.png" width=20 /> icon) at the top of the Histories panel.</li>
<li>Select <strong>Create New</strong></li>
</ul>
</li>
</ul>
<p>It is important to note that Galaxy has the concept of "File Type" built in. This means that each file stored needs to have its type described to Galaxy as it is being made available. Examples of file types are: text, fasta, fastq, vcf, GFF, Genbank, tabular etc.</p>
<p>We will tell Galaxy what type of file each one is as we upload it.</p>
<h3 id="method-1-upload-a-file-from-your-own-computer">Method 1: Upload a file from your own computer</h3>
<p>With this method you can get most of the files on your own computer into Galaxy. (there is a size limit)</p>
<ul>
<li>
<p>Download the following file to your computer: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz</em></p>
<ul>
<li>(To download this file, copy the link into a new browser tab, and press enter. The file should now download.)</li>
<li>From the Galaxy tool panel, click on <strong>Get Data -&gt; Upload File</strong></li>
<li>Click the <strong>Choose File</strong> button</li>
<li>Find and select the <em>Contig_stats.txt.gz</em> file you downloaded and click <strong>Open</strong></li>
<li>Set the "Type" (= file format) to <em>tabular</em></li>
<li>Click the <strong>Start</strong> button</li>
<li>Once the progress bar reaches 100%, click the <strong>Close</strong> button</li>
</ul>
</li>
</ul>
<p>The file will now upload to your current history.</p>
<h3 id="method-2-upload-a-file-from-a-url">Method 2: Upload a file from a URL</h3>
<p>If a file exists on a web resource somewhere and you know its URL (Unique resource location - a web address) you can directly load it into Galaxy.</p>
<ul>
<li>
<p>From the tool panel, click on <strong>Get Data -&gt; Upload File</strong></p>
<ul>
<li>Click on the <strong>Paste/Fetch Data</strong> button</li>
<li>Copy and paste the following web address into the URL/Text box: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz</em></li>
<li>Set the file format to <em>fastqsanger</em> (not fastqcsanger)</li>
<li>Click <strong>Start</strong></li>
<li>Once the progress bar has reached 100%, click <strong>Close</strong></li>
</ul>
</li>
</ul>
<p>Note that Galaxy is smart enough to recognize that this is a compressed file and so it will uncompress it as it loads it.</p>
<h3 id="method-2-again-get-data-from-a-data-library">Method 2 (again): Get data from a Data Library</h3>
<p>Now we are going to get another file from a shared Data Library.</p>
<ul>
<li>
<p>Go to the menu at the top of the screen and click <strong>Shared Data -&gt; Data Libraries</strong>.</p>
</li>
<li>
<p>Click on the Library: "Galaxy Australia Training Material" then "Galaxy_101"</p>
</li>
<li>
<p>To add the MRSA0252.fa file to our history</p>
</li>
<li>click on the checkbox next to it</li>
<li>Then click the "To History" button at the top of the page and select "As Datasets"</li>
<li>Click the "Import" button</li>
<li>Finally, click "Analyse Data" in the menu at the top of the screen to return to your history.</li>
</ul>
<p>The DNA sequence of <em>Staphlococcus aureus MRSA252</em> will be loaded into your history as a fasta file.</p>
<p>Your history should now look like this.</p>
<p><img src="../media/starting-history.png" width="300px" align="center" /></p>
<h3 id="the-data">The data</h3>
<p>Though we aren't going to focus on the contents of these files and what they mean from a bioinformatics standpoint, here is a brief description of each one.</p>
<ul>
<li>
<p><em>Contigs_stats.txt</em></p>
<ul>
<li>this file contains a table of summary data from a de novo genome assembly (the process of attempting to recover the full genome of an organism from the short read sequences produced by most DNA sequencing machines. )</li>
<li>The columns contain a lot of information but the ones we will be using indicate the amount of data (or coverage) that went into making up each piece of the final assembly.  </li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>bacterial_std_err_1.fastq.gz</em></p>
<ul>
<li>This file contains sequence reads as they would come off an Illumina sequencing machine. They are in <a href="https://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format.  </li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>MRSA0252.fna</em></p>
<ul>
<li>This file contains the genome sequence of <em>Staphylococcus aureus MRSA252</em>. It is in <a href="https://en.wikipedia.org/wiki/FASTA_format">fasta</a> format.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-3-play-with-the-tools">Section 3: Play with the tools</h2>
<p>The purpose of this section is to get you used to using the available tools in Galaxy and point out some of the more basic manipulation tools.</p>
<p>Firstly however, you’ll notice that two of the files have very long and confusing names. So we might want to change them. To do this we need to “edit” the file. So:</p>
<ol>
<li>Click on the <img src="../media/Galaxy-edit.png" width=20 /> icon (edit) next to the file in the history called: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq</em></li>
<li>In the "Name" text box, give it a new name. Call it: <em>Typical Fastq File</em></li>
<li>Click the <strong>Save</strong> button.</li>
</ol>
<p>Repeat the process for the MRSA252 fasta file. Rename it to <em>MRSA252.fna</em></p>
<p>Now that’s better. There was a lot of other functionality hidden behind that edit (<img src="../media/Galaxy-edit.png" width=20 />) icon. You can change a file’s data type, convert its format and many other things. Feel free to play around with them at a later date.</p>
<p>Ok, back to the tools.</p>
<h3 id="example-1-histogram-and-summary-statistics">Example 1: Histogram and summary statistics</h3>
<p>The first thing we are going to do is produce a histogram of contig read coverage depths and calculate the summary statistics from the Contig_stats.txt file. To do this we need to cut out a couple of columns, remove a line and then produce a histogram. This will introduce some of the text manipulation tools.</p>
<p>Click on the <img src="../media/Galaxy-view.png" width=20 /> icon of the <em>Contig_stats.txt</em> file to have a look at it. Note that there are 18 columns in this file. We want column 1 and column 6. To do this:</p>
<p><strong>1. Cut out column 1 and column 6.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Cut</strong> and set the following:</li>
<li>Set "Cut Columns" to: <em>c1,c6</em></li>
<li>"Delimited by": <em>Tab</em></li>
<li>"Cut from": <em>Contig_stats.txt</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Examine the new file by clicking on its <img src="../media/Galaxy-view.png" width=20 /> icon. We now have 2 columns instead of the 18 in the original file.</p>
<p><strong>2. Remove the Header lines of the new file.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Remove beginning</strong> and set the following:</li>
<li>"Remove First": <em>1</em></li>
<li>"from": <em>Cut on data1</em></li>
<li>click <strong>Execute</strong></li>
</ul>
<p>Note the the new file is the same as the previous one without the header line.</p>
<p><strong>3. Make a histogram.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Graph/Display Data -&gt; Histogram</strong> and set the following:</li>
<li>"Dataset": <em>Remove beginning on Data X</em></li>
<li>"Numerical column for X axis": <em>c2</em></li>
<li>"Number of breaks": <em>25</em></li>
<li>"Plot title": <em>Histogram of Contig Coverage</em></li>
<li>"Label for X axis": <em>Coverage depth</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Click on the <img src="../media/Galaxy-view.png" width=20 /> icon of the histogram to have a look at it. Note there are a few peaks. Maybe these correspond to single, double and triple copy number of these contigs.</p>
<p><strong>4. Calculate summary statistics for contig coverage depth.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Statistics -&gt; Summary Statisitics</strong> and set the following:</li>
<li>"Summary statistics on": <em>Remove beginning on Data X</em></li>
<li>"Column or expression": <em>c2</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<h3 id="example-2-convert-fastq-to-fasta">Example 2: Convert Fastq to Fasta</h3>
<p>This shows how to convert a fastq file to a fasta file. The tool creates a new file with the converted data.</p>
<p><strong>Converter tool</strong></p>
<ul>
<li>From the tool panel, click on <strong>Convert Formats -&gt; FASTQ to FASTA</strong> and set the following:</li>
<li>"FASTQ file to convert": <em>Typical Fastq File</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>This will have created a new Fasta file called FASTQ to FASTA on data 2.</p>
<h3 id="example-3-find-ribosomal-rna-features-in-a-dna-sequence">Example 3: Find Ribosomal RNA Features in a DNA Sequence</h3>
<p>This example shows how to use a tool called “barrnap” to search for rRNAs in a DNA sequence.</p>
<p><strong>1. Find all of the ribosomal RNA's in a sequence</strong></p>
<ul>
<li>From the tool panel, click on <strong>NGS: Annotation -&gt; barrnap</strong> and set the following:</li>
<li>"Fasta file": MRSA252.fna</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>A new file called <em>barrnap on data 3</em> will be produced. It is a gff3 file. (This stands for genome feature format - version 3. It is a file format for describing features contained by a DNA sequence.) Change its name to something more appropriate (click on the <img src="../media/Galaxy-edit.png" width=20 /> icon.)</p>
<p>Now let's say you only want the lines of the file for the 23S rRNA annotations. We can do this using a Filter tool.</p>
<p><strong>2. Filter the annotations to get the 23S RNAs</strong></p>
<ul>
<li>From the tool panel, click on <strong>Filter and Sort -&gt; Select</strong> and set the following:</li>
<li>"Select lines from":  (whatever you called the barrnap gff3 output)</li>
<li>"that": <em>Matching</em></li>
<li>"the pattern": <em>23S</em>     (this will look for all the lines in the file that contain “23S”)</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Now you have a gff3 file with just the 23S annotations!</p>
<h2 id="what-now">What now?</h2>
<p>Remember how we started a new history at the beginning? If you want to see any of your old histories, click on the history menu button <img src="../media/Galaxy-menu.png" width=20 /> at the top of the histories panel and then select “Saved Histories.” This will give you a list of all the histories you have worked on in this Galaxy server.</p>
<p>That's it. You now know a bit about the Galaxy interface and how to load data, run tools and view their outputs. For more tutorials, see <a href="http://genome.edu.au/learn">http://genome.edu.au/learn</a></p></article></div></div></main><footer class="md-footer"><div class="md-footer-nav"><nav class="md-footer-nav__inner md-grid"><a href="../../gvl_launch/gvl_launch/" title="Launching a Personal GVL Server" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev"><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-back md-footer-nav__button"></i></div><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Previous</span>Launching a Personal GVL Server</span></div></a><a href="../../galaxy-workflows/galaxy-workflows/" title="Tutorial" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next"><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Next</span>Tutorial</span></div><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i></div></a></nav></div><div class="md-footer-meta md-typeset"><div class="md-footer-meta__inner md-grid"><div class="md-footer-copyright"><div class="md-footer-copyright__highlight"><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a></div>powered by <a href="https://www.mkdocs.org">MkDocs</a> and <a href="https://squidfunk.github.io/mkdocs-material/">Material for MkDocs</a></div></div></div></footer></div><script src="../../../assets/javascripts/application.b260a35d.js"></script><script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script></body></html>