<!doctype html><html lang="en" class="no-js"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="lang:clipboard.copy" content="Copy to clipboard"><meta name="lang:clipboard.copied" content="Copied to clipboard"><meta name="lang:search.language" content="en"><meta name="lang:search.pipeline.stopwords" content="True"><meta name="lang:search.pipeline.trimmer" content="True"><meta name="lang:search.result.none" content="No matching documents"><meta name="lang:search.result.one" content="1 matching document"><meta name="lang:search.result.other" content="# matching documents"><meta name="lang:search.tokenizer" content="[\s\-]+"><link rel="shortcut icon" href="../../../assets/images/favicon.png"><meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.3.1"><title>Launching a Personal GVL Server - Bioinformatics Documentation</title><link rel="stylesheet" href="../../../assets/stylesheets/application.4031d38b.css"><script src="../../../assets/javascripts/modernizr.74668098.js"></script><link href="https://fonts.gstatic.com" rel="preconnect" crossorigin><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=swap"><style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style><link rel="stylesheet" href="../../../assets/fonts/material-icons.css"><link rel="stylesheet" href="../../../extra.css"><script>window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-21794901-5", "www.melbournebioinformatics.org.au/tutorials/")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })</script><script async src="https://www.google-analytics.com/analytics.js"></script></head><body dir="ltr"><svg class="md-svg"><defs><svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg></defs></svg> <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off"> <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off"> <label class="md-overlay" data-md-component="overlay" for="__drawer"></label><a href="#launching-a-personal-gvl-server-on-the-nectar-research-cloud" tabindex="1" class="md-skip">Skip to content </a><header class="md-header" data-md-component="header"><nav class="md-header-nav md-grid"><div class="md-flex"><div class="md-flex__cell md-flex__cell--shrink"><a href="../../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo"><i class="md-icon"></i></a></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label></div><div class="md-flex__cell md-flex__cell--stretch"><div class="md-flex__ellipsis md-header-nav__title" data-md-component="title"><span class="md-header-nav__topic">Bioinformatics Documentation</span><span class="md-header-nav__topic">Launching a Personal GVL Server</span></div></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--search md-header-nav__button" for="__search"></label><div class="md-search" data-md-component="search" role="dialog"><label class="md-search__overlay" for="__search"></label><div class="md-search__inner" role="search"><form class="md-search__form" name="search"><input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active"> <label class="md-icon md-search__icon" for="__search"></label> <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">&#xE5CD;</button></form><div class="md-search__output"><div class="md-search__scrollwrap" data-md-scrollfix><div class="md-search-result" data-md-component="result"><div class="md-search-result__meta">Type to start searching</div><ol class="md-search-result__list"></ol></div></div></div></div></div></div><div class="md-flex__cell md-flex__cell--shrink"><div class="md-header-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div></div></div></nav></header><div class="md-container"><main class="md-main"><div class="md-main__inner md-grid" data-md-component="container"><div class="md-sidebar md-sidebar--primary" data-md-component="navigation"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--primary" data-md-level="0"><label class="md-nav__title md-nav__title--site" for="__drawer"><a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo"><i class="md-icon"></i></a>Bioinformatics Documentation</label><div class="md-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../.." title="Home" class="md-nav__link">Home</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2"><label class="md-nav__link" for="nav-2">Guides</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-2">Guides</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../../guides/bioinfo/" title="What is bioinformatics?" class="md-nav__link">What is bioinformatics?</a></li><li class="md-nav__item"><a href="../../../guides/galaxy/" title="The Galaxy platform" class="md-nav__link">The Galaxy platform</a></li></ul></nav></li><li class="md-nav__item md-nav__item--active md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked><label class="md-nav__link" for="nav-3">General Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-3">General Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../python_overview/python_overview/" title="Python Overview" class="md-nav__link">Python Overview</a></li><li class="md-nav__item"><a href="../../python_pandas_tidy_data/python_pandas_tidy_data/" title="Data tidying with Python and Pandas" class="md-nav__link">Data tidying with Python and Pandas</a></li><li class="md-nav__item"><a href="../../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">Data visualisation with Python</a></li><li class="md-nav__item"><a href="../../using_git/Using_Git/" title="Using Git and Github for revision control" class="md-nav__link">Using Git and Github for revision control</a></li><li class="md-nav__item"><a href="../../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">Introduction to HPC</a></li><li class="md-nav__item"><a href="../../unix/unix/" title="Introduction to Unix" class="md-nav__link">Introduction to Unix</a></li><li class="md-nav__item md-nav__item--active"><input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc"><label class="md-nav__link md-nav__link--active" for="__toc">Launching a Personal GVL Server</label><a href="./" title="Launching a Personal GVL Server" class="md-nav__link md-nav__link--active">Launching a Personal GVL Server</a><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#tutorial-overview" title="Tutorial Overview" class="md-nav__link">Tutorial Overview</a></li><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#what-is-the-gvl" title="What is the GVL?" class="md-nav__link">What is the GVL?</a></li><li class="md-nav__item"><a href="#what-is-nectar" title="What is NeCTAR?" class="md-nav__link">What is NeCTAR?</a></li></ul></nav></li><li class="md-nav__item"><a href="#section-1-access-the-nectar-dashboard" title="Section 1: Access the NeCTAR dashboard" class="md-nav__link">Section 1: Access the NeCTAR dashboard</a></li><li class="md-nav__item"><a href="#section-2-get-your-cloud-credentials" title="Section 2: Get your cloud credentials" class="md-nav__link">Section 2: Get your cloud credentials</a></li><li class="md-nav__item"><a href="#section-3-launch-your-personal-gvl-instance" title="Section 3: Launch your personal GVL instance" class="md-nav__link">Section 3: Launch your personal GVL instance</a></li><li class="md-nav__item"><a href="#section-4-access-your-gvl-instance" title="Section 4: Access your GVL instance" class="md-nav__link">Section 4: Access your GVL instance</a></li><li class="md-nav__item"><a href="#section-5-gvl-services" title="Section 5: GVL services" class="md-nav__link">Section 5: GVL services</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#galaxy" title="Galaxy" class="md-nav__link">Galaxy</a></li><li class="md-nav__item"><a href="#cloudman" title="CloudMan" class="md-nav__link">CloudMan</a></li><li class="md-nav__item"><a href="#lubuntu-desktop" title="Lubuntu Desktop" class="md-nav__link">Lubuntu Desktop</a></li><li class="md-nav__item"><a href="#ssh" title="SSH" class="md-nav__link">SSH</a></li><li class="md-nav__item"><a href="#jupyterhub" title="JupyterHub" class="md-nav__link">JupyterHub</a></li><li class="md-nav__item"><a href="#rstudio" title="RStudio" class="md-nav__link">RStudio</a></li><li class="md-nav__item"><a href="#public-html" title="Public HTML" class="md-nav__link">Public HTML</a></li><li class="md-nav__item"><a href="#pacbio-smrt-portal" title="PacBio SMRT Portal" class="md-nav__link">PacBio SMRT Portal</a></li></ul></nav></li><li class="md-nav__item"><a href="#section-6-shutting-your-machine-down" title="Section 6: Shutting your machine down" class="md-nav__link">Section 6: Shutting your machine down</a></li></ul></nav></li><li class="md-nav__item"><a href="../../galaxy_101/galaxy_101/" title="Introduction to Galaxy" class="md-nav__link">Introduction to Galaxy</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-9" type="checkbox" id="nav-3-9"><label class="md-nav__link" for="nav-3-9">Galaxy Workflows</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-3-9">Galaxy Workflows</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../galaxy-workflows/galaxy-workflows/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../galaxy-workflows/history_creation/" title="History Creation" class="md-nav__link">History Creation</a></li></ul></nav></li><li class="md-nav__item"><a href="../../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">Containerized Bioinformatics</a></li><li class="md-nav__item"><a href="../../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">Common Workflow Language for Bioinformatics</a></li><li class="md-nav__item"><a href="../../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">Command-line software development</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4"><label class="md-nav__link" for="nav-4">Genomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-4">Genomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" title="Introduction to Variant Calling" class="md-nav__link">Introduction to Variant Calling</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2"><label class="md-nav__link" for="nav-4-2">Variant Calling Advanced</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-2">Variant Calling Advanced</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3"><label class="md-nav__link" for="nav-4-3">RNA-seq DGE Basic</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-3">RNA-seq DGE Basic</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="Tuxedo Protocol Tutorial" class="md-nav__link">Tuxedo Protocol Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNA-seq DGE Advanced" class="md-nav__link">RNA-seq DGE Advanced</a></li><li class="md-nav__item"><a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq DGE Experimental Design" class="md-nav__link">RNA-seq DGE Experimental Design</a></li><li class="md-nav__item"><a href="../../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">10X single-cell RNA-seq analysis in R</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-7" type="checkbox" id="nav-4-7"><label class="md-nav__link" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../assembly/assembly-background/" title="Method Background" class="md-nav__link">Method Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../assembly/spades/" title="Microbial assembly: Spades" class="md-nav__link">Microbial assembly: Spades</a></li><li class="md-nav__item"><a href="../../pacbio/" title="Long read assembly" class="md-nav__link">Long read assembly</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5"><label class="md-nav__link" for="nav-5">Proteomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-5">Proteomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-1" type="checkbox" id="nav-5-1"><label class="md-nav__link" for="nav-5-1">Identifying proteins from mass spectrometry data</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-5-1">Identifying proteins from mass spectrometry data</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../proteomics_basic/proteomics_basic/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_galaxy/" title="Galaxy Background" class="md-nav__link">Galaxy Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">Data Formats Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_findcountitems_workflow/" title="Create find count items workflow" class="md-nav__link">Create find count items workflow</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">Protein Databases</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">Search Engines</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">Protein Prophet</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6"><label class="md-nav__link" for="nav-6">Molecular Modelling Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-6">Molecular Modelling Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">Molecular Dynamics - Introduction to cluster computing</a></li><li class="md-nav__item"><a href="../../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">Molecular Dynamics - Building input files, visualising the trajectory</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7"><label class="md-nav__link" for="nav-7">Protocols</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-7">Protocols</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly-protocol/" title="De Novo Genome Assembly for Illumina Data" class="md-nav__link">De Novo Genome Assembly for Illumina Data</a></li></ul></nav></li></ul></nav></div></div></div><div class="md-sidebar md-sidebar--secondary" data-md-component="toc"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#tutorial-overview" title="Tutorial Overview" class="md-nav__link">Tutorial Overview</a></li><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#what-is-the-gvl" title="What is the GVL?" class="md-nav__link">What is the GVL?</a></li><li class="md-nav__item"><a href="#what-is-nectar" title="What is NeCTAR?" class="md-nav__link">What is NeCTAR?</a></li></ul></nav></li><li class="md-nav__item"><a href="#section-1-access-the-nectar-dashboard" title="Section 1: Access the NeCTAR dashboard" class="md-nav__link">Section 1: Access the NeCTAR dashboard</a></li><li class="md-nav__item"><a href="#section-2-get-your-cloud-credentials" title="Section 2: Get your cloud credentials" class="md-nav__link">Section 2: Get your cloud credentials</a></li><li class="md-nav__item"><a href="#section-3-launch-your-personal-gvl-instance" title="Section 3: Launch your personal GVL instance" class="md-nav__link">Section 3: Launch your personal GVL instance</a></li><li class="md-nav__item"><a href="#section-4-access-your-gvl-instance" title="Section 4: Access your GVL instance" class="md-nav__link">Section 4: Access your GVL instance</a></li><li class="md-nav__item"><a href="#section-5-gvl-services" title="Section 5: GVL services" class="md-nav__link">Section 5: GVL services</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#galaxy" title="Galaxy" class="md-nav__link">Galaxy</a></li><li class="md-nav__item"><a href="#cloudman" title="CloudMan" class="md-nav__link">CloudMan</a></li><li class="md-nav__item"><a href="#lubuntu-desktop" title="Lubuntu Desktop" class="md-nav__link">Lubuntu Desktop</a></li><li class="md-nav__item"><a href="#ssh" title="SSH" class="md-nav__link">SSH</a></li><li class="md-nav__item"><a href="#jupyterhub" title="JupyterHub" class="md-nav__link">JupyterHub</a></li><li class="md-nav__item"><a href="#rstudio" title="RStudio" class="md-nav__link">RStudio</a></li><li class="md-nav__item"><a href="#public-html" title="Public HTML" class="md-nav__link">Public HTML</a></li><li class="md-nav__item"><a href="#pacbio-smrt-portal" title="PacBio SMRT Portal" class="md-nav__link">PacBio SMRT Portal</a></li></ul></nav></li><li class="md-nav__item"><a href="#section-6-shutting-your-machine-down" title="Section 6: Shutting your machine down" class="md-nav__link">Section 6: Shutting your machine down</a></li></ul></nav></div></div></div><div class="md-content"><article class="md-content__inner md-typeset"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/gvl_launch/gvl_launch.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a><style type="text/css">
    body{
        line-height: 2;
        font-size: 16px;
    }

    ol li{padding: 4px;}
    ul li{padding: 0px;}
    h4 {margin: 30px 0px 15px 0px;}

    div.code {
        font-family: "Courier New";
        border: 1px solid;
        border-color: #999999;
        background-color: #eeeeee;
        padding: 5px 10px;
        margin: 10px;
        border-radius: 5px;
        overflow: auto;
    }

    div.question {
        color: #666666;
        background-color: #e1eaf9;
        padding: 15px 25px;
        margin: 20px;
        font-size: 15px;
        border-radius: 20px;
    }

    div.question h4 {
        font-style: italic;
        margin: 10px 0px !important;
    }
</style>

<p><img src="../media/gvl_logo.jpg" height=100px align=right></p>
<h1 id="launching-a-personal-gvl-server-on-the-nectar-research-cloud">Launching a Personal GVL Server on the NeCTAR Research Cloud</h1>
<hr />
<h2 id="tutorial-overview">Tutorial Overview</h2>
<p>This document guides you to launch your own GVL Analysis platform, with Galaxy,
using your default NeCTAR allocation.</p>
<p>The tutorial will go over:</p>
<ol>
<li>Accessing the NeCTAR dashboard using Australian Access Federation (AAF)
   credentials</li>
<li>Getting your NeCTAR Research Cloud credentials</li>
<li>Launching the GVL image</li>
<li>Accessing your GVL instance</li>
<li>GVL services</li>
<li>Shutting your machine down</li>
</ol>
<hr />
<h2 id="background">Background</h2>
<h4 id="what-is-the-gvl">What is the GVL?</h4>
<p>The Genomics Virtual Laboratory (GVL) is a research platform that can be
deployed on the cloud.</p>
<p><img src="../media/gvl_dashboard.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>A private GVL server is a virtual machine running on the cloud and contains a
pre-installed suite of tools for performing bioinformatics analyses. It differs
from public GVL servers (such as the
<a href="http://galaxy-tut.genome.edu.au/">Galaxy Tutorial Server</a>,
<a href="https://usegalaxy.org.au/">Galaxy Australia</a>, and
<a href="http://galaxy-qld.genome.edu.au/">Galaxy Queensland</a>) by providing full
administrative access to the server, as well as the full suite of GVL services,
whereas public GVL servers provide restricted access for security reasons.
For example, public GVL servers do not provide access to the Ubuntu desktop,
the Linux command line or JupyterHub at present.</p>
<p>Accessing the GVL server is completely free on the Australian NeCTAR Research
Cloud, provided that you have a account with NeCTAR with allocated resources.</p>
<h4 id="what-is-nectar">What is NeCTAR?</h4>
<p>The <a href="http://nectar.org.au/">National eResearch Collaboration Tools and Resources project</a>
(NeCTAR) is an Australian programme that provides computing infrastructure
and services to Australian researchers. The NeCTAR Cloud allows us to
deploy virtual machines as a platform for research.</p>
<p>While it is possible to launch the GVL on <a href="https://aws.amazon.com/">Amazon</a>,
you may have to pay Amazon usage charges (the GVL software itself is free).</p>
<hr />
<h2 id="section-1-access-the-nectar-dashboard">Section 1: Access the NeCTAR dashboard</h2>
<ol>
<li>
<p>Login into the NeCTAR dashboard at
    <a href="https://dashboard.rc.nectar.org.au">dashboard.rc.nectar.org.au</a>.<br />
    Only members of the Australian Access Federation (AAF) can access Australian
    Research Cloud resources. Most Australian universities are members of the
    AAF, however, if you belong to an institution that is not a member of AAF,
    <a href="http://genome.edu.au/help/gvl-help">GVL Help</a> may be able to provide you
    with credentials. You also have the option of using a commercial cloud
    provider such as <a href="https://aws.amazon.com/">AWS</a> to host your server.</p>
</li>
<li>
<p>Choose your organisation from the list and login using your credentials.
    If you are from the University of Melbourne, choose 'The University of
    Melbourne' and not 'The University of Melbourne (with ECP)'.</p>
</li>
<li>
<p>Login with your institutional username and password. If this is the first
    time you have accessed the Australian Research Cloud, you must agree to some
    terms and conditions.</p>
<p>When you log in for the first time, you are automatically allocated a trial
project which lasts for 3 months. This trial project allows you to launch a
medium instance (with 2 cores) which is sufficient for launching a GVL
instance.</p>
<p><img src="../media/nectar_dashboard.png" height="400px" style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If you have projects which require more compute resources, you can apply
for more allocation <a href="https://dashboard.rc.nectar.org.au/allocation/">here</a>.</p>
</li>
</ol>
<hr />
<h2 id="section-2-get-your-cloud-credentials">Section 2: Get your cloud credentials</h2>
<p>Launching a GVL instance requires OpenStack API credentials from NeCTAR. These credentials allows the GVL launcher
to create a new GVL Virtual Machine on your behalf. Obtaining these credentials is a 5 step process.</p>
<p><img src="../media/openstack_creds_annotated_step_1.png" height="400px" /></p>
<ol>
<li>
<p>From the NeCTAR dashboard, on the left sidebar, navigate to<br />
    Project &gt; Compute &gt; Access &amp; Security</p>
</li>
<li>
<p>Click on the top <strong>'API Access'</strong> tab.</p>
</li>
<li>
<p>Click on the <strong>'Download OpenStack RC File'</strong> button on the top right.
    A file containing your credentials will be saved to your downloads folder.
    This file will be needed later in the launch process.</p>
</li>
<li>
<p>Next you must obtain your OpenStack password and record it securely for future use.
    If you have ever done this step before, you should reuse your previously saved
    password.</p>
<p>To obtain your OpenStack password, click on your username on the top right hand corner
and go to settings as shown below.</p>
<p><img src="../media/openstack_creds_annotated_step_2.png" /></p>
<p>Click on the reset password link.</p>
<p><img src="../media/openstack_creds_annotated_step_3.png" /></p>
</li>
<li>
<p>Once reset, your password will be displayed. Record this securely for all future
    GVL launches.</p>
<p><img src="../media/openstack_creds_annotated_step_4.png" /></p>
</li>
</ol>
<hr />
<h2 id="section-3-launch-your-personal-gvl-instance">Section 3: Launch your personal GVL instance</h2>
<ol>
<li>
<p>In a new browser tab, go to <a href="https://launch.usegalaxy.org">launch.usegalaxy.org</a></p>
</li>
<li>
<p>You will see the screen below. Select the first option from the list - "Genomics Virtual Lab".</p>
<p><img src="../media/cloudlaunch_gvl_step_1.png" height="400px" /> </p>
</li>
<li>
<p>You will be asked to login with your preferred social network account.</p>
<p><img src="../media/cloudlaunch_gvl_step_2.png" /> </p>
</li>
<li>
<p>Once logged in, perform the following steps.</p>
<p><img src="../media/cloudlaunch_gvl_step_3.png" style="display:block; margin-left: auto; margin-right:auto; margin-top:10px;"></p>
<ul>
<li>
<ol>
<li>Select NeCTAR for the question "On which cloud would you like to launch your appliance"</li>
</ol>
</li>
<li>
<ol>
<li>Click "load credentials from file" and provide file you downloaded in Section 2.3</li>
</ol>
</li>
<li>
<ol>
<li>Provide the OpenStack password you obtained in Section 2.5.</li>
</ol>
</li>
<li>
<ol>
<li>Click "Test and Use these Credentials". The Next button will now be activated.</li>
</ol>
</li>
</ul>
</li>
<li>
<p>Click the next button, and provide the following options.</p>
<p><img src="../media/cloudlaunch_gvl_step_4.png" /> </p>
<ul>
<li><strong>Password:</strong> Choose a strong password and remember it. This is the
  password you will use later to log into your instance.</li>
</ul>
</li>
<li>
<p>Optional advanced options<br />
    Toggle the 'Advanced cloudlaunch options' option to see more options. For
    this tutorial, it is not necessary to modify any of the advanced options.</p>
<ul>
<li><strong>Deployment name:</strong> You can override the name with a name of your choice.
  It is recommended you choose a unique name if you launch multiple instances.</li>
<li><strong>Instance type:</strong> Keep the default: Medium (2 vcpu / 8GB RAM)</li>
<li><strong>Root Volume Storage:</strong> Keep the default: Instance storage</li>
<li><strong>Placement Zone:</strong> You can also choose the region of where your server is
  hosted. If you are doing lots of data transfer, it may be beneficial to
  pick a location close to your physical location. More information about
  zones can be found <a href="https://support.rc.nectar.org.au/docs/availability-zones">here</a>.</li>
<li><strong>Key pair:</strong> Key pairs are SSH credentials that can be used to access
  your instance. You can create and import key pairs in the NeCTAR dashboard
  by navigating to Project &gt; Compute &gt; Access &amp; Security &gt; Key Pairs and
  creating or importing a key pair.</li>
</ul>
</li>
<li>
<p>Click <strong>'Launch'</strong> to launch a GVL.<br />
    The launch process takes 2-5 minutes to start the machine and another 5
    minutes to start and configure Galaxy.</p>
<p><img src="../media/cloudlaunch_gvl_wait.png" style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If an error occurs, or the launch does not complete in 10 minutes, navigate back to the launch page and try
selecting a different availability zone under the advanced cloudlaunch options -&gt; <strong>Placement Zone</strong> field.</p>
</li>
</ol>
<hr />
<h2 id="section-4-access-your-gvl-instance">Section 4: Access your GVL instance</h2>
<ol>
<li>
<p>Once your instance has finished launching, click on the Access address
    to access your GVL dashboard.</p>
<p><img src="../media/cloudlaunch_gvl_ready.png" style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If you accidentally closed the launch page, you can find your access address at any time
by logging back into <a href="https://launch.usegalaxy.org">launch.usegalaxy.org</a> and navigating
to the "My Appliances" section through the menu bar.</p>
</li>
<li>
<p>Explore the GVL dashboard.
    Have a read through of the services provided by the GVL.</p>
</li>
</ol>
<p><img src="../media/gvl_dashboard.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<hr />
<h2 id="section-5-gvl-services">Section 5: GVL services</h2>
<p>Listed below are short descriptions of the services the GVL provides.</p>
<h4 id="galaxy">Galaxy</h4>
<p><a href="https://galaxyproject.org/">Galaxy</a> is a web-based platform for computational
biomedical research. The GVL has a number of Galaxy tutorials available
<a href="http://genome.edu.au/learn">here</a>.</p>
<p><img src="../media/gvl_galaxy.png" height=400px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>To begin using Galaxy, register as a user by navigating to User &gt; Register
on the top Galaxy bar.</p>
<h4 id="cloudman">CloudMan</h4>
<p>CloudMan is a cloud manager that manages services on your GVL instance. Use
Cloudman to start and manage your Galaxy service and to add additional nodes
to your compute cluster (if you have enough resources).</p>
<p>You can log into CloudMan by using the username 'ubuntu' and your cluster
password.</p>
<p><img src="../media/cloudman.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>You can also shut down your instance (permanently) with CloudMan.</p>
<h4 id="lubuntu-desktop">Lubuntu Desktop</h4>
<p>Lubuntu is a lightweight desktop environment through which you can run desktop
applications on your virtual machine through your web browser. You can also
access the GVL command line utilities through the desktop.</p>
<p>You can log into Lubuntu Desktop using the username 'ubuntu' and your cluster
password.</p>
<p><img src="../media/lubuntu.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="ssh">SSH</h4>
<p>Secure Shell (SSH) is a network protocol that allows us to connect to a remotely
machine. You can login to your virtual machine remotely through an SSH client.</p>
<p>If you are using Windows, you will need to download an SSH client such as
<a href="http://www.chiark.greenend.org.uk/~sgtatham/putty/">PuTTY</a>. If you are using
OSX, open up a Terminal window.</p>
<p>If you are unfamiliar with the command line and UNIX, many
<a href="http://swcarpentry.github.io/shell-novice/">tutorials</a>
<a href="../unix/">on UNIX</a>
<a href="http://www.ee.surrey.ac.uk/Teaching/Unix/">can be</a>
<a href="http://www.doc.ic.ac.uk/~wjk/UnixIntro/">found online</a>.</p>
<p>You can ssh into your machine using the either the username 'ubuntu' or the
username 'researcher' and using your cluster password. It is recommended to use
the researcher account when you are doing your computational research and use
the ubuntu account when you need administrative powers (such as installing
software).</p>
<p><img src="../media/command_line.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="jupyterhub">JupyterHub</h4>
<p><a href="http://jupyter.org/">JupyterHub</a> is a web-based interactive computational
environment where you can combine code execution, text, mathematics, plots and
rich media into a single document. Currently, JupyterHub can connect to Python2
and Python3 kernals.</p>
<p>If you are unfamiliar with Python, there are many
<a href="http://swcarpentry.github.io/python-novice-inflammation/">tutorials</a>
<a href="../python_overview/python_overview/">available</a>
<a href="http://pythonprogramminglanguage.com/">online</a>.</p>
<p>You can log into JupyterHub with the username 'researcher' and your cluster
password.</p>
<p>You may need to install Python packages you intend to use via the command line
beforehand.</p>
<p><img src="../media/jupyterhub.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="rstudio">RStudio</h4>
<p>RStudio Server gives browser-based access to RStudio, the popular programming
and analysis environment for the R programming language. You can find out more
about RStudio <a href="https://www.rstudio.com/home/">here</a>, and the R programming
language <a href="https://www.r-project.org/">here</a>.</p>
<p>You can log into RStudio with the username 'researcher' and your cluster
password.</p>
<p><img src="../media/rstudio.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="public-html">Public HTML</h4>
<p>This is a shared web-accessible folder. Any files you place in the directory
<code>/home/researcher/public_html</code> will be publicly accessible.</p>
<h4 id="pacbio-smrt-portal">PacBio SMRT Portal</h4>
<p><a href="http://www.pacb.com/products-and-services/analytical-software/smrt-analysis/">PacBio's SMRT Portal</a> is an open source software suite for the analysis of single molecule, real-time sequencing data.</p>
<p>Before you use SMRT Portal, you need to firstly install it through the Admin console. <em>Please note PacBio recommends the use of a 16 core instance with 64GB of RAM (or higher) for this package.</em></p>
<p>To install SMRT Portal from the GVL dashboard:</p>
<ul>
<li>
<p>Click on 'Admin' in the top navigation bar.</p>
</li>
<li>
<p>Log in with the username 'ubuntu' and your cluster password. The screen will now show all the tools available to be installed.</p>
</li>
<li>
<p>Scroll down to 'SMRT Analysis', and click 'install'.</p>
</li>
<li>
<p>The install is complete when SMRT Portal is available as a tool on the GVL dashboard, and a green tick is displayed.</p>
</li>
</ul>
<p>When launching SMRT Portal for the first time, you will need to register yourself as a new user.</p>
<hr />
<h2 id="section-6-shutting-your-machine-down">Section 6: Shutting your machine down</h2>
<p>There are two ways to terminate your instance. Terminating your instance is
permanent and all data will be deleted (unless you have persistent
<a href="https://support.rc.nectar.org.au/docs/volumes">volume storage</a> which you will
need to apply for).</p>
<ol>
<li>
<p>Via Cloudman<br />
    Log into CloudMan by using the username 'ubuntu' and your cluster password.
    Click the <strong>Shut down...</strong> button under Cluster Controls.</p>
</li>
<li>
<p>Via the NeCTAR dashboard<br />
    Navigate to the <a href="https://dashboard.rc.nectar.org.au/project/instances/">Instances</a>
    page by navigating to Project &gt; Compute &gt; Instances on the left panel. Find
    the instance you want to terminate, and on the right-most column (Actions),
    click on the arrow button, and select <strong>Terminate Instance</strong>.</p>
</li>
</ol></article></div></div></main><footer class="md-footer"><div class="md-footer-nav"><nav class="md-footer-nav__inner md-grid"><a href="../../unix/unix/" title="Introduction to Unix" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev"><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-back md-footer-nav__button"></i></div><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Previous</span>Introduction to Unix</span></div></a><a href="../../galaxy_101/galaxy_101/" title="Introduction to Galaxy" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next"><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Next</span>Introduction to Galaxy</span></div><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i></div></a></nav></div><div class="md-footer-meta md-typeset"><div class="md-footer-meta__inner md-grid"><div class="md-footer-copyright"><div class="md-footer-copyright__highlight"><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a></div>powered by <a href="https://www.mkdocs.org">MkDocs</a> and <a href="https://squidfunk.github.io/mkdocs-material/">Material for MkDocs</a></div></div></div></footer></div><script src="../../../assets/javascripts/application.b260a35d.js"></script><script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script></body></html>