<!doctype html><html lang="en" class="no-js"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="lang:clipboard.copy" content="Copy to clipboard"><meta name="lang:clipboard.copied" content="Copied to clipboard"><meta name="lang:search.language" content="en"><meta name="lang:search.pipeline.stopwords" content="True"><meta name="lang:search.pipeline.trimmer" content="True"><meta name="lang:search.result.none" content="No matching documents"><meta name="lang:search.result.one" content="1 matching document"><meta name="lang:search.result.other" content="# matching documents"><meta name="lang:search.tokenizer" content="[\s\-]+"><link rel="shortcut icon" href="../../../assets/images/favicon.png"><meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.3.1"><title>Introduction to Variant Calling - Bioinformatics Documentation</title><link rel="stylesheet" href="../../../assets/stylesheets/application.4031d38b.css"><script src="../../../assets/javascripts/modernizr.74668098.js"></script><link href="https://fonts.gstatic.com" rel="preconnect" crossorigin><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=swap"><style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style><link rel="stylesheet" href="../../../assets/fonts/material-icons.css"><link rel="stylesheet" href="../../../extra.css"><script>window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-21794901-5", "www.melbournebioinformatics.org.au/tutorials/")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })</script><script async src="https://www.google-analytics.com/analytics.js"></script></head><body dir="ltr"><svg class="md-svg"><defs><svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg></defs></svg> <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off"> <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off"> <label class="md-overlay" data-md-component="overlay" for="__drawer"></label><a href="#introduction-to-variant-calling-using-galaxy" tabindex="1" class="md-skip">Skip to content </a><header class="md-header" data-md-component="header"><nav class="md-header-nav md-grid"><div class="md-flex"><div class="md-flex__cell md-flex__cell--shrink"><a href="../../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo"><i class="md-icon"></i></a></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label></div><div class="md-flex__cell md-flex__cell--stretch"><div class="md-flex__ellipsis md-header-nav__title" data-md-component="title"><span class="md-header-nav__topic">Bioinformatics Documentation</span><span class="md-header-nav__topic">Introduction to Variant Calling</span></div></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--search md-header-nav__button" for="__search"></label><div class="md-search" data-md-component="search" role="dialog"><label class="md-search__overlay" for="__search"></label><div class="md-search__inner" role="search"><form class="md-search__form" name="search"><input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active"> <label class="md-icon md-search__icon" for="__search"></label> <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">&#xE5CD;</button></form><div class="md-search__output"><div class="md-search__scrollwrap" data-md-scrollfix><div class="md-search-result" data-md-component="result"><div class="md-search-result__meta">Type to start searching</div><ol class="md-search-result__list"></ol></div></div></div></div></div></div><div class="md-flex__cell md-flex__cell--shrink"><div class="md-header-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div></div></div></nav></header><div class="md-container"><main class="md-main"><div class="md-main__inner md-grid" data-md-component="container"><div class="md-sidebar md-sidebar--primary" data-md-component="navigation"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--primary" data-md-level="0"><label class="md-nav__title md-nav__title--site" for="__drawer"><a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo"><i class="md-icon"></i></a>Bioinformatics Documentation</label><div class="md-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../.." title="Home" class="md-nav__link">Home</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2"><label class="md-nav__link" for="nav-2">Guides</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-2">Guides</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../../guides/bioinfo/" title="What is bioinformatics?" class="md-nav__link">What is bioinformatics?</a></li><li class="md-nav__item"><a href="../../../guides/galaxy/" title="The Galaxy platform" class="md-nav__link">The Galaxy platform</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3"><label class="md-nav__link" for="nav-3">General Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-3">General Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../python_overview/python_overview/" title="Python Overview" class="md-nav__link">Python Overview</a></li><li class="md-nav__item"><a href="../../python_pandas_tidy_data/python_pandas_tidy_data/" title="Data tidying with Python and Pandas" class="md-nav__link">Data tidying with Python and Pandas</a></li><li class="md-nav__item"><a href="../../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">Data visualisation with Python</a></li><li class="md-nav__item"><a href="../../using_git/Using_Git/" title="Using Git and Github for revision control" class="md-nav__link">Using Git and Github for revision control</a></li><li class="md-nav__item"><a href="../../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">Introduction to HPC</a></li><li class="md-nav__item"><a href="../../unix/unix/" title="Introduction to Unix" class="md-nav__link">Introduction to Unix</a></li><li class="md-nav__item"><a href="../../gvl_launch/gvl_launch/" title="Launching a Personal GVL Server" class="md-nav__link">Launching a Personal GVL Server</a></li><li class="md-nav__item"><a href="../../galaxy_101/galaxy_101/" title="Introduction to Galaxy" class="md-nav__link">Introduction to Galaxy</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-9" type="checkbox" id="nav-3-9"><label class="md-nav__link" for="nav-3-9">Galaxy Workflows</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-3-9">Galaxy Workflows</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../galaxy-workflows/galaxy-workflows/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../galaxy-workflows/history_creation/" title="History Creation" class="md-nav__link">History Creation</a></li></ul></nav></li><li class="md-nav__item"><a href="../../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">Containerized Bioinformatics</a></li><li class="md-nav__item"><a href="../../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">Common Workflow Language for Bioinformatics</a></li><li class="md-nav__item"><a href="../../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">Command-line software development</a></li></ul></nav></li><li class="md-nav__item md-nav__item--active md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4" checked><label class="md-nav__link" for="nav-4">Genomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-4">Genomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--active"><input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc"><label class="md-nav__link md-nav__link--active" for="__toc">Introduction to Variant Calling</label><a href="./" title="Introduction to Variant Calling" class="md-nav__link md-nav__link--active">Introduction to Variant Calling</a><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#overview" title="Overview" class="md-nav__link">Overview</a></li><li class="md-nav__item"><a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">Learning Objectives</a></li><li class="md-nav__item"><a href="#requirements" title="Requirements" class="md-nav__link">Requirements</a></li><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a></li><li class="md-nav__item"><a href="#1-preparation" title="1. Preparation" class="md-nav__link">1. Preparation</a></li><li class="md-nav__item"><a href="#2-quality-control" title="2. Quality Control" class="md-nav__link">2. Quality Control</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-take-a-look-at-the-fastq-file" title="1. Take a look at the FASTQ file" class="md-nav__link">1. Take a look at the FASTQ file</a></li><li class="md-nav__item"><a href="#2-assessing-read-quality-from-the-fastq-files" title="2. Assessing read quality from the FASTQ files" class="md-nav__link">2. Assessing read quality from the FASTQ files</a></li></ul></nav></li><li class="md-nav__item"><a href="#3-alignment-to-the-reference-fastq-to-bam" title="3. Alignment to the reference - (FASTQ to BAM)" class="md-nav__link">3. Alignment to the reference - (FASTQ to BAM)</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-align-the-reads-with-bwa" title="1. Align the reads with BWA" class="md-nav__link">1. Align the reads with BWA</a></li><li class="md-nav__item"><a href="#2-examine-the-alignment" title="2. Examine the alignment" class="md-nav__link">2. Examine the alignment</a></li><li class="md-nav__item"><a href="#3-assess-the-alignment-data" title="3. Assess the alignment data" class="md-nav__link">3. Assess the alignment data</a></li></ul></nav></li><li class="md-nav__item"><a href="#4-visualise-the-bam-file-with-igv" title="4. Visualise the BAM file with IGV" class="md-nav__link">4. Visualise the BAM file with IGV</a></li><li class="md-nav__item"><a href="#5-generate-a-pileup-file" title="5. Generate a pileup file" class="md-nav__link">5. Generate a pileup file</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-generate-a-pileup-file" title="1. Generate a pileup file" class="md-nav__link">1. Generate a pileup file</a></li><li class="md-nav__item"><a href="#2-filter-the-pileup-file" title="2. Filter the pileup file" class="md-nav__link">2. Filter the pileup file</a></li></ul></nav></li><li class="md-nav__item"><a href="#6-call-variants-with-freebayes" title="6. Call variants with FreeBayes" class="md-nav__link">6. Call variants with FreeBayes</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-call-variants-with-freebayes" title="1. Call variants with FreeBayes." class="md-nav__link">1. Call variants with FreeBayes.</a></li><li class="md-nav__item"><a href="#2-check-the-generated-list-of-variants" title="2. Check the generated list of variants" class="md-nav__link">2. Check the generated list of variants</a></li><li class="md-nav__item"><a href="#3-visualise-the-variants-and-compare-files" title="3. Visualise the variants and compare files" class="md-nav__link">3. Visualise the variants and compare files</a></li><li class="md-nav__item"><a href="#4-optional-filter-variants" title="4. Optional: filter variants" class="md-nav__link">4. Optional: filter variants</a></li></ul></nav></li><li class="md-nav__item"><a href="#7-further-steps" title="7. Further steps" class="md-nav__link">7. Further steps</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2"><label class="md-nav__link" for="nav-4-2">Variant Calling Advanced</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-2">Variant Calling Advanced</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3"><label class="md-nav__link" for="nav-4-3">RNA-seq DGE Basic</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-3">RNA-seq DGE Basic</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="Tuxedo Protocol Tutorial" class="md-nav__link">Tuxedo Protocol Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNA-seq DGE Advanced" class="md-nav__link">RNA-seq DGE Advanced</a></li><li class="md-nav__item"><a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq DGE Experimental Design" class="md-nav__link">RNA-seq DGE Experimental Design</a></li><li class="md-nav__item"><a href="../../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">10X single-cell RNA-seq analysis in R</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-7" type="checkbox" id="nav-4-7"><label class="md-nav__link" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../assembly/assembly-background/" title="Method Background" class="md-nav__link">Method Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../assembly/spades/" title="Microbial assembly: Spades" class="md-nav__link">Microbial assembly: Spades</a></li><li class="md-nav__item"><a href="../../pacbio/" title="Long read assembly" class="md-nav__link">Long read assembly</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5"><label class="md-nav__link" for="nav-5">Proteomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-5">Proteomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-1" type="checkbox" id="nav-5-1"><label class="md-nav__link" for="nav-5-1">Identifying proteins from mass spectrometry data</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-5-1">Identifying proteins from mass spectrometry data</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../proteomics_basic/proteomics_basic/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_galaxy/" title="Galaxy Background" class="md-nav__link">Galaxy Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">Data Formats Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_findcountitems_workflow/" title="Create find count items workflow" class="md-nav__link">Create find count items workflow</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">Protein Databases</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">Search Engines</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">Protein Prophet</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6"><label class="md-nav__link" for="nav-6">Molecular Modelling Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-6">Molecular Modelling Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">Molecular Dynamics - Introduction to cluster computing</a></li><li class="md-nav__item"><a href="../../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">Molecular Dynamics - Building input files, visualising the trajectory</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7"><label class="md-nav__link" for="nav-7">Protocols</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-7">Protocols</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly-protocol/" title="De Novo Genome Assembly for Illumina Data" class="md-nav__link">De Novo Genome Assembly for Illumina Data</a></li></ul></nav></li></ul></nav></div></div></div><div class="md-sidebar md-sidebar--secondary" data-md-component="toc"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#overview" title="Overview" class="md-nav__link">Overview</a></li><li class="md-nav__item"><a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">Learning Objectives</a></li><li class="md-nav__item"><a href="#requirements" title="Requirements" class="md-nav__link">Requirements</a></li><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a></li><li class="md-nav__item"><a href="#1-preparation" title="1. Preparation" class="md-nav__link">1. Preparation</a></li><li class="md-nav__item"><a href="#2-quality-control" title="2. Quality Control" class="md-nav__link">2. Quality Control</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-take-a-look-at-the-fastq-file" title="1. Take a look at the FASTQ file" class="md-nav__link">1. Take a look at the FASTQ file</a></li><li class="md-nav__item"><a href="#2-assessing-read-quality-from-the-fastq-files" title="2. Assessing read quality from the FASTQ files" class="md-nav__link">2. Assessing read quality from the FASTQ files</a></li></ul></nav></li><li class="md-nav__item"><a href="#3-alignment-to-the-reference-fastq-to-bam" title="3. Alignment to the reference - (FASTQ to BAM)" class="md-nav__link">3. Alignment to the reference - (FASTQ to BAM)</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-align-the-reads-with-bwa" title="1. Align the reads with BWA" class="md-nav__link">1. Align the reads with BWA</a></li><li class="md-nav__item"><a href="#2-examine-the-alignment" title="2. Examine the alignment" class="md-nav__link">2. Examine the alignment</a></li><li class="md-nav__item"><a href="#3-assess-the-alignment-data" title="3. Assess the alignment data" class="md-nav__link">3. Assess the alignment data</a></li></ul></nav></li><li class="md-nav__item"><a href="#4-visualise-the-bam-file-with-igv" title="4. Visualise the BAM file with IGV" class="md-nav__link">4. Visualise the BAM file with IGV</a></li><li class="md-nav__item"><a href="#5-generate-a-pileup-file" title="5. Generate a pileup file" class="md-nav__link">5. Generate a pileup file</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-generate-a-pileup-file" title="1. Generate a pileup file" class="md-nav__link">1. Generate a pileup file</a></li><li class="md-nav__item"><a href="#2-filter-the-pileup-file" title="2. Filter the pileup file" class="md-nav__link">2. Filter the pileup file</a></li></ul></nav></li><li class="md-nav__item"><a href="#6-call-variants-with-freebayes" title="6. Call variants with FreeBayes" class="md-nav__link">6. Call variants with FreeBayes</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#1-call-variants-with-freebayes" title="1. Call variants with FreeBayes." class="md-nav__link">1. Call variants with FreeBayes.</a></li><li class="md-nav__item"><a href="#2-check-the-generated-list-of-variants" title="2. Check the generated list of variants" class="md-nav__link">2. Check the generated list of variants</a></li><li class="md-nav__item"><a href="#3-visualise-the-variants-and-compare-files" title="3. Visualise the variants and compare files" class="md-nav__link">3. Visualise the variants and compare files</a></li><li class="md-nav__item"><a href="#4-optional-filter-variants" title="4. Optional: filter variants" class="md-nav__link">4. Optional: filter variants</a></li></ul></nav></li><li class="md-nav__item"><a href="#7-further-steps" title="7. Further steps" class="md-nav__link">7. Further steps</a></li></ul></nav></div></div></div><div class="md-content"><article class="md-content__inner md-typeset"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/variant_calling_galaxy_1/variant_calling_galaxy_1.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a><p>
<img src="../../../img/melbioinf_logo.png" alt="Melbourne Bioinformatics logo" align="left" width="40%"/><br />
<img src="../media/gvl_logo.jpg" alt="GVL logo" align="right" width="112"/>
</p>

<p><br></p>

<p><br></p>

<h1 id="introduction-to-variant-calling-using-galaxy">Introduction to Variant Calling using Galaxy</h1>
<h2 id="overview">Overview</h2>
<p>This tutorial is designed to introduce the tools, data types and workflow of variant detection. We will align reads to the genome, look for differences between reads and reference genome sequence, and filter the detected genomic variation manually to understand the computational basis of variant calling.</p>
<p>We cover the concepts of detecting small variants (SNVs and indels) in human genomic DNA using a small set of reads from chromosome 22.</p>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of the course, you will be able to:</p>
<ul>
<li>Work with the FASTQ format and base quality scores</li>
<li>Align reads to generate a BAM file and subsequently generate a pileup file</li>
<li>Run the FreeBayes variant caller to find SNVs and indels</li>
<li>Visualise BAM files using the Integrative Genomics Viewer (IGV) and identify likely SNVs and indels by eye</li>
</ul>
<h2 id="requirements">Requirements</h2>
<p>This workshop uses Galaxy as a platform. It is recommended that participants who have not used Galaxy before either sign up for our Intro to GVL workshop, or work through this tutorial themselves beforehand.</p>
<p>This is a hands-on workshop and attendees are required to bring their own laptops.</p>
<hr />
<h2 id="background">Background</h2>
<p>Some background reading material - <a href="https://www.google.com/url?q=https://docs.google.com/document/pub?id%3D1NfythYcSrkQwldGMrbHRKLRORFFn-WnBm3gOMHwIgmE&amp;sa=D&amp;usg=AFQjCNE7C6wmK6Fiu-_ZJhc0RSBaxFSRbg">background</a></p>
<p><strong>Where is the data in this tutorial from?</strong></p>
<p>The workshop is based on analysis of short read data from the exome of chromosome 22 of a single human individual. There are one million 76bp reads in the dataset, produced on an Illumina GAIIx from exome-enriched DNA. This data was generated as part of the <a href="http://www.1000genomes.org/">1000 genomes</a> Genomes project.</p>
<hr />
<h2 id="1-preparation">1. Preparation</h2>
<ol>
<li><strong>Make sure you have an instance of Galaxy ready to go.</strong><ul>
<li>For example, you can use the <a href="https://usegalaxy.org.au/">Galaxy Australia server</a>.</li>
</ul>
</li>
<li><strong>Create a new history for this tutorial.</strong><ul>
<li>In the history pane, click on the cog icon at the top right.</li>
<li>Click <em>Create New</em>.</li>
<li>Click on <em>Unnamed history</em> and re-name it.</li>
</ul>
</li>
<li>
<p><strong>Import data for the tutorial.</strong></p>
<ul>
<li>In this case, we are uploading a <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> file.</li>
<li><strong>Method 1</strong><br />
<em>Paste/Fetch data from a URL to Galaxy.</em><ol>
<li>In the Galaxy tools panel (left), click on <code>Get Data</code> and choose <code>Upload File</code>.</li>
<li>Click <em>Paste/Fetch data</em> and paste the following URL into the box
<div class="code">
https://swift.rc.nectar.org.au:8888/v1/AUTH_a3929895f9e94089ad042c9900e1ee82/VariantDet_BASIC/NA12878.GAIIx.exome_chr22.1E6reads.76bp.fastq
<br>
</div></li>
<li>Select <em>Type</em> as <strong>fastqsanger</strong> and click <em>Start</em>.</li>
<li>Once the upload status turns <em>green</em>, it means the upload is complete. You should now be able to see the file in the Galaxy history panel (right).</li>
<li>The dataset will have a very long name, as it's named after the full URL we got it from. Optionally, once the file is in your History, click the pencil icon in the upper-right corner of the green dataset box, then select the <em>Name</em> box and give the file a shorter name by removing the URL. Then click <em>Save</em>.</li>
</ol>
</li>
</ul>
<p>Alternatively, if you have a local file to upload (<em>For the purpose of this tutorial we can stick with the option above</em>):</p>
<ul>
<li><strong>Method 2</strong><br />
<em>Upload data to Galaxy.</em><ol>
<li>In the Galaxy tools panel (left), click on <code>Get Data</code> and choose <code>Upload File</code>.</li>
<li>From <em>Choose local file</em> select the downloaded FASTQ file. Select <em>Type</em> as <strong>fastqsanger</strong> and click <em>Start</em>.</li>
<li>Once the upload status turns <em>green</em>, it means the upload is complete. You should now be able to see the file in the Galaxy history panel (right).</li>
</ol>
</li>
</ul>
</li>
</ol>
<p>Summary:<br />
So far, we have started a Galaxy instance, got hold of our data and uploaded it to
the Galaxy instance.<br />
Now we are ready to perform our analysis.</p>
<hr />
<h2 id="2-quality-control">2. Quality Control</h2>
<p>The first step is to evaluate the quality of the raw sequence data. If the quality is poor, then adjustments can be made - e.g. trimming the short reads, or adjusting your expectations of the final outcome!</p>
<h4 id="1-take-a-look-at-the-fastq-file">1. Take a look at the FASTQ file</h4>
<p>Click on the eye icon to the top right of the FASTQ file to view the a snippet of the file.</p>
<p>Note that each read is represented by 4 lines:
    * read identifier
    * short read sequence
    * separator
    * short read sequence quality scores</p>
<pre><code class="text">e.g.
identifier:    @61CC3AAXX100125:7:72:14903:20386/1
read sequence: TTCCTCCTGAGGCCCCACCCACTATACATCATCCCTTCATGGTGAGGGAGACTTCAGCCCTCAATGCCACCTTCAT
separator:     +
quality score: ?ACDDEFFHBCHHHHHFHGGCHHDFDIFFIFFIIIIHIGFIIFIEEIIEFEIIHIGFIIIIIGHCIIIFIID?@&lt;6
</code></pre>

<p>For more details see <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a>.</p>
<h4 id="2-assessing-read-quality-from-the-fastq-files">2. Assessing read quality from the FASTQ files</h4>
<ol>
<li>From the Galaxy tools panel, in the search box at the top, type in "FastQC".</li>
<li>Click on <ss>FastQC</ss></li>
</ol>
<p><br>The input FASTQ file will be selected by default. Keep the other defaults and click execute.</p>
<blockquote>
<p><img src="../media/tips.png" alt="Tip" height="42" width="42"/>
   Tip:
   Note the batch processing interface of Galaxy:<br />
   <ul>
   <li> grey = waiting in queue<br />
   <li> yellow = running<br />
   <li> green = finished<br />
   <li> red = tried to run and failed
   </ul></p>
</blockquote>
<p>When the job has finished, click on the eye icon to view the newly generated data (in this case a set of quality metrics for the FASTQ data). This will be a file called <em>FastQC on data 1: Web page</em>.
Look at the various quality scores. The data looks pretty good - <em>high Per base sequence quality</em> (avg. above 30).</p>
<hr />
<h2 id="3-alignment-to-the-reference-fastq-to-bam">3. Alignment to the reference - (FASTQ to BAM)</h2>
<p>The basic process here to map individual reads - from the input sample FASTQ file - to a matching region on the reference genome.</p>
<h4 id="1-align-the-reads-with-bwa">1. Align the reads with BWA</h4>
<p>We will map (align) the reads with the <a href="http://bio-bwa.sourceforge.net/">BWA</a> tool to the human reference genome. For this tutorial, use <em>Human reference genome 19 (hg19)</em> - this is <a href="http://genome.ucsc.edu/cgi-bin/hgTracks">hg19 from UCSC</a>.</p>
<ol>
<li>
<p>Align the reads [3-5mins]: from the Galaxy tools panel, search for <ss>Map with BWA-MEM</ss><br>
   From the options:<br></p>
<ul>
<li><em>Will you selection a reference genome...:</em> Use a built-in genome index</li>
<li><em>Using reference genome:</em> set to hg19</li>
<li><em>Single or Paired-end reads:</em> set to Single</li>
<li>Make sure your fastq file is the input file.</li>
<li>Keep other options as default and click execute.</li>
</ul>
<p><em>Note: This is the longest step in the workshop and will take a few minutes, possibly more depending on how many people are also scheduling mappings</em></p>
</li>
<li>
<p>Sort the BAM file: from the Galaxy tools panel, search for <ss>SortSam</ss>
    <br>From the options:<br></p>
<ul>
<li>Set the input file to be the output BAM file from the previous step.</li>
<li><em>Sort by:</em> set to Coordinate</li>
<li>Keep other options as default and click execute</li>
</ul>
</li>
</ol>
<h4 id="2-examine-the-alignment">2. Examine the alignment</h4>
<ol>
<li>
<p>To examine the output sorted BAM file, we need to first convert it into readable <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM</a> format.
   From the Galaxy tools panel, search for <ss>BAM-to-SAM</ss>
   <br>From the options:<br></p>
<ul>
<li><em>BAM File to Convert:</em> select your sorted BAM file</li>
<li>Keep all options as default and click execute</li>
</ul>
</li>
<li>
<p>Examine the generated Sequence Alignment Map (SAM) file.</p>
<ul>
<li>Click the eye icon in the History pane next to the newly generated file</li>
<li>Familiarise yourself with the <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM</a> format</li>
<li>Note that some reads have mapped to non-chr22 chromosomes (see column 3).</li>
</ul>
</li>
</ol>
<p>This is the essence of alignment algorithms - the aligner does the best it can, but because of compromises in accuracy vs performance and repetitive sequences in the genome, not all the reads will necessarily align to the ‘correct’ sequence or could this be suggesting the presence of a structural variant?</p>
<blockquote>
<p><img src="../media/tips.png" alt="Tip" height="42" width="42"/>
Tip:
Galaxy auto-generates a name for all outputs. Therefore, it is advisable to choose a more meaningful name to these outputs.</p>
<p>This can be done as follows:</p>
<p>Click on the pencil icon (edit attributes) and change Name e.g. Sample.bam or Sample.sam or Sample.sorted.bam etc.</p>
</blockquote>
<h4 id="3-assess-the-alignment-data">3. Assess the alignment data</h4>
<p>We can generate some mapping statistics from the BAM file to assess the quality of our alignment.</p>
<ol>
<li>
<p>Run IdxStats. From the Galaxy tools panel, search for the tool <ss>IdxStats</ss></p>
<ul>
<li>Select the sorted BAM file as input.</li>
<li>Keep other options as default and click execute.</li>
</ul>
<p>IdxStats generates a tab-delimited output with four columns.
Each line consists of a reference sequence name (e.g. a chromosome),
reference sequence length, number of mapped reads and number of placed but
unmapped reads.</p>
<p>We can see that most of the reads are aligning to chromosome 22 as expected.</p>
</li>
<li>
<p>Run Flagstat. From the Galaxy tools panel, search for <ss>Flagstat</ss></p>
<ul>
<li>Select the sorted BAM file as input.</li>
<li>Keep other options as default and click execute.</li>
</ul>
<p>Note that in this case the statistics are not very informative. This is because the dataset has been generated for this workshop and much of the noise has been removed (and in fact we just removed a lot more noise in the previous step); also we are using single ended read data rather than paired-end so some of the metrics are not relevant.</p>
</li>
</ol>
<hr />
<h2 id="4-visualise-the-bam-file-with-igv">4. Visualise the BAM file with IGV</h2>
<p>To visualise the alignment data:</p>
<ol>
<li>Click on the sorted BAM file dataset in the History panel.</li>
<li>Click on "Display with IGV <strong>web current</strong>". This should download a .jnlp
    Java Web Start file to your computer. Open this file to run IGV.
    (You will need Java installed on your computer to run IGV). NOTE: If IGV is already open on your computer, you can click "<strong>local</strong>" instead of "web current", and this will open the BAM file in your current IGV session.</li>
<li>Once IGV opens, it will show you the BAM file. This may take a bit of time as the data is downloaded.</li>
<li>Our reads for this tutorial are from chromosome 22, so select <strong>chr22</strong> from the second
    drop box under the toolbar. Zoom in to view alignments of reads to the reference genome.</li>
</ol>
<p><a href="../media/igv1.jpg"><img src="../media/igv1.jpg" alt="IGV view 1" width="640px" style="display:block; margin-left: auto; margin-right:auto;"/></a></p>
<p>Try looking at region <code>chr22:36,006,744-36,007,406</code></p>
<p>Can you see a few variants?  </p>
<p><a href="../media/igv_mb.jpg"><img src="../media/igv_mb.jpg" alt="IGV view 1" width="640px" style="display:block; margin-left: auto; margin-right:auto;"/></a></p>
<p>Don't close IGV yet as we'll be using it later.</p>
<hr />
<h2 id="5-generate-a-pileup-file">5. Generate a pileup file</h2>
<p>A pileup is essentially a column-wise representation of the aligned reads - at the base level - to the reference. The pileup file summarises all data from the reads at each genomic region that is covered by at least one read. Each row of the pileup file gives similar information to a single vertical column of reads in the IGV view.</p>
<p>The current generation of variant calling tools do not output pileup files, and you don't need to do this section in order to use FreeBayes in the next section. However, a pileup file is a good illustration of the evidence the variant caller is looking at internally, and we will produce one to see this evidence.</p>
<h4 id="1-generate-a-pileup-file">1. Generate a pileup file</h4>
<p>From the Galaxy tools panel, search for <ss>Generate pileup</ss><br>
From the options:</p>
<ul>
<li><em>Call consensus according to MAQ model:</em> set to Yes<br>
  This generates a called 'consensus base' for each chromosomal position.
  This would allow us to use this pileup directly for variant detection, if we wanted.</li>
<li>Keep other options as default and click execute</li>
</ul>
<p>Galaxy tries to assign a datatype attribute to every output file. In this case, you'll need to manually set the datatype to <em>pileup</em>.</p>
<ol>
<li>First, rename the output to something more meaningful by clicking on the pencil icon</li>
<li>To change the datatype, click on the Datatype link from the top tab while you're editing attributes.</li>
<li>For downstream processing we want to tell Galaxy that this is a <em>pileup</em> file. From the drop-down, select Pileup and click save.</li>
</ol>
<blockquote>
<p><img src="../media/tips.png" alt="Tip" height="42" width="42"/>
 Tip:
The pileup file we generated has 10 columns:<br>
* 1. chromosome<br>
* 2. position<br>
* 3. current reference base<br>
* 4. consensus base from the mapped reads<br>
* 5. consensus quality<br>
* 6. SNV quality<br>
* 7. maximum mapping quality<br>
* 8. coverage<br>
* 9. bases within reads<br>
* 10. quality values<br></p>
<p>Further information on (9):<br>
Each character represents one of the following (the longer this string, higher the coverage):
  <ul>
  <li> . = match on forward strand for that base
  <li> , = match on reverse strand
  <li> ACGTN = mismatch on forward
  <li> acgtn = mismatch on reverse
  <li> +[0-9]+[ACGTNacgtn]+' = insertion between this reference position and the next
  <li> -[0-9]+[ACGTNacgtn]+' = deletion between this reference position and the next
  <li> ^ = start of read
  <li> $ = end of read
  <li> BaseQualities = one character per base in ReadBases, ASCII encoded Phred scores
  </ul></p>
</blockquote>
<h4 id="2-filter-the-pileup-file">2. Filter the pileup file</h4>
<p>If you click the eye icon to view the contents of your pileup file, you'll see that the visible rows of the file aren't very interesting as they are outside chromosome 22 and have very low coverage. Let's filter to regions with coverage of at least 10 reads.</p>
<p>From the Galaxy tools panel, search for <ss>Filter Pileup</ss><br>
From the options:</p>
<ul>
<li><em>which contains:</em> set to Pileup with ten columns (with consensus).
  This is to match the <em>Call consensus according to MAQ model</em> option we selected earlier -
  the filter tool needs to know to expect the 10-column rather than 6-column format.</li>
<li><em>Do not report positions with coverage lower than:</em> set to 10</li>
</ul>
<p>Try this filtering step two ways:</p>
<ul>
<li>First, set the parameter <em>Only report variants?</em> to <strong>No</strong>. This will give you all locations with a coverage of at least 10 and sufficient read quality. This is similar to the information you see when you look at IVG: each pilup row corresponds to a column of aligned bases at one genomic location.</li>
<li>Then, repeat the step but set <em>Only report variants?</em> to <strong>Yes</strong>. This will effectively do variant calling: it will give you only locations that have some evidence that there might be a variant present. This variant calling is not very stringent, so you will still get lots of rows. We could filter further to, for instance, variants with high quality scores.</li>
</ul>
<p>Examine your two pileup files and understand the difference between them. Which coordinates are present in each? What do the bases look like in one compared to the other? Compare the variant quality score (in column 6) to the bases listed on each row.</p>
<h2 id="6-call-variants-with-freebayes">6. Call variants with FreeBayes</h2>
<p>FreeBayes is a Bayesian variant caller which assesses the likelihood of each possible genotype for each position in the reference genome, given the observed reads at that position, and reports back the list of possible variants. We look at it in more detail in the <a href="../var_detect_advanced/var_detect_advanced">Advanced Variant Calling</a> tutorial.</p>
<h4 id="1-call-variants-with-freebayes">1. Call variants with FreeBayes.</h4>
<p>In the tool panel search for <ss>FreeBayes</ss>.</p>
<ul>
<li>Select your sorted BAM file as input, and select the correct reference genome.</li>
<li>Under <em>Choose parameter selection level</em>, select "Simple diploid calling with filtering and coverage". This will consider only aligned reads with sufficient mapping quality and base quality. You can see the exact parameters this sets by scrolling down in the main Galaxy window to the Galaxy FreeBayes documentation section on <em>Galaxy-specific options</em>.</li>
<li><em>Execute</em> FreeBayes. (This may take a while.)</li>
</ul>
<h4 id="2-check-the-generated-list-of-variants">2. Check the generated list of variants</h4>
<ul>
<li>Click the eye icon to examine the VCF file contents. The VCF format is described below - make sure you can identify the header rows and the data, and understand the important columns.</li>
<li>How many variants exactly are in your list? (Hint: you can look at the number of lines in the dataset, listed in the green box in the History, but remember the top lines are header lines.)</li>
<li>What sort of quality scores do your variants have?</li>
</ul>
<blockquote>
<p><img src="../media/tips.png" alt="Tip" height="42" width="42"/>
FreeBayes, like most variant callers, produces a <a href="http://vcftools.sourceforge.net/specs.html">Variant Call Format (VCF)</a> file.</p>
<p>VCF consists of a header section and a data section. The header section has some information about the file and the parameters used to produce it. The header also specifies what information is stored in the INFO, FILTER and FORMAT columns, as this is different for different variant callers.</p>
<p>The data section has several columns. For this tutorial, you should concentrate on CHROM, POS, REF, ALT and QUAL.
CHROM and POS describe the variant's genomic location, REF and ALT describe the variant's nucleotides relative to the reference, and QUAL is a quality score giving FreeBayes' confidence in the correctness of this variant call.</p>
<p>The columns in more detail are:</p>
</blockquote>
<table>
<thead>
<tr>
<th>Col</th>
<th>Field</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>CHROM</td>
<td>Chromosome name</td>
</tr>
<tr>
<td>2</td>
<td>POS</td>
<td>1-based position. For an indel, this is the position preceding the indel.</td>
</tr>
<tr>
<td>3</td>
<td>ID</td>
<td>Variant identifier (optional). Usually the dbSNP rsID.</td>
</tr>
<tr>
<td>4</td>
<td>REF</td>
<td>Reference sequence at POS involved in the variant. For a SNP, it is a single base.</td>
</tr>
<tr>
<td>5</td>
<td>ALT</td>
<td>Comma delimited list of alternative sequence(s) seen in our reads.</td>
</tr>
<tr>
<td>6</td>
<td>QUAL</td>
<td>Phred-scaled probability of all samples being homozygous reference.</td>
</tr>
<tr>
<td>7</td>
<td>FILTER</td>
<td>Semicolon delimited list of filters that the variant fails to pass.</td>
</tr>
<tr>
<td>8</td>
<td>INFO</td>
<td>Semicolon delimited list of variant information.</td>
</tr>
<tr>
<td>9</td>
<td>FORMAT</td>
<td>Colon delimited list of the format of individual genotypes in the following fields.</td>
</tr>
<tr>
<td>10+</td>
<td>Sample(s)</td>
<td>Individual genotype information defined by FORMAT.</td>
</tr>
</tbody>
</table>
<blockquote>
<p>For even more detail on VCF files, you can look at the <a href="http://vcftools.sourceforge.net/specs.html"><em>VCF format specification</em></a>.</p>
</blockquote>
<h4 id="3-visualise-the-variants-and-compare-files">3. Visualise the variants and compare files</h4>
<ul>
<li>Open the VCF file in IGV using the dataset's <em>display in IGV local</em> link (using the <em>web current</em> link will open IGV again, and using <em>local</em> should use your already-running IGV). This will give an annotation track in IGV to visualise where variants have been called. Compare it to your BAM file.</li>
<li>Take a look again at the same region as earlier: <code>chr22:36,006,744-36,007,406</code></li>
<li>Try comparing to the corresponding location in the pileup file. You can filter to the same window as we just opened in IGV with the tool <code>Filter and Sort &gt; Filter</code>. Choose your previously-filtered pileup file as input, and set the filter condition to <code>c1=="chr22" and c2 &gt; 36006744 and c2 &lt; 36007406</code>.</li>
</ul>
<h4 id="4-optional-filter-variants">4. Optional: filter variants</h4>
<p>See if you can work out how to filter your VCF file to variants with quality scores greater than 50. You can use the <code>Filter and Sort: Filter</code> tool we used above.</p>
<h2 id="7-further-steps">7. Further steps</h2>
<p>We've seen how to:</p>
<ul>
<li>Align the raw data (sequence reads) to a reference genome</li>
<li>Generate variant calls from aligned reads</li>
<li>Interpret the various file formats used in storing reads, alignments and variant calls</li>
<li>Visualise the data using IGV</li>
</ul>
<p>For real variant calling, you will probably want to carry out clean-up steps on your BAM file to improve the quality of the calls, and do further filtering and selection on the resulting variants.</p>
<p>We look at some further steps in the <a href="../var_detect_advanced/var_detect_advanced">Advanced Variant Calling</a> tutorial.</p>
<hr /></article></div></div></main><footer class="md-footer"><div class="md-footer-nav"><nav class="md-footer-nav__inner md-grid"><a href="../../bionitio/bionitio/" title="Command-line software development" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev"><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-back md-footer-nav__button"></i></div><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Previous</span>Command-line software development</span></div></a><a href="../../var_detect_advanced/var_detect_advanced/" title="Tutorial" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next"><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Next</span>Tutorial</span></div><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i></div></a></nav></div><div class="md-footer-meta md-typeset"><div class="md-footer-meta__inner md-grid"><div class="md-footer-copyright"><div class="md-footer-copyright__highlight"><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a></div>powered by <a href="https://www.mkdocs.org">MkDocs</a> and <a href="https://squidfunk.github.io/mkdocs-material/">Material for MkDocs</a></div></div></div></footer></div><script src="../../../assets/javascripts/application.b260a35d.js"></script><script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script></body></html>