<!doctype html><html lang="en" class="no-js"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="lang:clipboard.copy" content="Copy to clipboard"><meta name="lang:clipboard.copied" content="Copied to clipboard"><meta name="lang:search.language" content="en"><meta name="lang:search.pipeline.stopwords" content="True"><meta name="lang:search.pipeline.trimmer" content="True"><meta name="lang:search.result.none" content="No matching documents"><meta name="lang:search.result.one" content="1 matching document"><meta name="lang:search.result.other" content="# matching documents"><meta name="lang:search.tokenizer" content="[\s\-]+"><link rel="shortcut icon" href="../../../assets/images/favicon.png"><meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.3.1"><title>Introduction to Unix - Bioinformatics Documentation</title><link rel="stylesheet" href="../../../assets/stylesheets/application.4031d38b.css"><script src="../../../assets/javascripts/modernizr.74668098.js"></script><link href="https://fonts.gstatic.com" rel="preconnect" crossorigin><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=swap"><style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style><link rel="stylesheet" href="../../../assets/fonts/material-icons.css"><link rel="stylesheet" href="../../../extra.css"><script>window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-21794901-5", "www.melbournebioinformatics.org.au/tutorials/")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })</script><script async src="https://www.google-analytics.com/analytics.js"></script></head><body dir="ltr"><svg class="md-svg"><defs><svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg></defs></svg> <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off"> <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off"> <label class="md-overlay" data-md-component="overlay" for="__drawer"></label><a href="#introduction-to-unix" tabindex="1" class="md-skip">Skip to content </a><header class="md-header" data-md-component="header"><nav class="md-header-nav md-grid"><div class="md-flex"><div class="md-flex__cell md-flex__cell--shrink"><a href="../../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo"><i class="md-icon"></i></a></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label></div><div class="md-flex__cell md-flex__cell--stretch"><div class="md-flex__ellipsis md-header-nav__title" data-md-component="title"><span class="md-header-nav__topic">Bioinformatics Documentation</span><span class="md-header-nav__topic">Introduction to Unix</span></div></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--search md-header-nav__button" for="__search"></label><div class="md-search" data-md-component="search" role="dialog"><label class="md-search__overlay" for="__search"></label><div class="md-search__inner" role="search"><form class="md-search__form" name="search"><input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active"> <label class="md-icon md-search__icon" for="__search"></label> <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">&#xE5CD;</button></form><div class="md-search__output"><div class="md-search__scrollwrap" data-md-scrollfix><div class="md-search-result" data-md-component="result"><div class="md-search-result__meta">Type to start searching</div><ol class="md-search-result__list"></ol></div></div></div></div></div></div><div class="md-flex__cell md-flex__cell--shrink"><div class="md-header-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div></div></div></nav></header><div class="md-container"><main class="md-main"><div class="md-main__inner md-grid" data-md-component="container"><div class="md-sidebar md-sidebar--primary" data-md-component="navigation"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--primary" data-md-level="0"><label class="md-nav__title md-nav__title--site" for="__drawer"><a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo"><i class="md-icon"></i></a>Bioinformatics Documentation</label><div class="md-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../.." title="Home" class="md-nav__link">Home</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2"><label class="md-nav__link" for="nav-2">Guides</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-2">Guides</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../../guides/bioinfo/" title="What is bioinformatics?" class="md-nav__link">What is bioinformatics?</a></li><li class="md-nav__item"><a href="../../../guides/galaxy/" title="The Galaxy platform" class="md-nav__link">The Galaxy platform</a></li></ul></nav></li><li class="md-nav__item md-nav__item--active md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked><label class="md-nav__link" for="nav-3">General Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-3">General Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../python_overview/python_overview/" title="Python Overview" class="md-nav__link">Python Overview</a></li><li class="md-nav__item"><a href="../../python_pandas_tidy_data/python_pandas_tidy_data/" title="Data tidying with Python and Pandas" class="md-nav__link">Data tidying with Python and Pandas</a></li><li class="md-nav__item"><a href="../../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">Data visualisation with Python</a></li><li class="md-nav__item"><a href="../../using_git/Using_Git/" title="Using Git and Github for revision control" class="md-nav__link">Using Git and Github for revision control</a></li><li class="md-nav__item"><a href="../../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">Introduction to HPC</a></li><li class="md-nav__item md-nav__item--active"><input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc"><label class="md-nav__link md-nav__link--active" for="__toc">Introduction to Unix</label><a href="./" title="Introduction to Unix" class="md-nav__link md-nav__link--active">Introduction to Unix</a><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#overview" title="Overview" class="md-nav__link">Overview</a></li><li class="md-nav__item"><a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">Learning Objectives</a></li><li class="md-nav__item"><a href="#requirements" title="Requirements" class="md-nav__link">Requirements</a></li><li class="md-nav__item"><a href="#introduction" title="Introduction" class="md-nav__link">Introduction</a></li><li class="md-nav__item"><a href="#topic-1-remote-log-in" title="Topic 1: Remote log in" class="md-nav__link">Topic 1: Remote log in</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#connecting-to-a-unix-computer" title="Connecting to a Unix computer" class="md-nav__link">Connecting to a Unix computer</a></li><li class="md-nav__item"><a href="#exercises" title="Exercises" class="md-nav__link">Exercises</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#11-when-youve-logged-into-the-unix-server-run-the-following-commands-and-see-what-they-do" title="1.1) When you've logged into the Unix server, run the following commands and see what they do:" class="md-nav__link">1.1) When you've logged into the Unix server, run the following commands and see what they do:</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item"><a href="#topic-2-exploring-your-home-directory" title="Topic 2: Exploring your home directory" class="md-nav__link">Topic 2: Exploring your home directory</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#21-use-the-ls-command-to-list-the-files-in-your-home-directory-how-many-files-are-there" title="2.1) Use the ls command to list the files in your home directory.  How many files are there?" class="md-nav__link">2.1) Use the ls command to list the files in your home directory.  How many files are there?</a></li><li class="md-nav__item"><a href="#22-what-flag-might-you-use-to-display-all-files-with-the-ls-command-how-many-files-are-really-there" title="2.2) What flag might you use to display all files with the ls command?  How many files are really there?" class="md-nav__link">2.2) What flag might you use to display all files with the ls command?  How many files are really there?</a></li><li class="md-nav__item"><a href="#23-what-is-the-full-path-name-of-your-home-directory" title="2.3) What is the full path name of your home directory?" class="md-nav__link">2.3) What is the full path name of your home directory?</a></li><li class="md-nav__item"><a href="#24-run-ls-using-the-long-flag-l-how-did-the-output-change" title="2.4) Run ls using the long flag (-l), how did the output change?" class="md-nav__link">2.4) Run ls using the long flag (-l), how did the output change?</a></li><li class="md-nav__item"><a href="#25-what-type-of-file-is-exp01-and-musclefq" title="2.5) What type of file is exp01 and muscle.fq?" class="md-nav__link">2.5) What type of file is exp01 and muscle.fq?</a></li><li class="md-nav__item"><a href="#26-who-has-permission-to-read-write-and-execute-your-home-directory" title="2.6) Who has permission to read, write and execute your home directory?" class="md-nav__link">2.6) Who has permission to read, write and execute your home directory?</a></li><li class="md-nav__item"><a href="#27-use-the-man-command-to-find-out-what-the-h-flag-does-for-ls" title="2.7) Use the man command to find out what the -h flag does for ls" class="md-nav__link">2.7) Use the man command to find out what the -h flag does for ls</a></li><li class="md-nav__item"><a href="#28-use-the-h-how-did-the-output-change-of-musclefq" title="2.8) Use the -h, how did the output change of muscle.fq?" class="md-nav__link">2.8) Use the -h, how did the output change of muscle.fq?</a></li></ul></nav></li><li class="md-nav__item"><a href="#topic-3-exploring-the-file-system" title="Topic 3: Exploring the file system" class="md-nav__link">Topic 3: Exploring the file system</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#31-print-the-value-of-your-current-working-directory" title="3.1) Print the value of your current working directory." class="md-nav__link">3.1) Print the value of your current working directory.</a></li><li class="md-nav__item"><a href="#32-list-the-contents-of-the-root-directory-called-forward-slash" title="3.2) List the contents of the root directory, called '/' (forward slash)." class="md-nav__link">3.2) List the contents of the root directory, called '/' (forward slash).</a></li><li class="md-nav__item"><a href="#33-use-the-cd-command-to-change-your-working-directory-to-the-root-directory-did-your-prompt-change" title="3.3) Use the cd command to change your working directory to the root directory.  Did your prompt change?" class="md-nav__link">3.3) Use the cd command to change your working directory to the root directory.  Did your prompt change?</a></li><li class="md-nav__item"><a href="#34-list-the-contents-of-the-cwd-and-verify-it-matches-the-list-in-32" title="3.4) List the contents of the CWD and verify it matches the list in 3.2" class="md-nav__link">3.4) List the contents of the CWD and verify it matches the list in 3.2</a></li><li class="md-nav__item"><a href="#35-change-your-current-working-directory-back-to-your-home-directory-what-is-the-simplest-unix-command-that-will-get-you-back-to-your-home-directory-from-anywhere-else-in-the-file-system" title="3.5) Change your current working directory back to your home directory. What is the simplest Unix command that will get you back to your home directory from anywhere else in the file system?" class="md-nav__link">3.5) Change your current working directory back to your home directory. What is the simplest Unix command that will get you back to your home directory from anywhere else in the file system?</a></li><li class="md-nav__item"><a href="#36-change-your-working-directory-to-the-following-directory" title="3.6) Change your working directory to the following directory:" class="md-nav__link">3.6) Change your working directory to the following directory:</a></li><li class="md-nav__item"><a href="#37-list-the-contents-of-that-directory-how-many-files-does-it-contain" title="3.7) List the contents of that directory. How many files does it contain?" class="md-nav__link">3.7) List the contents of that directory. How many files does it contain?</a></li><li class="md-nav__item"><a href="#38-what-kind-of-file-is-vlscitrainingsharedintro_to_unixsleepy" title="3.8) What kind of file is /vlsci/TRAINING/shared/Intro_to_Unix/sleepy?" class="md-nav__link">3.8) What kind of file is /vlsci/TRAINING/shared/Intro_to_Unix/sleepy?</a></li><li class="md-nav__item"><a href="#310-what-are-the-file-permissions-of-the-following-file-and-what-do-they-mean" title="3.10) What are the file permissions of the following file and what do they mean?" class="md-nav__link">3.10) What are the file permissions of the following file and what do they mean?</a></li><li class="md-nav__item"><a href="#311-change-your-working-directory-back-to-your-home-directory-ready-for-the-next-topic" title="3.11) Change your working directory back to your home directory ready for the next topic." class="md-nav__link">3.11) Change your working directory back to your home directory ready for the next topic.</a></li></ul></nav></li><li class="md-nav__item"><a href="#topic-4-working-with-files-and-directories" title="Topic 4: Working with files and directories" class="md-nav__link">Topic 4: Working with files and directories</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#41-in-your-home-directory-make-a-sub-directory-called-test" title="4.1) In your home directory make a sub-directory called test." class="md-nav__link">4.1) In your home directory make a sub-directory called test.</a></li><li class="md-nav__item"><a href="#42-copy-all-the-files-from-the-following-directory-into-the-newly-created-test-directory" title="4.2) Copy all the files from the following directory into the newly created test directory:" class="md-nav__link">4.2) Copy all the files from the following directory into the newly created test directory:</a></li><li class="md-nav__item"><a href="#43-check-that-the-file-size-of-expectationstxt-is-the-same-in-both-the-directory-that-you-copied-it-from-and-the-directory-that-you-copied-it-to" title="4.3) Check that the file size of expectations.txt is the same in both the directory that you copied it from and the directory that you copied it to." class="md-nav__link">4.3) Check that the file size of expectations.txt is the same in both the directory that you copied it from and the directory that you copied it to.</a></li><li class="md-nav__item"><a href="#44-check-that-the-contents-of-expectationstxt-are-the-same-in-both-the-directory-that-you-copied-it-from-and-the-directory-that-you-copied-it-to" title="4.4) Check that the contents of expectations.txt are the same in both the directory that you copied it from and the directory that you copied it to." class="md-nav__link">4.4) Check that the contents of expectations.txt are the same in both the directory that you copied it from and the directory that you copied it to.</a></li><li class="md-nav__item"><a href="#45-how-many-lines-words-and-characters-are-in-expectationstxt" title="4.5) How many lines, words and characters are in expectations.txt?" class="md-nav__link">4.5) How many lines, words and characters are in expectations.txt?</a></li><li class="md-nav__item"><a href="#46-open-testexpectationstxt-in-the-nano-text-editor-delete-the-first-line-of-text-and-save-your-changes-to-the-file-exit-nano" title="4.6) Open ~/test/expectations.txt in the nano text editor, delete the first line of text, and save your changes to the file. Exit nano." class="md-nav__link">4.6) Open ~/test/expectations.txt in the nano text editor, delete the first line of text, and save your changes to the file. Exit nano.</a></li><li class="md-nav__item"><a href="#47-did-the-changes-you-made-to-testexpectationstxt-have-any-effect-on-vlscitrainingsharedintro_to_unix" title="4.7) Did the changes you made to ~/test/expectations.txt have any effect on /vlsci/TRAINING/shared/Intro_to_Unix?" class="md-nav__link">4.7) Did the changes you made to ~/test/expectations.txt have any effect on /vlsci/TRAINING/shared/Intro_to_Unix?</a></li><li class="md-nav__item"><a href="#48-in-your-test-subdirectory-rename-expectationstxt-to-footxt" title="4.8) In your test subdirectory, rename expectations.txt to foo.txt." class="md-nav__link">4.8) In your test subdirectory, rename expectations.txt to foo.txt.</a></li><li class="md-nav__item"><a href="#49-rename-footxt-back-to-expectationstxt" title="4.9) Rename foo.txt back to expectations.txt." class="md-nav__link">4.9) Rename foo.txt back to expectations.txt.</a></li><li class="md-nav__item"><a href="#410-remove-the-file-expectationstxt-from-your-test-directory" title="4.10) Remove the file expectations.txt from your test directory." class="md-nav__link">4.10) Remove the file expectations.txt from your test directory.</a></li><li class="md-nav__item"><a href="#411-remove-the-entire-test-directory-and-all-the-files-within-it" title="4.11) Remove the entire test directory and all the files within it." class="md-nav__link">4.11) Remove the entire test directory and all the files within it.</a></li><li class="md-nav__item"><a href="#412-recreate-the-test-directory-in-your-home-directory-and-copy-all-the-files-from-vlscitrainingsharedintro_to_unix-back-into-the-test-directory" title="4.12) Recreate the test directory in your home directory and copy all the files from /vlsci/TRAINING/shared/Intro_to_Unix back into the test directory." class="md-nav__link">4.12) Recreate the test directory in your home directory and copy all the files from /vlsci/TRAINING/shared/Intro_to_Unix back into the test directory.</a></li><li class="md-nav__item"><a href="#413-change-directories-to-test-and-use-the-cat-command-to-display-the-entire-contents-of-the-file-helloc" title="4.13) Change directories to ~/test and use the cat command to display the entire contents of the file hello.c" class="md-nav__link">4.13) Change directories to ~/test and use the cat command to display the entire contents of the file hello.c</a></li><li class="md-nav__item"><a href="#414-use-the-head-command-to-view-the-first-20-lines-of-the-file-sample_1fastq" title="4.14) Use the head command to view the first 20 lines of the file sample_1.fastq" class="md-nav__link">4.14) Use the head command to view the first 20 lines of the file sample_1.fastq</a></li><li class="md-nav__item"><a href="#415-use-the-tail-command-to-view-the-last-8-lines-of-the-file-sample_1fastq" title="4.15) Use the tail command to view the last 8 lines of the file sample_1.fastq" class="md-nav__link">4.15) Use the tail command to view the last 8 lines of the file sample_1.fastq</a></li><li class="md-nav__item"><a href="#416-use-the-grep-command-to-find-out-all-the-lines-in-mobytxt-that-contain-the-word-ahab" title="4.16) Use the grep command to find out all the lines in moby.txt that contain the word "Ahab"" class="md-nav__link">4.16) Use the grep command to find out all the lines in moby.txt that contain the word "Ahab"</a></li><li class="md-nav__item"><a href="#417-use-the-grep-command-to-find-out-all-the-lines-in-expectationstxt-that-contain-the-word-the-with-a-case-insensitive-search-it-should-count-the-the-the-the-etcetera" title="4.17) Use the grep command to find out all the lines in expectations.txt that contain the word "the" with a case insensitive search (it should count "the" "The" "THE" "tHe" etcetera)." class="md-nav__link">4.17) Use the grep command to find out all the lines in expectations.txt that contain the word "the" with a case insensitive search (it should count "the" "The" "THE" "tHe" etcetera).</a></li><li class="md-nav__item"><a href="#418-use-the-gzip-command-to-compress-the-file-sample_1fastq-use-gunzip-to-decompress-it-back-to-the-original-contents" title="4.18) Use the gzip command to compress the file sample_1.fastq. Use gunzip to decompress it back to the original contents." class="md-nav__link">4.18) Use the gzip command to compress the file sample_1.fastq. Use gunzip to decompress it back to the original contents.</a></li></ul></nav></li><li class="md-nav__item"><a href="#topic-5-pipes-output-redirection-and-shell-scripts" title="Topic 5: Pipes, output redirection and shell scripts" class="md-nav__link">Topic 5: Pipes, output redirection and shell scripts</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#51-how-many-reads-are-contained-in-the-file-sample_1fastq" title="5.1) How many reads are contained in the file sample_1.fastq?" class="md-nav__link">5.1) How many reads are contained in the file sample_1.fastq?</a></li><li class="md-nav__item"><a href="#52-how-many-reads-in-sample_1fastq-contain-the-sequence-gattaca" title="5.2) How many reads in sample_1.fastq contain the sequence GATTACA?" class="md-nav__link">5.2) How many reads in sample_1.fastq contain the sequence GATTACA?</a></li><li class="md-nav__item"><a href="#53-on-what-line-numbers-do-the-sequences-containing-gattaca-occur" title="5.3) On what line numbers do the sequences containing GATTACA occur?" class="md-nav__link">5.3) On what line numbers do the sequences containing GATTACA occur?</a></li><li class="md-nav__item"><a href="#54-use-the-nl-command-to-print-each-line-of-sample_1fastq-with-its-corresponding-line-number-at-the-beginning" title="5.4) Use the nl command to print each line of sample_1.fastq with its corresponding line number at the beginning." class="md-nav__link">5.4) Use the nl command to print each line of sample_1.fastq with its corresponding line number at the beginning.</a></li><li class="md-nav__item"><a href="#55-redirect-the-output-of-the-previous-command-to-a-file-called-sample_1fastqnl" title="5.5) Redirect the output of the previous command to a file called sample_1.fastq.nl." class="md-nav__link">5.5) Redirect the output of the previous command to a file called sample_1.fastq.nl.</a></li><li class="md-nav__item"><a href="#56-the-four-lines-per-read-format-of-fastq-is-cumbersome-to-deal-with-often-it-would-be-preferable-if-we-could-convert-it-to-tab-separated-value-tsv-format-such-that-each-read-appears-on-a-single-line-with-each-of-its-fields-separated-by-tabs-use-the-following-command-to-convert-sample_1fastq-into-tsv-format" title="5.6) The four-lines-per-read format of FASTQ is cumbersome to deal with. Often it would be preferable if we could convert it to tab-separated-value (TSV) format, such that each read appears on a single line with each of its fields separated by tabs. Use the following command to convert sample_1.fastq into TSV format:" class="md-nav__link">5.6) The four-lines-per-read format of FASTQ is cumbersome to deal with. Often it would be preferable if we could convert it to tab-separated-value (TSV) format, such that each read appears on a single line with each of its fields separated by tabs. Use the following command to convert sample_1.fastq into TSV format:</a></li><li class="md-nav__item"><a href="#57-do-you-expect-the-output-of-the-following-command-to-produce-the-same-output-as-above-and-why" title="5.7) Do you expect the output of the following command to produce the same output as above? and why?" class="md-nav__link">5.7) Do you expect the output of the following command to produce the same output as above? and why?</a></li><li class="md-nav__item"><a href="#58-check-that-sample_1tsv-has-the-correct-number-of-lines-use-the-head-command-to-view-the-first-20-lines-of-the-file" title="5.8) Check that sample_1.tsv has the correct number of lines. Use the head command to view the first 20 lines of the file." class="md-nav__link">5.8) Check that sample_1.tsv has the correct number of lines. Use the head command to view the first 20 lines of the file.</a></li><li class="md-nav__item"><a href="#59-use-the-cut-command-to-print-out-the-second-column-of-sample_1tsv-redirect-the-output-to-a-file-called-sample_1dnatxt" title="5.9) Use the cut command to print out the second column of sample_1.tsv. Redirect the output to a file called sample_1.dna.txt." class="md-nav__link">5.9) Use the cut command to print out the second column of sample_1.tsv. Redirect the output to a file called sample_1.dna.txt.</a></li><li class="md-nav__item"><a href="#510-use-the-sort-command-to-sort-the-lines-of-sample_1dnatxt-and-redirect-the-output-to-sample_1dnasortedtxt-use-head-to-look-at-the-first-few-lines-of-the-output-file-you-should-see-a-lot-of-repeated-sequences-of-as" title="5.10) Use the sort command to sort the lines of sample_1.dna.txt and redirect the output to sample_1.dna.sorted.txt. Use head to look at the first few lines of the output file. You should see a lot of repeated sequences of As." class="md-nav__link">5.10) Use the sort command to sort the lines of sample_1.dna.txt and redirect the output to sample_1.dna.sorted.txt. Use head to look at the first few lines of the output file. You should see a lot of repeated sequences of As.</a></li><li class="md-nav__item"><a href="#511-use-the-uniq-command-to-remove-duplicate-consecutive-lines-from-sample_1dnasortedtxt-redirect-the-result-to-sample_1dnauniqtxt-compare-the-number-of-lines-in-sample1_dnatxt-to-the-number-of-lines-in-sample_1dnauniqtxt" title="5.11) Use the uniq command to remove duplicate consecutive lines from sample_1.dna.sorted.txt, redirect the result to sample_1.dna.uniq.txt. Compare the number of lines in sample1_dna.txt to the number of lines in sample_1.dna.uniq.txt." class="md-nav__link">5.11) Use the uniq command to remove duplicate consecutive lines from sample_1.dna.sorted.txt, redirect the result to sample_1.dna.uniq.txt. Compare the number of lines in sample1_dna.txt to the number of lines in sample_1.dna.uniq.txt.</a></li><li class="md-nav__item"><a href="#512-can-you-modify-the-command-from-above-to-produce-only-those-sequences-of-dna-which-were-duplicated-in-sample_1dnasortedtxt" title="5.12) Can you modify the command from above to produce only those sequences of DNA which were duplicated in sample_1.dna.sorted.txt?" class="md-nav__link">5.12) Can you modify the command from above to produce only those sequences of DNA which were duplicated in sample_1.dna.sorted.txt?</a></li><li class="md-nav__item"><a href="#513-write-a-shell-pipeline-which-will-print-the-number-of-duplicated-dna-sequences-in-sample_1fastq" title="5.13) Write a shell pipeline which will print the number of duplicated DNA sequences in sample_1.fastq." class="md-nav__link">5.13) Write a shell pipeline which will print the number of duplicated DNA sequences in sample_1.fastq.</a></li><li class="md-nav__item"><a href="#514-advanced-write-a-shell-script-which-will-print-the-number-of-duplicated-dna-sequences-in-sample_1fastq" title="5.14) (Advanced) Write a shell script which will print the number of duplicated DNA sequences in sample_1.fastq." class="md-nav__link">5.14) (Advanced) Write a shell script which will print the number of duplicated DNA sequences in sample_1.fastq.</a></li><li class="md-nav__item"><a href="#515-advanced-modify-your-shell-script-so-that-it-accepts-the-name-of-the-input-fastq-file-as-a-command-line-parameter" title="5.15) (Advanced) Modify your shell script so that it accepts the name of the input FASTQ file as a command line parameter." class="md-nav__link">5.15) (Advanced) Modify your shell script so that it accepts the name of the input FASTQ file as a command line parameter.</a></li><li class="md-nav__item"><a href="#516-advanced-modify-your-shell-script-so-that-it-accepts-zero-or-more-fastq-files-on-the-command-line-argument-and-outputs-the-number-of-duplicated-dna-sequences-in-each-file" title="5.16) (Advanced) Modify your shell script so that it accepts zero or more FASTQ files on the command line argument and outputs the number of duplicated DNA sequences in each file." class="md-nav__link">5.16) (Advanced) Modify your shell script so that it accepts zero or more FASTQ files on the command line argument and outputs the number of duplicated DNA sequences in each file.</a></li></ul></nav></li><li class="md-nav__item"><a href="#finished" title="Finished" class="md-nav__link">Finished</a></li></ul></nav></li><li class="md-nav__item"><a href="../../gvl_launch/gvl_launch/" title="Launching a Personal GVL Server" class="md-nav__link">Launching a Personal GVL Server</a></li><li class="md-nav__item"><a href="../../galaxy_101/galaxy_101/" title="Introduction to Galaxy" class="md-nav__link">Introduction to Galaxy</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-9" type="checkbox" id="nav-3-9"><label class="md-nav__link" for="nav-3-9">Galaxy Workflows</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-3-9">Galaxy Workflows</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../galaxy-workflows/galaxy-workflows/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../galaxy-workflows/history_creation/" title="History Creation" class="md-nav__link">History Creation</a></li></ul></nav></li><li class="md-nav__item"><a href="../../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">Containerized Bioinformatics</a></li><li class="md-nav__item"><a href="../../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">Common Workflow Language for Bioinformatics</a></li><li class="md-nav__item"><a href="../../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">Command-line software development</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4"><label class="md-nav__link" for="nav-4">Genomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-4">Genomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" title="Introduction to Variant Calling" class="md-nav__link">Introduction to Variant Calling</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2"><label class="md-nav__link" for="nav-4-2">Variant Calling Advanced</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-2">Variant Calling Advanced</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3"><label class="md-nav__link" for="nav-4-3">RNA-seq DGE Basic</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-3">RNA-seq DGE Basic</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="Tuxedo Protocol Tutorial" class="md-nav__link">Tuxedo Protocol Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNA-seq DGE Advanced" class="md-nav__link">RNA-seq DGE Advanced</a></li><li class="md-nav__item"><a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq DGE Experimental Design" class="md-nav__link">RNA-seq DGE Experimental Design</a></li><li class="md-nav__item"><a href="../../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">10X single-cell RNA-seq analysis in R</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-7" type="checkbox" id="nav-4-7"><label class="md-nav__link" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../assembly/assembly-background/" title="Method Background" class="md-nav__link">Method Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../assembly/spades/" title="Microbial assembly: Spades" class="md-nav__link">Microbial assembly: Spades</a></li><li class="md-nav__item"><a href="../../pacbio/" title="Long read assembly" class="md-nav__link">Long read assembly</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5"><label class="md-nav__link" for="nav-5">Proteomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-5">Proteomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-1" type="checkbox" id="nav-5-1"><label class="md-nav__link" for="nav-5-1">Identifying proteins from mass spectrometry data</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-5-1">Identifying proteins from mass spectrometry data</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../proteomics_basic/proteomics_basic/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_galaxy/" title="Galaxy Background" class="md-nav__link">Galaxy Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">Data Formats Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_findcountitems_workflow/" title="Create find count items workflow" class="md-nav__link">Create find count items workflow</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">Protein Databases</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">Search Engines</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">Protein Prophet</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6"><label class="md-nav__link" for="nav-6">Molecular Modelling Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-6">Molecular Modelling Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">Molecular Dynamics - Introduction to cluster computing</a></li><li class="md-nav__item"><a href="../../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">Molecular Dynamics - Building input files, visualising the trajectory</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7"><label class="md-nav__link" for="nav-7">Protocols</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-7">Protocols</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly-protocol/" title="De Novo Genome Assembly for Illumina Data" class="md-nav__link">De Novo Genome Assembly for Illumina Data</a></li></ul></nav></li></ul></nav></div></div></div><div class="md-sidebar md-sidebar--secondary" data-md-component="toc"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#overview" title="Overview" class="md-nav__link">Overview</a></li><li class="md-nav__item"><a href="#learning-objectives" title="Learning Objectives" class="md-nav__link">Learning Objectives</a></li><li class="md-nav__item"><a href="#requirements" title="Requirements" class="md-nav__link">Requirements</a></li><li class="md-nav__item"><a href="#introduction" title="Introduction" class="md-nav__link">Introduction</a></li><li class="md-nav__item"><a href="#topic-1-remote-log-in" title="Topic 1: Remote log in" class="md-nav__link">Topic 1: Remote log in</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#connecting-to-a-unix-computer" title="Connecting to a Unix computer" class="md-nav__link">Connecting to a Unix computer</a></li><li class="md-nav__item"><a href="#exercises" title="Exercises" class="md-nav__link">Exercises</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#11-when-youve-logged-into-the-unix-server-run-the-following-commands-and-see-what-they-do" title="1.1) When you've logged into the Unix server, run the following commands and see what they do:" class="md-nav__link">1.1) When you've logged into the Unix server, run the following commands and see what they do:</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item"><a href="#topic-2-exploring-your-home-directory" title="Topic 2: Exploring your home directory" class="md-nav__link">Topic 2: Exploring your home directory</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#21-use-the-ls-command-to-list-the-files-in-your-home-directory-how-many-files-are-there" title="2.1) Use the ls command to list the files in your home directory.  How many files are there?" class="md-nav__link">2.1) Use the ls command to list the files in your home directory.  How many files are there?</a></li><li class="md-nav__item"><a href="#22-what-flag-might-you-use-to-display-all-files-with-the-ls-command-how-many-files-are-really-there" title="2.2) What flag might you use to display all files with the ls command?  How many files are really there?" class="md-nav__link">2.2) What flag might you use to display all files with the ls command?  How many files are really there?</a></li><li class="md-nav__item"><a href="#23-what-is-the-full-path-name-of-your-home-directory" title="2.3) What is the full path name of your home directory?" class="md-nav__link">2.3) What is the full path name of your home directory?</a></li><li class="md-nav__item"><a href="#24-run-ls-using-the-long-flag-l-how-did-the-output-change" title="2.4) Run ls using the long flag (-l), how did the output change?" class="md-nav__link">2.4) Run ls using the long flag (-l), how did the output change?</a></li><li class="md-nav__item"><a href="#25-what-type-of-file-is-exp01-and-musclefq" title="2.5) What type of file is exp01 and muscle.fq?" class="md-nav__link">2.5) What type of file is exp01 and muscle.fq?</a></li><li class="md-nav__item"><a href="#26-who-has-permission-to-read-write-and-execute-your-home-directory" title="2.6) Who has permission to read, write and execute your home directory?" class="md-nav__link">2.6) Who has permission to read, write and execute your home directory?</a></li><li class="md-nav__item"><a href="#27-use-the-man-command-to-find-out-what-the-h-flag-does-for-ls" title="2.7) Use the man command to find out what the -h flag does for ls" class="md-nav__link">2.7) Use the man command to find out what the -h flag does for ls</a></li><li class="md-nav__item"><a href="#28-use-the-h-how-did-the-output-change-of-musclefq" title="2.8) Use the -h, how did the output change of muscle.fq?" class="md-nav__link">2.8) Use the -h, how did the output change of muscle.fq?</a></li></ul></nav></li><li class="md-nav__item"><a href="#topic-3-exploring-the-file-system" title="Topic 3: Exploring the file system" class="md-nav__link">Topic 3: Exploring the file system</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#31-print-the-value-of-your-current-working-directory" title="3.1) Print the value of your current working directory." class="md-nav__link">3.1) Print the value of your current working directory.</a></li><li class="md-nav__item"><a href="#32-list-the-contents-of-the-root-directory-called-forward-slash" title="3.2) List the contents of the root directory, called '/' (forward slash)." class="md-nav__link">3.2) List the contents of the root directory, called '/' (forward slash).</a></li><li class="md-nav__item"><a href="#33-use-the-cd-command-to-change-your-working-directory-to-the-root-directory-did-your-prompt-change" title="3.3) Use the cd command to change your working directory to the root directory.  Did your prompt change?" class="md-nav__link">3.3) Use the cd command to change your working directory to the root directory.  Did your prompt change?</a></li><li class="md-nav__item"><a href="#34-list-the-contents-of-the-cwd-and-verify-it-matches-the-list-in-32" title="3.4) List the contents of the CWD and verify it matches the list in 3.2" class="md-nav__link">3.4) List the contents of the CWD and verify it matches the list in 3.2</a></li><li class="md-nav__item"><a href="#35-change-your-current-working-directory-back-to-your-home-directory-what-is-the-simplest-unix-command-that-will-get-you-back-to-your-home-directory-from-anywhere-else-in-the-file-system" title="3.5) Change your current working directory back to your home directory. What is the simplest Unix command that will get you back to your home directory from anywhere else in the file system?" class="md-nav__link">3.5) Change your current working directory back to your home directory. What is the simplest Unix command that will get you back to your home directory from anywhere else in the file system?</a></li><li class="md-nav__item"><a href="#36-change-your-working-directory-to-the-following-directory" title="3.6) Change your working directory to the following directory:" class="md-nav__link">3.6) Change your working directory to the following directory:</a></li><li class="md-nav__item"><a href="#37-list-the-contents-of-that-directory-how-many-files-does-it-contain" title="3.7) List the contents of that directory. How many files does it contain?" class="md-nav__link">3.7) List the contents of that directory. How many files does it contain?</a></li><li class="md-nav__item"><a href="#38-what-kind-of-file-is-vlscitrainingsharedintro_to_unixsleepy" title="3.8) What kind of file is /vlsci/TRAINING/shared/Intro_to_Unix/sleepy?" class="md-nav__link">3.8) What kind of file is /vlsci/TRAINING/shared/Intro_to_Unix/sleepy?</a></li><li class="md-nav__item"><a href="#310-what-are-the-file-permissions-of-the-following-file-and-what-do-they-mean" title="3.10) What are the file permissions of the following file and what do they mean?" class="md-nav__link">3.10) What are the file permissions of the following file and what do they mean?</a></li><li class="md-nav__item"><a href="#311-change-your-working-directory-back-to-your-home-directory-ready-for-the-next-topic" title="3.11) Change your working directory back to your home directory ready for the next topic." class="md-nav__link">3.11) Change your working directory back to your home directory ready for the next topic.</a></li></ul></nav></li><li class="md-nav__item"><a href="#topic-4-working-with-files-and-directories" title="Topic 4: Working with files and directories" class="md-nav__link">Topic 4: Working with files and directories</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#41-in-your-home-directory-make-a-sub-directory-called-test" title="4.1) In your home directory make a sub-directory called test." class="md-nav__link">4.1) In your home directory make a sub-directory called test.</a></li><li class="md-nav__item"><a href="#42-copy-all-the-files-from-the-following-directory-into-the-newly-created-test-directory" title="4.2) Copy all the files from the following directory into the newly created test directory:" class="md-nav__link">4.2) Copy all the files from the following directory into the newly created test directory:</a></li><li class="md-nav__item"><a href="#43-check-that-the-file-size-of-expectationstxt-is-the-same-in-both-the-directory-that-you-copied-it-from-and-the-directory-that-you-copied-it-to" title="4.3) Check that the file size of expectations.txt is the same in both the directory that you copied it from and the directory that you copied it to." class="md-nav__link">4.3) Check that the file size of expectations.txt is the same in both the directory that you copied it from and the directory that you copied it to.</a></li><li class="md-nav__item"><a href="#44-check-that-the-contents-of-expectationstxt-are-the-same-in-both-the-directory-that-you-copied-it-from-and-the-directory-that-you-copied-it-to" title="4.4) Check that the contents of expectations.txt are the same in both the directory that you copied it from and the directory that you copied it to." class="md-nav__link">4.4) Check that the contents of expectations.txt are the same in both the directory that you copied it from and the directory that you copied it to.</a></li><li class="md-nav__item"><a href="#45-how-many-lines-words-and-characters-are-in-expectationstxt" title="4.5) How many lines, words and characters are in expectations.txt?" class="md-nav__link">4.5) How many lines, words and characters are in expectations.txt?</a></li><li class="md-nav__item"><a href="#46-open-testexpectationstxt-in-the-nano-text-editor-delete-the-first-line-of-text-and-save-your-changes-to-the-file-exit-nano" title="4.6) Open ~/test/expectations.txt in the nano text editor, delete the first line of text, and save your changes to the file. Exit nano." class="md-nav__link">4.6) Open ~/test/expectations.txt in the nano text editor, delete the first line of text, and save your changes to the file. Exit nano.</a></li><li class="md-nav__item"><a href="#47-did-the-changes-you-made-to-testexpectationstxt-have-any-effect-on-vlscitrainingsharedintro_to_unix" title="4.7) Did the changes you made to ~/test/expectations.txt have any effect on /vlsci/TRAINING/shared/Intro_to_Unix?" class="md-nav__link">4.7) Did the changes you made to ~/test/expectations.txt have any effect on /vlsci/TRAINING/shared/Intro_to_Unix?</a></li><li class="md-nav__item"><a href="#48-in-your-test-subdirectory-rename-expectationstxt-to-footxt" title="4.8) In your test subdirectory, rename expectations.txt to foo.txt." class="md-nav__link">4.8) In your test subdirectory, rename expectations.txt to foo.txt.</a></li><li class="md-nav__item"><a href="#49-rename-footxt-back-to-expectationstxt" title="4.9) Rename foo.txt back to expectations.txt." class="md-nav__link">4.9) Rename foo.txt back to expectations.txt.</a></li><li class="md-nav__item"><a href="#410-remove-the-file-expectationstxt-from-your-test-directory" title="4.10) Remove the file expectations.txt from your test directory." class="md-nav__link">4.10) Remove the file expectations.txt from your test directory.</a></li><li class="md-nav__item"><a href="#411-remove-the-entire-test-directory-and-all-the-files-within-it" title="4.11) Remove the entire test directory and all the files within it." class="md-nav__link">4.11) Remove the entire test directory and all the files within it.</a></li><li class="md-nav__item"><a href="#412-recreate-the-test-directory-in-your-home-directory-and-copy-all-the-files-from-vlscitrainingsharedintro_to_unix-back-into-the-test-directory" title="4.12) Recreate the test directory in your home directory and copy all the files from /vlsci/TRAINING/shared/Intro_to_Unix back into the test directory." class="md-nav__link">4.12) Recreate the test directory in your home directory and copy all the files from /vlsci/TRAINING/shared/Intro_to_Unix back into the test directory.</a></li><li class="md-nav__item"><a href="#413-change-directories-to-test-and-use-the-cat-command-to-display-the-entire-contents-of-the-file-helloc" title="4.13) Change directories to ~/test and use the cat command to display the entire contents of the file hello.c" class="md-nav__link">4.13) Change directories to ~/test and use the cat command to display the entire contents of the file hello.c</a></li><li class="md-nav__item"><a href="#414-use-the-head-command-to-view-the-first-20-lines-of-the-file-sample_1fastq" title="4.14) Use the head command to view the first 20 lines of the file sample_1.fastq" class="md-nav__link">4.14) Use the head command to view the first 20 lines of the file sample_1.fastq</a></li><li class="md-nav__item"><a href="#415-use-the-tail-command-to-view-the-last-8-lines-of-the-file-sample_1fastq" title="4.15) Use the tail command to view the last 8 lines of the file sample_1.fastq" class="md-nav__link">4.15) Use the tail command to view the last 8 lines of the file sample_1.fastq</a></li><li class="md-nav__item"><a href="#416-use-the-grep-command-to-find-out-all-the-lines-in-mobytxt-that-contain-the-word-ahab" title="4.16) Use the grep command to find out all the lines in moby.txt that contain the word "Ahab"" class="md-nav__link">4.16) Use the grep command to find out all the lines in moby.txt that contain the word "Ahab"</a></li><li class="md-nav__item"><a href="#417-use-the-grep-command-to-find-out-all-the-lines-in-expectationstxt-that-contain-the-word-the-with-a-case-insensitive-search-it-should-count-the-the-the-the-etcetera" title="4.17) Use the grep command to find out all the lines in expectations.txt that contain the word "the" with a case insensitive search (it should count "the" "The" "THE" "tHe" etcetera)." class="md-nav__link">4.17) Use the grep command to find out all the lines in expectations.txt that contain the word "the" with a case insensitive search (it should count "the" "The" "THE" "tHe" etcetera).</a></li><li class="md-nav__item"><a href="#418-use-the-gzip-command-to-compress-the-file-sample_1fastq-use-gunzip-to-decompress-it-back-to-the-original-contents" title="4.18) Use the gzip command to compress the file sample_1.fastq. Use gunzip to decompress it back to the original contents." class="md-nav__link">4.18) Use the gzip command to compress the file sample_1.fastq. Use gunzip to decompress it back to the original contents.</a></li></ul></nav></li><li class="md-nav__item"><a href="#topic-5-pipes-output-redirection-and-shell-scripts" title="Topic 5: Pipes, output redirection and shell scripts" class="md-nav__link">Topic 5: Pipes, output redirection and shell scripts</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#51-how-many-reads-are-contained-in-the-file-sample_1fastq" title="5.1) How many reads are contained in the file sample_1.fastq?" class="md-nav__link">5.1) How many reads are contained in the file sample_1.fastq?</a></li><li class="md-nav__item"><a href="#52-how-many-reads-in-sample_1fastq-contain-the-sequence-gattaca" title="5.2) How many reads in sample_1.fastq contain the sequence GATTACA?" class="md-nav__link">5.2) How many reads in sample_1.fastq contain the sequence GATTACA?</a></li><li class="md-nav__item"><a href="#53-on-what-line-numbers-do-the-sequences-containing-gattaca-occur" title="5.3) On what line numbers do the sequences containing GATTACA occur?" class="md-nav__link">5.3) On what line numbers do the sequences containing GATTACA occur?</a></li><li class="md-nav__item"><a href="#54-use-the-nl-command-to-print-each-line-of-sample_1fastq-with-its-corresponding-line-number-at-the-beginning" title="5.4) Use the nl command to print each line of sample_1.fastq with its corresponding line number at the beginning." class="md-nav__link">5.4) Use the nl command to print each line of sample_1.fastq with its corresponding line number at the beginning.</a></li><li class="md-nav__item"><a href="#55-redirect-the-output-of-the-previous-command-to-a-file-called-sample_1fastqnl" title="5.5) Redirect the output of the previous command to a file called sample_1.fastq.nl." class="md-nav__link">5.5) Redirect the output of the previous command to a file called sample_1.fastq.nl.</a></li><li class="md-nav__item"><a href="#56-the-four-lines-per-read-format-of-fastq-is-cumbersome-to-deal-with-often-it-would-be-preferable-if-we-could-convert-it-to-tab-separated-value-tsv-format-such-that-each-read-appears-on-a-single-line-with-each-of-its-fields-separated-by-tabs-use-the-following-command-to-convert-sample_1fastq-into-tsv-format" title="5.6) The four-lines-per-read format of FASTQ is cumbersome to deal with. Often it would be preferable if we could convert it to tab-separated-value (TSV) format, such that each read appears on a single line with each of its fields separated by tabs. Use the following command to convert sample_1.fastq into TSV format:" class="md-nav__link">5.6) The four-lines-per-read format of FASTQ is cumbersome to deal with. Often it would be preferable if we could convert it to tab-separated-value (TSV) format, such that each read appears on a single line with each of its fields separated by tabs. Use the following command to convert sample_1.fastq into TSV format:</a></li><li class="md-nav__item"><a href="#57-do-you-expect-the-output-of-the-following-command-to-produce-the-same-output-as-above-and-why" title="5.7) Do you expect the output of the following command to produce the same output as above? and why?" class="md-nav__link">5.7) Do you expect the output of the following command to produce the same output as above? and why?</a></li><li class="md-nav__item"><a href="#58-check-that-sample_1tsv-has-the-correct-number-of-lines-use-the-head-command-to-view-the-first-20-lines-of-the-file" title="5.8) Check that sample_1.tsv has the correct number of lines. Use the head command to view the first 20 lines of the file." class="md-nav__link">5.8) Check that sample_1.tsv has the correct number of lines. Use the head command to view the first 20 lines of the file.</a></li><li class="md-nav__item"><a href="#59-use-the-cut-command-to-print-out-the-second-column-of-sample_1tsv-redirect-the-output-to-a-file-called-sample_1dnatxt" title="5.9) Use the cut command to print out the second column of sample_1.tsv. Redirect the output to a file called sample_1.dna.txt." class="md-nav__link">5.9) Use the cut command to print out the second column of sample_1.tsv. Redirect the output to a file called sample_1.dna.txt.</a></li><li class="md-nav__item"><a href="#510-use-the-sort-command-to-sort-the-lines-of-sample_1dnatxt-and-redirect-the-output-to-sample_1dnasortedtxt-use-head-to-look-at-the-first-few-lines-of-the-output-file-you-should-see-a-lot-of-repeated-sequences-of-as" title="5.10) Use the sort command to sort the lines of sample_1.dna.txt and redirect the output to sample_1.dna.sorted.txt. Use head to look at the first few lines of the output file. You should see a lot of repeated sequences of As." class="md-nav__link">5.10) Use the sort command to sort the lines of sample_1.dna.txt and redirect the output to sample_1.dna.sorted.txt. Use head to look at the first few lines of the output file. You should see a lot of repeated sequences of As.</a></li><li class="md-nav__item"><a href="#511-use-the-uniq-command-to-remove-duplicate-consecutive-lines-from-sample_1dnasortedtxt-redirect-the-result-to-sample_1dnauniqtxt-compare-the-number-of-lines-in-sample1_dnatxt-to-the-number-of-lines-in-sample_1dnauniqtxt" title="5.11) Use the uniq command to remove duplicate consecutive lines from sample_1.dna.sorted.txt, redirect the result to sample_1.dna.uniq.txt. Compare the number of lines in sample1_dna.txt to the number of lines in sample_1.dna.uniq.txt." class="md-nav__link">5.11) Use the uniq command to remove duplicate consecutive lines from sample_1.dna.sorted.txt, redirect the result to sample_1.dna.uniq.txt. Compare the number of lines in sample1_dna.txt to the number of lines in sample_1.dna.uniq.txt.</a></li><li class="md-nav__item"><a href="#512-can-you-modify-the-command-from-above-to-produce-only-those-sequences-of-dna-which-were-duplicated-in-sample_1dnasortedtxt" title="5.12) Can you modify the command from above to produce only those sequences of DNA which were duplicated in sample_1.dna.sorted.txt?" class="md-nav__link">5.12) Can you modify the command from above to produce only those sequences of DNA which were duplicated in sample_1.dna.sorted.txt?</a></li><li class="md-nav__item"><a href="#513-write-a-shell-pipeline-which-will-print-the-number-of-duplicated-dna-sequences-in-sample_1fastq" title="5.13) Write a shell pipeline which will print the number of duplicated DNA sequences in sample_1.fastq." class="md-nav__link">5.13) Write a shell pipeline which will print the number of duplicated DNA sequences in sample_1.fastq.</a></li><li class="md-nav__item"><a href="#514-advanced-write-a-shell-script-which-will-print-the-number-of-duplicated-dna-sequences-in-sample_1fastq" title="5.14) (Advanced) Write a shell script which will print the number of duplicated DNA sequences in sample_1.fastq." class="md-nav__link">5.14) (Advanced) Write a shell script which will print the number of duplicated DNA sequences in sample_1.fastq.</a></li><li class="md-nav__item"><a href="#515-advanced-modify-your-shell-script-so-that-it-accepts-the-name-of-the-input-fastq-file-as-a-command-line-parameter" title="5.15) (Advanced) Modify your shell script so that it accepts the name of the input FASTQ file as a command line parameter." class="md-nav__link">5.15) (Advanced) Modify your shell script so that it accepts the name of the input FASTQ file as a command line parameter.</a></li><li class="md-nav__item"><a href="#516-advanced-modify-your-shell-script-so-that-it-accepts-zero-or-more-fastq-files-on-the-command-line-argument-and-outputs-the-number-of-duplicated-dna-sequences-in-each-file" title="5.16) (Advanced) Modify your shell script so that it accepts zero or more FASTQ files on the command line argument and outputs the number of duplicated DNA sequences in each file." class="md-nav__link">5.16) (Advanced) Modify your shell script so that it accepts zero or more FASTQ files on the command line argument and outputs the number of duplicated DNA sequences in each file.</a></li></ul></nav></li><li class="md-nav__item"><a href="#finished" title="Finished" class="md-nav__link">Finished</a></li></ul></nav></div></div></div><div class="md-content"><article class="md-content__inner md-typeset"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/unix/unix.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a><style src="../../includes/media/tute.css" ></style>

<style>em {font-style: normal; font-family: courier new;}</style>

<h1 id="introduction-to-unix">Introduction to Unix</h1>
<p>A hands-on workshop covering the basics of the Unix/Linux command line interface.</p>
<h2 id="overview">Overview</h2>
<p>Knowledge of the Unix operating system is fundamental to being
productive on HPC systems. This workshop will introduce you to the
fundamental Unix concepts by way of a series of hands-on exercises.</p>
<p>The workshop is facilitated by experienced Unix users who will be able
to guide you through the exercises and offer assistance where needed.</p>
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of the course, you will be able to:</p>
<ul>
<li>Log into a Unix machine remotely</li>
<li>Organise your files into directories</li>
<li>Change file permissions to improve security and safety</li>
<li>Create and edit files with a text editor</li>
<li>Copy files between directories</li>
<li>Use command line programs to manipulate files</li>
<li>Automate your workflow using shell scripts</li>
</ul>
<h2 id="requirements">Requirements</h2>
<ul>
<li>The workshop is intended for beginners with no prior experience in Unix.</li>
<li>Attendees are required to bring their own laptop computers.</li>
</ul>
<h2 id="introduction">Introduction</h2>
<p>Before we commence the hands-on part of this workshop we will first give a short 30 minute talk to introduce the Unix concepts.
The <a href="../slides.html">slides</a> are available if you would like.  Additionally the following reference material is available for later
use.</p>
<details>
  <summary>Reference Material</summary>

### Why use unix

* **Powerful**: Unix computers are typically very powerful in comparision to your desktop/laptop computers.
  Additionally they don't typically use a Graphical User Interface which can free up much resources for actual
  computing.
* **Big data**: Unix programs are designed to handle large data sets
* **Flexible**: Small programs that can be arranged in many ways to solve your problems
* **Automation**: Scripting allows you to do many tasks in one step and repeat steps many times
* **Pipelines**: Unix programs are designed to be 'chained' together to form long multi-step pipelines
* **Science Software**: Lots of Scientific software is designed to run in a Unix environment


### User interface

The Unix interface is a text-based command driven one; often known as a Command Line Interface (CLI).  This means
that you control it by issuing (i.e. typing) commands at the command prompt.  Consequently, the Mouse does not perform
any function in the unix environment.

<img src="../media/unix-screen.png" title="Unix Screen" alt="Unix Screen" width="500px" />


### Command prompt

The command prompt is the first thing you see when you connect to a Unix Computer.  Its purpose is to receive the next
command from you, the user.

<img src="../media/cli-parts.png" title="Command prompt" alt="Command prompt" width="500px" />

There are several parts that make up the command prompt:

* **Time**: the time (when the last command finished)
* **Username**: the username that you are logged in as
* **Hostname**: the name of the computer that your are connected to
* **Current working directory**: the current position within the file system that your are working.  More to follow
* **Prompt**: this is simply a sign to the user that the computer is ready to accept the next command

From this point forward in this document, the command prompt will be simply represented as a '$' rather than
the full command prompt as shown above.  When copying and pasting commands you should NOT copy the '$' sign.


### Command line

Below is an example command with various flags and options.

<img src="../media/cli-command.png" title="Command line" alt="Command line" width="500px" />

There are a number of parts which may be included in a command; each is separated by one of more 'white-space' characters (i.e. space, tab):

* **Command**: this is the name of the program (command) that you want to run
* **Flag**: these turn on (or off) specific features in the program.  They consist of a dash (-) followed by
  a single character.
* **Long flag**: same as flag except they are generally two dashes (--) followed by a word (or two)
* **Option**: set the value of a configurable option.  They are a flag (or long flag) followed by a value
* **Anonymous options**: these are one or more options that are specified in the required order
* **Quoted value**: if you need to specify a space (or tab) in an option then you will need to use double (") or
  single (') quotes on each side of the value.


### File system

The file-system of a unix computer can be thought of as an up-side-down tree.  The topmost directory has a special name
called 'root'; it contains all files and directories that are on the computer system.  It is represented by a single
slash (/).  The figure below shows an example file system with directories (black outline boxes) and files (grey dashed
boxes).

<img src="../media/file-system.png" title="File system" alt="File system" width="400px" />

At the top level we have one file (settings) and one directory (home).  Inside the home directory we have two directories
(user1 and user2) and so on.


#### Absolute file names

Absolute file names are a single unique name for each file and directory within the computer.  They start with the slash (/)
character and follow all the parent directories above the file/directory.

<img src="../media/file-system-settings.png" title="File system" alt="File system" width="400px" />

**Absolute file name**: */settings*

<img src="../media/file-system-file01.png" title="File system" alt="File system" width="400px" />

**Absolute file name**: */home/user1/file01.txt*

<img src="../media/file-system-user2.png" title="File system" alt="File system" width="400px" />

**Absolute file name**: */home/user2*  

**Note**: the final slash is not needed (but generally doesn't hurt if it is present).


#### Current working directory

The *current working directory* is the current location within the file system that you are currently using.
When you first login to a unix computer it will begin with the current working directory set to your home directory, that
is, a place that is unique to you and generally nobody else will have access to it.

<img src="../media/unix-cwd.png" title="Current working directory" alt="Current working directory" width="500px" />

Remember from earlier that the current working directory is shown in the command prompt.


#### Relative file names

Relative file names are a short cut to writing file names that are shorter.  The difference between an absolute file
name is that relative file names do NOT begin with a slash.

<img src="../media/file-system-muscle.png" title="File system" alt="File system" width="400px" />

If your current working directory is set to */home* you can leave this part from the beginning of the filename.

**Relative file name**: *user1/muscle.fq*

(Note: the absence of the leading slash)


**Special file names**:

There are a few further short cuts for typing relative file names:

* *~* (Tilde): is a short cut to your home directory
* *.* (dot): is a short cut for the current directory
* *..* (2x dot): means the parent (or one directory up) from current directory
* *...* (3x dot): does not mean anything (a gotcha for new users).  If you want 2 directories up then chain two
double dot's  e.g. *../..*

**Note**: the special file names above can be used within absolute and relative file name and used multiple times.


<img src="../media/file-system-muscle2.png" title="File system" alt="File system" width="400px" />

Now, if the current working directory is changed to */home/user2* the relative path to muscle.fq is different.

**Relative file name**: *../user1/muscle.fq*

</details>

<h2 id="topic-1-remote-log-in">Topic 1: Remote log in</h2>
<p>In this topic we will learn how to connect to a <em>Unix</em> computer via a program called <em>ssh</em> and run a few basic commands.</p>
<h3 id="connecting-to-a-unix-computer">Connecting to a Unix computer</h3>
<p>To begin this workshop you will need to connect to an HPC.  Today we will use <em>barcoo</em>.
The computer called <em>barcoo.vlsci.org.au</em> is the one that coordinates all the HPC's tasks.</p>
<p><strong>Server details</strong>:</p>
<ul>
<li><strong>host</strong>: barcoo.vlsci.org.au</li>
<li><strong>port</strong>: 22</li>
<li><strong>username</strong>: (provided at workshop)</li>
<li><strong>password</strong>: (provided at workshop)</li>
</ul>
<details>
  <summary>Mac OS X / Linux</summary>

Both Mac OS X and Linux come with a version of ssh (called OpenSSH) that can be used from the command line.  To use OpenSSH you must 
first start a terminal program on your computer.  On OS X the standard terminal is called Terminal, and it is installed by default. 
On Linux there are many popular terminal programs including: xterm, gnome-terminal, konsole (if you aren't sure, then xterm is a good 
default).  When you've started the terminal you should see a command prompt.  To log into *barcoo*, for example, type this command at 
the prompt and press return (where the word *username* is replaced with your *barcoo* username):

*$ ssh username@barcoo.vlsci.org.au*

The same procedure works for any other machine where you have an account except that if your Unix computer uses a port other than 22 you will
need to specify the port by adding the option *-p PORT* with PORT substituted with the port number.

You may be presented with a message along the lines of:


<pre><code class="text">The authenticity of host 'barcoo.vlsci.org.au (131.172.36.150)' can't be  established.
...
Are you sure you want to continue connecting (yes/no)?
</code></pre>


Although you should never ignore a warning, this particular one is nothing to be concerned about; type **yes** and then **press enter**. 
If all goes well you will be asked to enter your password.  Assuming you type the correct username and password the system should 
then display a welcome message, and then present you with a Unix prompt.  If you get this far then you are ready to start entering 
Unix commands and thus begin using the remote computer.

</details>

<details>
  <summary>Windows</summary>

On Microsoft Windows (Vista, 7, 8) we recommend that you use the PuTTY ssh client.  PuTTY (putty.exe) can be downloaded 
from this web page:

[http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html](http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html)

Documentation for using PuTTY is here:

[http://www.chiark.greenend.org.uk/~sgtatham/putty/docs.html](http://www.chiark.greenend.org.uk/~sgtatham/putty/docs.html)

When you start PuTTY you should see a window which looks something like this:

<img src="../../../includes/media/putty1.png" title="Putty Connection Dialog" alt="Putty Connection Dialog" width="400px" />

To connect to *barcoo* you should enter *barcoo.vlsci.org.au* into the box entitled "Host Name (or IP address)" 
and *22* in the port, 
then click on the Open button. All of the settings should remain the same as they were when PuTTY started (which should be the 
same as they are in the picture above).

In some circumstances you will be presented with a window entitled PuTTY Security Alert. It will say something along the lines 
of *"The server's host key is not cached in the registry"*. This is nothing to worry about, and you should agree to continue (by 
clicking on Yes). You usually see this message the first time you try to connect to a particular remote computer.

If all goes well, a terminal window will open, showing a prompt with the text *"login as:"*. An example terminal window is shown 
below. You should type your *barcoo* username and press enter. After entering your username you will be prompted for your 
password. Assuming you type the correct username and password the system should then display a welcome message, and then 
present you with a Unix prompt. If you get this far then you are ready to start entering Unix commands and thus begin using 
the remote computer.

<img src="../../../includes/media/putty2.png" title="Putty login screen" alt="Putty login screen" width="400px" />

</details>

<p><strong>Note</strong>: for security reasons ssh will not display any characters when you enter your password. This
can be confusing because it appears as if your typing is not recognised by the computer. Don’t be
alarmed; type your password in and press return at the end.</p>
<p>barcoo is a high performance computer for Melbourne Bioinformatics users.  Logging in connects your local computer
(e.g. laptop) to barcoo, and allows you to type commands into the Unix prompt which are run on
the HPC, and have the results displayed on your local screen.</p>
<p>You will be allocated a training account on barcoo for the duration of the workshop. Your
username and password will be supplied at the start of the workshop.</p>
<p>Log out of barcoo, and log back in again (to make sure you can repeat the process).</p>
<p>All the remaining parts assume that you are logged into barcoo over ssh.</p>
<h3 id="exercises">Exercises</h3>
<h4 id="11-when-youve-logged-into-the-unix-server-run-the-following-commands-and-see-what-they-do">1.1) When you've logged into the Unix server, run the following commands and see what they do:</h4>
<ul>
<li><em>who</em></li>
<li><em>whoami</em></li>
<li><em>date</em></li>
<li><em>cal</em></li>
<li><em>hostname</em></li>
<li><em>/vlsci/TRAINING/shared/Intro_to_Unix/hi</em></li>
</ul>
<details>
  <summary>Answer</summary>

* **who**: displays a list of the users who are currently using this Unix computer.
* **whoami**: displays your username (i.e. they person currently logged in).
* **date**: displays the current date and time.
* **cal**: displays a calendar on the terminal.  It can be configured to display more than just
the current month.
* **hostname**: displays the name of the computer we are logged in to.
* **/vlsci/TRAINING/shared/Intro_to_Unix/hi**: displays the text "Hello World"

</details>

<h2 id="topic-2-exploring-your-home-directory">Topic 2: Exploring your home directory</h2>
<p>In this topic we will learn how to "look" at the filesystem and further expand our repertoire of Unix commands.</p>
<p><strong>Duration</strong>: 20 minutes. <!-- See "The shell and the command line" and "The file system" section of the workshop notes. --></p>
<p><strong>Relevant commands</strong>: <em>ls</em>, <em>pwd</em>, <em>echo</em>, <em>man</em></p>
<p>Your home directory contains your own private working space.  Your <em>current working directory</em> is automatically set
to your <em>home</em> directory when you log into a Unix computer.</p>
<h4 id="21-use-the-ls-command-to-list-the-files-in-your-home-directory-how-many-files-are-there">2.1) Use the <em>ls</em> command to list the files in your <em>home</em> directory.  How many files are there?</h4>
<details>
  <summary>Hint</summary>

Literally, type *ls* and press the *ENTER* key.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ ls
exp01  file01  muscle.fq
</code></pre>


When running the *ls* command with no options it will list files in your current working directory.  The place
where you start when you first login is your *HOME* directory.

**Answer**: 3 (exp01, file01 and muscle.fq)

</details>

<hr />
<p>The above answer is not quite correct.  There are a number of <em>hidden</em> files in your home directory as well.</p>
<h4 id="22-what-flag-might-you-use-to-display-all-files-with-the-ls-command-how-many-files-are-really-there">2.2) What <em>flag</em> might you use to display <em>all</em> files with the <em>ls</em> command?  How many files are really there?</h4>
<details>
  <summary>Hint</summary>

Take the *all* quite literally.

</details>

<details>
  <summary>Additional Hint</summary>

Type *ls --all* and press the *ENTER* key.

</details>

<details>
  <summary>Answer</summary>

**Answer 1**: *--all* (or *-a*) flag

Now you should see several files in your home directory whose names all begin with a dot. All these files are
created automatically for your user account. They are mostly configuration options for various programs including
the shell. It is safe to ignore them for the moment.


<pre><code class="sh">$ ls --all
.              .bash_logout    exp01    .lesshst
..             .bash_profile   file01   muscle.fq
.bash_history  .bashrc         .kshrc   .viminfo
</code></pre>


There are two trick files here; namely *.* and *..* which are not real files but instead, shortcuts.  *.* is a shortcut
for the current directory and *..* a shortcut for the directory above the current one.

**Answer 2**: 10 files (don't count *.* and *..*)

</details>

<hr />
<h4 id="23-what-is-the-full-path-name-of-your-home-directory">2.3) What is the full path name of your <em>home</em> directory?</h4>
<details>
  <summary>Hint</summary>

Remember your *Current Working Directory* starts in your *home* directory.

</details>

<details>
  <summary>Additional Hint</summary>

Try a shortened version of *print working directory*

</details>

<details>
  <summary>Answer</summary>

You can find out the full path name of the current working directory with the *pwd* command. Your home directory
will look something like this:


<pre><code class="sh">$ pwd
/home/trainingXX
</code></pre>


**Answer**: */vlsci/TRAINING/trainXX*

where *XX* is replaced by some 2 digit sequence.

**Alternate method**:
You can also find out the name of your home directory by printing the value of the *$HOME* shell variable:


<pre><code class="sh">echo $HOME
</code></pre>


</details>

<hr />
<h4 id="24-run-ls-using-the-long-flag-l-how-did-the-output-change">2.4) Run <em>ls</em> using the long flag (<em>-l</em>), how did the output change?</h4>
<details>
  <summary>Hint</summary>

Run *ls -l*

</details>

<details>
  <summary>Answer</summary>

**Answer**: it changed the output to place 1 file/directory per line.  It also added some extra information
about each.


<pre><code class="sh">$ ls -l
total 32
drwxr-x--- 2 training01 training 2048 Jun 14 11:28 exp01
-rw-r----- 1 training01 training   97 Jun 14 11:28 file01
-rw-r----- 1 training01 training 2461 Jun 14 11:28 muscle.fq
</code></pre>


**Details**:


<pre><code class="sh">drwxr-x--- 2 training01 training 2048 Jun 14 11:28 exp01
\--------/ ^ \--------/ \------/ \--/ \----------/ \---/
permission |  username   group   size    date       name
       /---^---\
       linkcount
</code></pre>


Where:

* **permissions**: 4 parts, file type, user perms, group perms and other perms
    * *filetype*: 1 character, *d* = directory and *-* regular file
    * *user* permissions: 3 characters, *r* = read, *w* = write, *x* = execute and *-* no permission
    * *group* permissions: same as user except for users within the owner group
    * *other* permissions: same as user except for users that are not in either user *or* *group*
* **username**: the user who *owns* this file/directory
* **group**: the group name who *owns* this file/directory
* **size**: the number of bytes this file/directory takes to store on disk
* **date**: the date and time when this file/directory was *last edited*
* **name**: name of the file
* **linkcount**: technical detail which represents the number of links this file has in the file system (safe to ignore)

</details>

<hr />
<h4 id="25-what-type-of-file-is-exp01-and-musclefq">2.5) What type of file is <em>exp01</em> and <em>muscle.fq</em>?</h4>
<details>
  <summary>Hint</summary>

Check the output from the *ls -l*.

</details>

<details>
  <summary>Answer</summary>

**Answer**:

* *exp01*: Directory (given the 'd' as the first letter of its permissions)
* *muscle.fq*: Regular File (given the '-')

</details>

<hr />
<h4 id="26-who-has-permission-to-read-write-and-execute-your-home-directory">2.6) Who has permission to <em>read</em>, <em>write</em> and <em>execute</em> your <em>home</em> directory?</h4>
<details>
  <summary>Hint</summary>

You can also give *ls* a filename as the first option.

</details>

<details>
  <summary>Additional Hint</summary>

*ls -l* will show you the contents of the *CWD*; how might you see the contents of the *parent* directory? (remember
the slides)

</details>

<details>
  <summary>Answer</summary>

If you pass the *-l* flag to ls it will display a "long" listing of file information including file permissions.

There are various ways you could find out the permissions on your home directory.


**Method 1**: given we know the *CWD* is our home directory.


<pre><code class="sh">$ ls -l ..
...
drwxr-x--- 4 trainingXY training  512 Feb  9 14:18 trainingXY
...
</code></pre>

The *..* refers to the parent directory.


**Method 2**: using $HOME.  This works no matter what our *CWD* is set to.

You could list the permissions of all files and directories in the parent directory of your home:


<pre><code class="sh">$ ls -l $HOME/..
...
drwxr-x--- 4 trainingXY training  512 Feb  9 14:18 trainingXY
...
</code></pre>


In this case we use the shell variable to refer to our home directory.


**Method 3**: using *~* (tilde) shortcut

You may also refer to your home directory using the *~* (tilde) character:


<pre><code class="sh">$ ls -l ~/..
...
drwxr-x--- 4 trainingXY training  512 Feb  9 14:18 trainingXY
...
</code></pre>


All 3 of the methods above mean the same thing.

You will see a list of files and directories in the parent directory of your home directory. One of them will
be the name of your home directory, something like *trainXX*.  Where *XX* is replaced by a two digit string.

**Altername**: using the *-a* flag and looking at the *.* (dot) special file.


<pre><code class="sh">$ ls -la
...
drwxr-x--- 4 trainingXY training  512 Feb  9 14:18 .
...
</code></pre>



**Answer**: *drwxr-x---*

* **You**: read (see filenames), write (add, delete files), execute (change your CWD to this directory).
* **Training users**: read, execute
* **Everyone else**: No access

**Discussion on Permissions**:

The permission string is *"drwxr-x---"*. The *d* means it is a directory. The *rwx* means that the owner of the directory
(your user account) can *read*, *write* and *execute* the directory. Execute permissions on a directory means that you
can *cd* into the directory. The *r-x* means that anyone in the same user group as *training* can read or execute the
directory. The *---* means that nobody else (other users on the system) can do anything with the directory.

</details>

<hr />
<div class="info">
<p><b><em>man</em> is for manual</b>: and it will be your best friend!</p>
<p>Manual pages include a lot of detail about a command and its available flags/options.  It should be your first (or second)
port of call when you are trying to work out what a command or option does.</p>
<p>You can scroll <em>up</em> and <em>down</em> in the man page using the <em>arrow</em> keys.</p>
<p>You can search in the man page using the forward
slash followed by the search text followed by the <em>ENTER</em> key. e.g.
type <em>/hello</em> and press <em>ENTER</em> to search for the word <em>hello</em>.  Press <em>n</em> key to find next
occurance of <em>hello</em> etc.</p>
<p>You can <em>quit</em> the man page by pressing <em>q</em>.</p>
</div>

<hr />
<h4 id="27-use-the-man-command-to-find-out-what-the-h-flag-does-for-ls">2.7) Use the <em>man</em> command to find out what the <em>-h</em> flag does for <em>ls</em></h4>
<details>
  <summary>Hint</summary>

Give *ls* as an option to *man* command.

</details>

<details>
  <summary>Additional Hint</summary>

*man ls*

</details>

<details>
  <summary>Answer</summary>

Use the following command to view the *man* page for *ls*:


<pre><code class="sh">$ man ls
</code></pre>


**Answer**: You should discover that the *-h* option prints file sizes in human readable format


<pre><code class="sh">-h, --human-readable
              with -l, print sizes in human readable format (e.g., 1K 234M 2G)
</code></pre>


</details>

<hr />
<h4 id="28-use-the-h-how-did-the-output-change-of-musclefq">2.8) Use the <em>-h</em>, how did the output change of <em>muscle.fq</em>?</h4>
<details>
  <summary>Hint</summary>

Don't forget the *-l* option too.

</details>

<details>
  <summary>Additional Hint</summary>

Run *ls -lh*

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ ls -lh
...
-rw-r----- 1 training01 training 2.5K Jun 14 11:28 muscle.fq
</code></pre>


**Answer**: it changed the output so the *filesize* of *muscle.fq* is now *2.5K* instead of *2461*

</details>

<h2 id="topic-3-exploring-the-file-system">Topic 3: Exploring the file system</h2>
<p>In this topic we will learn how to move around the filesystem and see what is there.</p>
<p><strong>Duration</strong>: 30 minutes. <!-- See "The file system" section of the workshop notes. --></p>
<p><strong>Relevant commands</strong>: <em>pwd</em>, <em>cd</em>, <em>ls</em>, <em>file</em></p>
<h4 id="31-print-the-value-of-your-current-working-directory">3.1) Print the value of your current working directory.</h4>
<details>
  <summary>Answer</summary>

The *pwd* command prints the value of your current working directory.


<pre><code class="sh">$ pwd
/home/training01
</code></pre>


</details>

<hr />
<h4 id="32-list-the-contents-of-the-root-directory-called-forward-slash">3.2) List the contents of the root directory, called '<em>/</em>' (forward slash).</h4>
<details>
  <summary>Hint</summary>

*ls* expects one or more anonymous options which are the files/directories to list.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ ls /
applications-merged  etc         media    root     tmp
bin                  home        mnt      sbin     usr
boot                 lib         oldhome  selinux  var
data                 lib64       opt      srv
dev                  lost+found  proc     sys
</code></pre>


Here we see that *ls* can take a filepath as its argument, which allows you to list the contents of directories
other than your current working directory.

</details>

<hr />
<h4 id="33-use-the-cd-command-to-change-your-working-directory-to-the-root-directory-did-your-prompt-change">3.3) Use the <em>cd</em> command to change your working directory to the root directory.  Did your prompt change?</h4>
<details>
  <summary>Hint</summary>

*cd* expects a single option which is the directory to change to

</details>

<details>
  <summary>Answer</summary>

The *cd* command changes the value of your current working directory. To change to the root directory use the
following command:


<pre><code class="sh">$ cd /
</code></pre>


**Answer**: Yes, it now says the CWD is */* instead of *~*.

Some people imagine that changing the working directory is akin to moving your focus within the file system.
So people often say "move to", "go to" or "charge directory to" when they want to change the working directory.

The root directory is special in Unix. It is the topmost directory in the whole file system.

</details>

<hr />
<div class="info">
<b>Output on ERROR only</b>: Many Unix commands will not produce any output if everything went well; <em>cd</em> is one
such command.  However, it will get grumpy if something went wrong by way of an error message on-screen.
</div>

<hr />
<h4 id="34-list-the-contents-of-the-cwd-and-verify-it-matches-the-list-in-32">3.4) List the contents of the CWD and verify it matches the list in 3.2</h4>
<details>
  <summary>Hint</summary>

*ls*

</details>

<details>
  <summary>Answer</summary>

Assuming you have changed to the root directory then this can be achieved with *ls*, or *ls -a* (for all files) or
*ls -la* for a long listing of all files.

If you are not currently in the root directory then you can list its contents by passing it as an argument to ls:


<pre><code class="sh">$ ls
applications-merged  etc         media    root     tmp
bin                  home        mnt      sbin     usr
boot                 lib         oldhome  selinux  var
data                 lib64       opt      srv
dev                  lost+found  proc     sys
</code></pre>


**Answer**: Yes, we got the same output as exercise 3.2



</details>

<hr />
<h4 id="35-change-your-current-working-directory-back-to-your-home-directory-what-is-the-simplest-unix-command-that-will-get-you-back-to-your-home-directory-from-anywhere-else-in-the-file-system">3.5) Change your current working directory back to your home directory. What is the simplest Unix command that will get you back to your home directory from anywhere else in the file system?</h4>
<details>
  <summary>Hint</summary>

The answer to exercise 2.6 might give some hints on how to get back to the home directory

</details>

<details>
  <summary>Additional Hint</summary>

*$HOME*, *~*, */vlsci/TRAINING/trainXX* are all methods to name your home directory.  Yet there is a simpler method; the answer
is buried in *man cd* however *cd* doesn't have its own manpage so you will need to search for it.

</details>

<details>
  <summary>Answer</summary>

Use the *cd* command to change your working directory to your home directory. There are a number of ways to refer
to your home directory:


<pre><code class="sh">cd $HOME
</code></pre>


is equivalent to:


<pre><code class="sh">cd ~
</code></pre>


The simplest way to change your current working directory to your home directory is to run the *cd* command with
no arguments:

**Answer**: the simplest for is cd with NO options.


<pre><code class="sh">cd
</code></pre>


This is a special-case behaviour which is built into *cd* for convenience.

</details>

<hr />
<h4 id="36-change-your-working-directory-to-the-following-directory">3.6) Change your working directory to the following directory:</h4>
<p><em>/vlsci/TRAINING/shared/Intro_to_Unix</em></p>
<details>
  <summary>Answer</summary>

**Answer**: *cd /vlsci/TRAINING/shared/Intro_to_Unix*

</details>

<hr />
<h4 id="37-list-the-contents-of-that-directory-how-many-files-does-it-contain">3.7) List the contents of that directory. How many files does it contain?</h4>
<details>
  <summary>Hint</summary>

*ls*

</details>

<details>
  <summary>Answer</summary>

You can do this with *ls*


<pre><code class="sh">$ ls
expectations.txt  hello.c  hi  jude.txt  moby.txt  sample_1.fastq  sleepy
</code></pre>


**Answer**: 7 files (expectations.txt  hello.c  hi  jude.txt  moby.txt  sample_1.fastq  sleepy)

</details>

<hr />
<h4 id="38-what-kind-of-file-is-vlscitrainingsharedintro_to_unixsleepy">3.8) What kind of <em>file</em> is <em>/vlsci/TRAINING/shared/Intro_to_Unix/sleepy</em>?</h4>
<details>
  <summary>Hint</summary>

Take the word *file* quite literally.

</details>

<details>
  <summary>Additional Hint</summary>

*file sleepy*

</details>

<details>
  <summary>Answer</summary>

Use the *file* command to get extra information about the contents of a file:

Assuming your current working directory is */vlsci/TRAINING/shared/Intro_to_Unix*


<pre><code class="sh">$ file sleepy
Bourne-Again shell script text executable
</code></pre>


Otherwise specify the full path of sleepy:


<pre><code class="sh">$ file /vlsci/TRAINING/shared/Intro_to_Unix/sleepy
Bourne-Again shell script text executable
</code></pre>


**Answer**: Bourne-Again shell script text executable

The "Bourne-Again shell" is more commonly known as BASH. The *file* command is telling us that sleepy
is (probably) a shell script written in the language of BASH.

The file command uses various heuristics to guess the "type" of a file. If you want to know how it works
then read the Unix manual page like so:


<pre><code class="sh">man file
</code></pre>


</details>

<hr />
<p>3.9) What kind of <em>file</em> is <em>/vlsci/TRAINING/shared/Intro_to_Unix/hi</em>?</p>
<details>
  <summary>Hint</summary>

Take the word *file* quite literally.

</details>

<details>
  <summary>Answer</summary>

Use the file command again. If you are in the same directory as *hi* then:


<pre><code class="sh">$ file hi
ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux
2.6.9, not stripped
</code></pre>


**Answer**: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux

This rather complicated output is roughly saying that the file called *hi* contains a binary executable
program (raw instructions that the computer can execute directly).

</details>

<hr />
<h4 id="310-what-are-the-file-permissions-of-the-following-file-and-what-do-they-mean">3.10) What are the file permissions of the following file and what do they mean?</h4>
<p><em>/vlsci/TRAINING/shared/Intro_to_Unix/sleepy</em></p>
<details>
  <summary>Hint</summary>

Remember the *ls* command, and don't forget the *-l* flag

</details>

<details>
  <summary>Answer</summary>

You can find the permissions of *sleepy* using the *ls* command with the *-l* flag. If you are in the same
directory as *sleepy* then:


<pre><code class="sh">$ ls -l sleepy
-rw-r--r-- 1 arobinson common 183 Feb  9 16:36 sleepy
</code></pre>


**Answer**: The Answer is dependent on the computer you are connected too however will follow something like above.
We can see that this particular instance of sleepy is owned by the user arobinson, and is part of the common
user group. It is 183 bytes in size, and was last modified on the 9th of February at 4:36pm. The file is
readable to everyone, and write-able only to arobinson.  The digit '1' between the file permission string and
the owner indicates that there is one link to the file. The Unix file system allows files to be referred to
by multiple "links". When you create a file it is referred to by one link, but you may add others later. For
future reference: links are created with the *ln* command.

</details>

<hr />
<h4 id="311-change-your-working-directory-back-to-your-home-directory-ready-for-the-next-topic">3.11) Change your working directory back to your home directory ready for the next topic.</h4>
<details>
  <summary>Hint</summary>

*cd*

</details>

<details>
  <summary>Answer</summary>

You should know how to do this with the cd command:


<pre><code class="sh">cd
</code></pre>


</details>

<h2 id="topic-4-working-with-files-and-directories">Topic 4: Working with files and directories</h2>
<p>In this topic we will start to read, create, edit and delete files and directories.</p>
<p><strong>Duration</strong>: 50 minutes.  <!-- See "Working with files" from the workshop notes. --></p>
<p><strong>Relevant commands</strong>: <em>mkdir</em>, <em>cp</em>, <em>ls</em>, <em>diff</em>, <em>wc</em>, <em>nano</em>, <em>mv</em>, <em>rm</em>, <em>rmdir</em>, <em>head</em>, <em>tail</em>, <em>grep</em>, <em>gzip</em>, <em>gunzip</em></p>
<div class="info">
<b>Hint</b>: Look at the commands above; you will need them roughly in order for this topic.  Use the <em>man</em>
command find out what they do, in particular the NAME, SYNOPSIS and DESCRIPTION sections.
</div>

<hr />
<h4 id="41-in-your-home-directory-make-a-sub-directory-called-test">4.1) In your home directory make a sub-directory called test.</h4>
<details>
  <summary>Hint</summary>

You are trying to *make a directory*, which of the above commands looks like a shortened version of this?

</details>

<details>
  <summary>Additional Hint</summary>

*mkdir*

</details>

<details>
  <summary>Answer</summary>

Make sure you are in your home directory first. If not *cd* to your home directory.

Use the *mkdir* command to make new directories:


<pre><code class="sh">$ mkdir test
</code></pre>


Use the *ls* command to check that the new directory was created.


<pre><code class="sh">$ ls
exp01  file01  muscle.fq  test
</code></pre>


</details>

<hr />
<h4 id="42-copy-all-the-files-from-the-following-directory-into-the-newly-created-test-directory">4.2) Copy all the files from the following directory into the newly created test directory:</h4>
<p><em>/vlsci/TRAINING/shared/Intro_to_Unix</em></p>
<details>
  <summary>Hint</summary>

You are trying to *copy*, which of the above commands looks like a shortened version of this?

</details>

<details>
  <summary>Additional Hint</summary>

<pre><code class="sh">$ man cp
...
SYNOPSIS
       cp [OPTION]... [-T] SOURCE DEST
...
DESCRIPTION
       Copy SOURCE to DEST, or multiple SOURCE(s) to DIRECTORY.
</code></pre>


which means *cp* expects zero or more flags, a SOURCE file followed by a DEST file or directory

</details>

<details>
  <summary>Answer</summary>

Use the *cp* command to copy files.

<div class="info"><b>Wildcards</b>: You could copy them one-by-one, but that would be tedious, so use
the <em>*</em> wildcard to specify that you want to copy all the files.
</div>

There are a number of ways you could do this depending on how you specify the source and destination
paths to *cp*. You only need to perform one of these ways, but we show multiple ones for your reference.

**Answer 1**: From your home directory:


<pre><code class="sh">$ cp /vlsci/TRAINING/shared/Intro_to_Unix/* test
</code></pre>


**Answer 2**: Change to the test directory and then copy (assuming you started in your home directory):


<pre><code class="sh">$ cd test
$ cp /vlsci/TRAINING/shared/Intro_to_Unix/* .
</code></pre>


In the example above the '*.*' (dot) character refers to the current working directory. It should be
the test subdirectory of your home directory.

**Answer 3**: Change to the \end{UNIX_TRAINING_FILES_PATH} directory and then copy:


<pre><code class="sh">cd /vlsci/TRAINING/shared/Intro_to_Unix/
cp * ~/test
</code></pre>


Remember that ~ is a shortcut reference to your home directory.

</details>

<hr />
<p><strong>Note</strong>: This exercise assumes that the copy command from the previous exercise was successful.</p>
<h4 id="43-check-that-the-file-size-of-expectationstxt-is-the-same-in-both-the-directory-that-you-copied-it-from-and-the-directory-that-you-copied-it-to">4.3) Check that the file size of <em>expectations.txt</em> is the same in both the directory that you copied it from and the directory that you copied it to.</h4>
<details>
  <summary>Hint</summary>

Remember *ls* can show you the file size (with one of its flags)

</details>

<details>
  <summary>Additional Hint</summary>

*ls -l*

</details>

<details>
  <summary>Answer</summary>

Use *ls -l* to check the size of files.

You could do this in many ways depending on the value of your working directory. We just show one possible
way for each file:


<pre><code class="sh">$ ls -l /vlsci/TRAINING/shared/Intro_to_Unix/expectations.txt

$ ls -l ~/test/expectations.txt
</code></pre>


From the output of the above commands you should be able to see the size of each file and check that they
are the same.

**Answer**: They should each be *1033773* bytes

**Alternate**: Sometimes it is useful to get file sizes reported in more "human friendly" units than bytes. If this is
true then try the *-h* option for ls:


<pre><code class="sh">$ ls -lh /vlsci/TRAINING/shared/Intro_to_Unix/expectations.txt
-rw-r--r-- 1 arobinson common 1010K Mar 26  2012 /vlsci/TRAINING/shared/Intro_to_Unix/expectations.txt
</code></pre>


In this case the size is reported in kilobytes as *1010K*. Larger files are reported in megabytes, gigabytes
etcetera.

</details>

<hr />
<p><strong>Note</strong>: this exercise assumes your working directory is <em>~/test</em>; if not run <em>cd ~/test</em></p>
<h4 id="44-check-that-the-contents-of-expectationstxt-are-the-same-in-both-the-directory-that-you-copied-it-from-and-the-directory-that-you-copied-it-to">4.4) Check that the contents of expectations.txt are the same in both the directory that you copied it from and the directory that you copied it to.</h4>
<details>
  <summary>Hint</summary>

What is the opposite of *same*?

</details>

<details>
  <summary>Additional Hint</summary>

*diff*erence

</details>

<details>
  <summary>Answer</summary>

Use the *diff* command to compare the contents of two files.


<pre><code class="sh">$ diff /vlsci/TRAINING/shared/Intro_to_Unix/expectations.txt expectations.txt
</code></pre>


If the two files are identical the *diff* command will NOT produce any output)

**Answer**: Yes, they are the same since no output was given.

</details>

<hr />
<h4 id="45-how-many-lines-words-and-characters-are-in-expectationstxt">4.5) How many lines, words and characters are in expectations.txt?</h4>
<details>
  <summary>Hint</summary>

Initialisms are key

</details>

<details>
  <summary>Additional Hint</summary>

*w*ord *c*ount

</details>

<details>
  <summary>Answer</summary>

Use the *wc* (for "word count") to count the number of characters, lines and words in a file:


<pre><code class="sh">$ wc expectations.txt
  20415  187465 1033773 expectations.txt
</code></pre>


**Answer**: There are *20415* lines, *187465* words and *1033773* characters in expectations.txt.

To get just the line, word or character count:


<pre><code class="sh">$ wc -l expectations.txt
20415 expectations.txt
$ wc -w expectations.txt
187465 expectations.txt
$ wc -c expectations.txt
1033773 expectations.txt
</code></pre>


</details>

<hr />
<h4 id="46-open-testexpectationstxt-in-the-nano-text-editor-delete-the-first-line-of-text-and-save-your-changes-to-the-file-exit-nano">4.6) Open <em>~/test/expectations.txt</em> in the <em>nano</em> text editor, delete the first line of text, and save your changes to the file. Exit <em>nano</em>.</h4>
<details>
  <summary>Hint</summary>

*nano FILENAME*

Once *nano* is open it displays some command hints along the bottom of the screen.

</details>

<details>
  <summary>Additional Hint</summary>

*^O* means hold the *Control* (or CTRL) key while pressing the *o*.  Despite what it displays, you need to type
the lower-case letter that follows the *^* character.

WriteOut is another name for Save.

</details>

<details>
  <summary>Answer</summary>

Take some time to play around with the *nano* text editor.

*Nano* is a very simple text editor which is easy to use but limited in features. More powerful
editors exist such as *vim* and *emacs*, however they take a substantial amount of time to learn.

</details>

<hr />
<h4 id="47-did-the-changes-you-made-to-testexpectationstxt-have-any-effect-on-vlscitrainingsharedintro_to_unix">4.7) Did the changes you made to <em>~/test/expectations.txt</em> have any effect on <em>/vlsci/TRAINING/shared/Intro_to_Unix</em>?</h4>
<p>How can you tell if two files are the same or different in their contents?</p>
<details>
  <summary>Hint</summary>

Remember exercise 4.4

</details>

<details>
  <summary>Additional Hint</summary>

Use *diff*

</details>

<details>
  <summary>Answer</summary>

Use *diff* to check that the two files are different after you have made the change to the copy of
*expectations.txt* in your *~/test* directory.


<pre><code class="sh">diff ~/test/expectations.txt \
/vlsci/TRAINING/shared/Intro_to_Unix/expectations.txt
</code></pre>


You could also use *ls* to check that the files have different sizes.

</details>

<hr />
<h4 id="48-in-your-test-subdirectory-rename-expectationstxt-to-footxt">4.8) In your <em>test</em> subdirectory, rename <em>expectations.txt</em> to <em>foo.txt</em>.</h4>
<details>
  <summary>Hint</summary>

Another way to think of it is *moving* it from *expectations.txt* to *foo.txt*

</details>

<details>
  <summary>Additional Hint</summary>

*mv*

Use *man mv* if you need to work out how to use it.

</details>

<details>
  <summary>Answer</summary>

Use the *mv* command to rename the file:


<pre><code class="sh">$ mv expectations.txt foo.txt
$ ls
foo.txt  hello.c  hi  jude.txt  moby.txt  sample_1.fastq  sleepy
</code></pre>


</details>

<hr />
<h4 id="49-rename-footxt-back-to-expectationstxt">4.9) Rename foo.txt back to expectations.txt.</h4>
<details>
  <summary>Answer</summary>

Use the *mv* command to rename the file:


<pre><code class="sh">$ mv foo.txt expectations.txt
$ ls
expectations.txt  hello.c  hi  jude.txt  moby.txt  sample_1.fastq  sleepy
</code></pre>


Use *ls* to check that the file is in fact renamed.

</details>

<hr />
<h4 id="410-remove-the-file-expectationstxt-from-your-test-directory">4.10) Remove the file <em>expectations.txt</em> from your <em>test</em> directory.</h4>
<details>
  <summary>Hint</summary>

We are trying to *remove* a file, check the commands at the top of this topic.

</details>

<details>
  <summary>Additional Hint</summary>

*rm*

</details>

<details>
  <summary>Answer</summary>

Use the *rm* command to remove files (carefully):


<pre><code class="sh">$ rm expectations.txt
$ ls
hello.c  hi  jude.txt  moby.txt  sample_1.fastq  sleepy
</code></pre>


</details>

<hr />
<h4 id="411-remove-the-entire-test-directory-and-all-the-files-within-it">4.11) Remove the entire <em>test</em> directory and all the files within it.</h4>
<details>
  <summary>Hint</summary>

We are trying to *remove a directory*.

</details>

<details>
  <summary>Additional Hint</summary>

You could use *rmdir* but there is an easier way using just *rm* and a flag.

</details>

<details>
  <summary>Answer</summary>

You could use the *rm* command to remove each file individually, and then use the *rmdir* command
to remove the directory. Note that *rmdir* will only remove directories that are empty (i.e. do not
contain files or subdirectories).

A faster way is to pass the *-r* (for recursive) flag to *rm* to remove all the files and the
directory in one go:

**Logical Answer**:

<pre><code class="sh">cd ~
rm test/*
rmdir test
</code></pre>


**Easier Answer**:

<pre><code class="sh">cd ~
rm -r test
</code></pre>


<div class="error"><b>Warning</b>: Be very careful with <em>rm -r</em>, it will remove all files
and all subdirectories underneath the specified directory. This could be catastrophic if you do it
in the wrong location! Now is a good moment to pause and think about file backup strategies.</div>

</details>

<hr />
<h4 id="412-recreate-the-test-directory-in-your-home-directory-and-copy-all-the-files-from-vlscitrainingsharedintro_to_unix-back-into-the-test-directory">4.12) Recreate the test directory in your home directory and copy all the files from <em>/vlsci/TRAINING/shared/Intro_to_Unix</em> back into the test directory.</h4>
<details>
  <summary>Hint</summary>

See exercises 4.1 and 4.2

</details>

<details>
  <summary>Answer</summary>

Repeat exercises 4.1 and 4.2.


<pre><code class="sh">$ cd ~
$ mkdir test
$ cp /vlsci/TRAINING/shared/Intro_to_Unix/* test
</code></pre>


</details>

<hr />
<h4 id="413-change-directories-to-test-and-use-the-cat-command-to-display-the-entire-contents-of-the-file-helloc">4.13) Change directories to <em>~/test</em> and use the <em>cat</em> command to display the entire contents of the file <em>hello.c</em></h4>
<details>
  <summary>Hint</summary>

Use *man* if you can't guess how it might work.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ cd ~/test
$ cat hello.c
#include &lt;stdio.h&gt;
int main(void) {
    printf (&quot;Hello World\n&quot;);
    return 0;
}
</code></pre>


*hello.c* contains the source code of a C program. The compiled executable version of this code
is in the file called *hi*, which you can run like so:


<pre><code class="sh">$ ./hi
Hello World
</code></pre>


</details>

<hr />
<h4 id="414-use-the-head-command-to-view-the-first-20-lines-of-the-file-sample_1fastq">4.14) Use the <em>head</em> command to view the first <em>20</em> lines of the file <em>sample_1.fastq</em></h4>
<details>
  <summary>Hint</summary>

Remember your *best* friend!

</details>

<details>
  <summary>Additional Hint</summary>

Use *man* to find out what option you need to add to display a given number of *lines*.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ head -20 sample_1.fastq
@IRIS:7:1:17:394#0/1
GTCAGGACAAGAAAGACAANTCCAATTNACATTATG
+IRIS:7:1:17:394#0/1
aaabaa`]baaaaa_aab]D^^`b`aYDW]abaa`^
@IRIS:7:1:17:800#0/1
GGAAACACTACTTAGGCTTATAAGATCNGGTTGCGG
+IRIS:7:1:17:800#0/1
ababbaaabaaaaa`]`ba`]`aaaaYD\\_a``XT
@IRIS:7:1:17:1757#0/1
TTTTCTCGACGATTTCCACTCCTGGTCNACGAATCC
+IRIS:7:1:17:1757#0/1
aaaaaa``aaa`aaaa_^a```]][Z[DY^XYV^_Y
@IRIS:7:1:17:1479#0/1
CATATTGTAGGGTGGATCTCGAAAGATATGAAAGAT
+IRIS:7:1:17:1479#0/1
abaaaaa`a```^aaaaa`_]aaa`aaa__a_X]``
@IRIS:7:1:17:150#0/1
TGATGTACTATGCATATGAACTTGTATGCAAAGTGG
+IRIS:7:1:17:150#0/1
abaabaa`aaaaaaa^ba_]]aaa^aaaaa_^][aa
</code></pre>


</details>

<hr />
<h4 id="415-use-the-tail-command-to-view-the-last-8-lines-of-the-file-sample_1fastq">4.15) Use the <em>tail</em> command to view the last <em>8</em> lines of the file <em>sample_1.fastq</em></h4>
<details>
  <summary>Hint</summary>

It's very much like *head*.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">tail -8 sample_1.fastq
@IRIS:7:32:731:717#0/1
TAATAATTGGAGCCAAATCATGAATCAAAGGACATA
+IRIS:7:32:731:717#0/1
ababbababbab]abbaa`babaaabbb`bbbabbb
@IRIS:7:32:731:1228#0/1
CTGATGCCGAGGCACGCCGTTAGGCGCGTGCTGCAG
+IRIS:7:32:731:1228#0/1
`aaaaa``aaa`a``a`^a`a`a_[a_a`a`aa`__
</code></pre>


</details>

<hr />
<h4 id="416-use-the-grep-command-to-find-out-all-the-lines-in-mobytxt-that-contain-the-word-ahab">4.16) Use the <em>grep</em> command to find out all the lines in <em>moby.txt</em> that contain the word "Ahab"</h4>
<details>
  <summary>Hint</summary>

One might say we are 'looking for the *pattern* "Ahab"'

</details>

<details>
  <summary>Additional Hint</summary>

<pre><code class="sh">$ man grep
...
SYNOPSIS
       grep [OPTIONS] PATTERN [FILE...]
...
</code></pre>


</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ grep Ahab moby.txt
&quot;Want to see what whaling is, eh? Have ye clapped eye on Captain Ahab?&quot;
&quot;Who is Captain Ahab, sir?&quot;
&quot;Aye, aye, I thought so. Captain Ahab is the Captain of this ship.&quot;
... AND MUCH MUCH MORE ...
</code></pre>


If you want to know how many lines are in the output of the above command you can "pipe" it
into the *wc -l* command:


<pre><code class="sh">$ grep Ahab moby.txt | wc -l
491
</code></pre>


which shows that there are *491* lines in *moby.txt* that contain the word Ahab.

</details>

<hr />
<h4 id="417-use-the-grep-command-to-find-out-all-the-lines-in-expectationstxt-that-contain-the-word-the-with-a-case-insensitive-search-it-should-count-the-the-the-the-etcetera">4.17) Use the <em>grep</em> command to find out all the lines in <em>expectations.txt</em> that contain the word "the" with a case insensitive search (it should count "the" "The" "THE" "tHe" etcetera).</h4>
<details>
  <summary>Hint</summary>

One might say we are *ignoring case*.

</details>

<details>
  <summary>Additional Hint</summary>

<pre><code class="sh">$ man grep
...
       -i, --ignore-case
              Ignore case distinctions in both the PATTERN and the input files.  (-i is specified by POSIX.)
...
</code></pre>


</details>

<details>
  <summary>Answer</summary>

Use the *-i* flag to *grep* to make it perform case insensitive search:


<pre><code class="sh">$ grep -i the expectations.txt
The Project Gutenberg EBook of Great Expectations, by Charles Dickens
This eBook is for the use of anyone anywhere at no cost and with
re-use it under the terms of the Project Gutenberg License included
[Project Gutenberg Editor's Note: There is also another version of
... AND MUCH MUCH MORE ...
</code></pre>


Again, "pipe" the output to *wc -l* to count the number of lines:


<pre><code class="sh">$ grep -i the expectations.txt  | wc -l
8165
</code></pre>


</details>

<hr />
<h4 id="418-use-the-gzip-command-to-compress-the-file-sample_1fastq-use-gunzip-to-decompress-it-back-to-the-original-contents">4.18) Use the <em>gzip</em> command to compress the file <em>sample_1.fastq</em>. Use <em>gunzip</em> to decompress it back to the original contents.</h4>
<details>
  <summary>Hint</summary>

Use the above commands along with *man* and *ls* to see what happens to the file.

</details>

<details>
  <summary>Answer</summary>

Check the file size of sample_1.fastq before compressing it:


<pre><code class="sh"># check filesize
$ ls -l sample_1.fastq
-rw-r--r-- 1 training01 training 90849644 Jun 14 20:03 sample_1.fastq

# compress it (takes a few seconds)
$ gzip sample_1.fastq

# check filesize (Note: its name changed)
$ ls -l sample_1.fastq.gz
-rw-r--r-- 1 training01 training 26997595 Jun 14 20:03 sample_1.fastq.gz

# decompress it
$ gunzip sample_1.fastq.gz

$ ls -l sample_1.fastq
-rw-r--r-- 1 training01 training 90849644 Jun 14 20:03 sample_1.fastq
</code></pre>


You will see that when it was compressed it is *26997595* bytes in size, making it about *0.3* times the size of the
original file.

**Note**: in the above section the lines starting with *#* are comments so don't need to be copied but if you
do then they wont do anything.

</details>

<h2 id="topic-5-pipes-output-redirection-and-shell-scripts">Topic 5: Pipes, output redirection and shell scripts</h2>
<p>In this section we will cover a lot of the more advanced Unix concepts; it is here where you will start to see
the power of Unix.  I say <em>start</em> because this is only the "tip of the iceberg".</p>
<p><strong>Duration</strong>: 50 minutes. <!-- See "Processes" from the workshop notes. --></p>
<p><strong>Relevant commands</strong>: <em>wc</em>, <em>paste</em>, <em>grep</em>, <em>sort</em>, <em>uniq</em>, <em>nano</em>, <em>cut</em></p>
<h4 id="51-how-many-reads-are-contained-in-the-file-sample_1fastq">5.1) How many <em>reads</em> are contained in the file <em>sample_1.fastq</em>?</h4>
<details>
  <summary>Hint</summary>

Examine some of the file to work out how many lines each *read* takes up.

</details>

<details>
  <summary>Additional Hint</summary>

Count the number of lines

</details>

<details>
  <summary>Answer</summary>

We can answer this question by counting the number of lines in the file and dividing by 4:


<pre><code class="sh">$ wc -l sample_1.fastq
3000000
</code></pre>


**Answer**: There are *3000000* lines in the file representing *750000* reads.

If you want to do simple arithmetic at the command line then you can use the "basic calculator"
called *bc*:


<pre><code class="sh">$ echo &quot;3000000 / 4&quot; | bc
750000
</code></pre>


<div class="info"><b>Note</b>: that the vertical bar character "|" is the Unix pipe (and is often
called the "pipe symbol"). It is used for connecting the output of one command into the input of
another command. We'll see more examples soon.</div>

*bc* is suitable for small calculations, but it becomes cumbersome for more complex examples. If
you want to do more sophisticated calculations then we recommend to use a more general purpose
programming language (such as Python etcetera).

</details>

<hr />
<h4 id="52-how-many-reads-in-sample_1fastq-contain-the-sequence-gattaca">5.2) How many reads in <em>sample_1.fastq</em> contain the sequence <em>GATTACA</em>?</h4>
<details>
  <summary>Hint</summary>

Check out exercise 4.16

</details>

<details>
  <summary>Answer</summary>

Use *grep* to find all the lines that contain *GATTACA* and "pipe" the output to *wc -l* to count them:


<pre><code class="sh">$ grep GATTACA sample_1.fastq | wc -l
1119
</code></pre>


**Answer**: *1119*

If you are unsure about the possibility of upper and lower case characters then consider using
the *-i* (ignore case option for grep).

</details>

<hr />
<h4 id="53-on-what-line-numbers-do-the-sequences-containing-gattaca-occur">5.3) On what line numbers do the sequences containing <em>GATTACA</em> occur?</h4>
<details>
  <summary>Hint</summary>

We are looking for the *line numbers*.

</details>

<details>
  <summary>Additional Hint</summary>

Check out the manpage for *grep* and/or *nl*

</details>

<details>
  <summary>Answer</summary>

You can use the *-n* flag to grep to make it prefix each line with a line number:

**Answer 1**:

<pre><code class="sh">$ grep -n GATTACA sample_1.fastq
5078:AGGAAGATTACAACTCCAAGACACCAAACAAATTCC
7170:AACTACAAAGGTCAGGATTACAAGCTCTTGCCCTTC
8238:ATAGTTTTTTCGATTACATGGATTATATCTGTTTGC
... AND MUCH MUCH MORE ...
</code></pre>


**Answer 2**: Or you can use the *nl* command to number each line of sample_1.fastq and then search for *GATTACA*
in the numbered lines:


<pre><code class="sh">$ nl sample_1.fastq | grep GATTACA
  5078  AGGAAGATTACAACTCCAAGACACCAAACAAATTCC
  7170  AACTACAAAGGTCAGGATTACAAGCTCTTGCCCTTC
  8238  ATAGTTTTTTCGATTACATGGATTATATCTGTTTGC
... AND MUCH MUCH MORE ...
</code></pre>


**Just the line numbers**:

If you just want to see the line numbers then you can "pipe" the output of the above command into
*cut -f 1*:


<pre><code class="sh">$ nl sample_1.fastq | grep GATTACA | cut -f 1
  5078
  7170
  8238
... AND MUCH MUCH MORE ...
</code></pre>


*cut* will remove certain columns from the input; in this case it will remove all except column 1
(a.k.a. field 1, hence the *-f 1* option)


<pre><code class="sh">$ grep -n GATTACA sample_1.fastq | cut -d: -f 1
5078
7170
8238
... AND MUCH MUCH MORE ...
</code></pre>


</details>

<hr />
<h4 id="54-use-the-nl-command-to-print-each-line-of-sample_1fastq-with-its-corresponding-line-number-at-the-beginning">5.4) Use the <em>nl</em> command to print each line of <em>sample_1.fastq</em> with its corresponding line number at the beginning.</h4>
<details>
  <summary>Hint</summary>

Check answer to 5.3.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ nl sample_1.fastq
     1  @IRIS:7:1:17:394#0/1
     2  GTCAGGACAAGAAAGACAANTCCAATTNACATTATG
     3  +IRIS:7:1:17:394#0/1
     4  aaabaa`]baaaaa_aab]D^^`b`aYDW]abaa`^
     5  @IRIS:7:1:17:800#0/1
     6  GGAAACACTACTTAGGCTTATAAGATCNGGTTGCGG
     7  +IRIS:7:1:17:800#0/1
     8  ababbaaabaaaaa`]`ba`]`aaaaYD\\_a``XT
... AND MUCH MUCH MORE ...
</code></pre>


There are a lot of lines in that file so this command might take a while to print all its output.
If you get tired of looking at the output you can kill the command with *control-c* (hold the
*control* key down and simultaneously press the "*c*" character).

</details>

<hr />
<h4 id="55-redirect-the-output-of-the-previous-command-to-a-file-called-sample_1fastqnl">5.5) Redirect the output of the previous command to a file called <em>sample_1.fastq.nl</em>.</h4>
<p>Check the first <em>20</em> lines of <em>sample_1.fastq.nl</em> with the <em>head</em> command. Use the <em>less</em> command to
interactively view the contents of <em>sample_1.fastq.nl</em> (use the arrow keys to navigate up and down,
<em>q</em> to quit and '<em>/</em>' to search). Use the search facility in less to find occurrences of
<em>GATTACA</em>.</p>
<details>
  <summary>Hint</summary>

Ok that one was tough, *> FILENAME* is how you do it if you didn't break out an internet search for
"redirect the output in Unix"

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ nl sample_1.fastq &gt; sample_1.fastq.nl
</code></pre>


The greater-than sign "*>*" is the file redirection operator. It causes the standard output of the
command on the left-hand-side to be written to the file on the right-hand-side.

You should notice that the above command is much faster than printing the output to the screen.
This is because writing to disk can be performed much more quickly than rendering the output on
a terminal.

To check that the first 20 lines of the file look reasonable you can use the *head* command like so:


<pre><code class="sh">$ head -20 sample_1.fastq.nl
     1  @IRIS:7:1:17:394#0/1
     2  GTCAGGACAAGAAAGACAANTCCAATTNACATTATG
     3  +IRIS:7:1:17:394#0/1
     4  aaabaa`]baaaaa_aab]D^^`b`aYDW]abaa`^
     5  @IRIS:7:1:17:800#0/1
     6  GGAAACACTACTTAGGCTTATAAGATCNGGTTGCGG
     7  +IRIS:7:1:17:800#0/1
     8  ababbaaabaaaaa`]`ba`]`aaaaYD\\_a``XT
...
</code></pre>


The *less* command allows you to interactively view a file. The arrow keys move the page up and
down. You can search using the '*/*' followed by the search term. You can quit by pressing "*q*". Note
that the *less* command is used by default to display man pages.


<pre><code class="sh">$ less sample_1.fastq.nl
</code></pre>


</details>

<hr />
<h4 id="56-the-four-lines-per-read-format-of-fastq-is-cumbersome-to-deal-with-often-it-would-be-preferable-if-we-could-convert-it-to-tab-separated-value-tsv-format-such-that-each-read-appears-on-a-single-line-with-each-of-its-fields-separated-by-tabs-use-the-following-command-to-convert-sample_1fastq-into-tsv-format">5.6) The four-lines-per-read format of FASTQ is cumbersome to deal with. Often it would be preferable if we could convert it to tab-separated-value (TSV) format, such that each read appears on a single line with each of its fields separated by tabs. Use the following command to convert sample_1.fastq into TSV format:</h4>
<pre><code class="sh">$ cat sample_1.fastq | paste - - - - &gt; sample_1.tsv
</code></pre>

<details>
  <summary>Answer</summary>

The *'-'* (dash) character has a special meaning when used in place of a file; it means use the standard
input instead of a real file.  Note: while it is fairly common in most Unix programs, not all will support it.

The *paste* command is useful for merging multiple files together line-by-line, such that the *Nth*
line from each file is joined together into one line in the output, separated by default with a
*tab* character. In the above example we give paste 4 copies of the contents of *sample_1.fastq*,
which causes it to join consecutive groups of 4 lines from the file into one line of output.

</details>

<hr />
<h4 id="57-do-you-expect-the-output-of-the-following-command-to-produce-the-same-output-as-above-and-why">5.7) Do you expect the output of the following command to produce the same output as above? and why?</h4>
<pre><code class="sh">$ paste sample_1.fastq sample_1.fastq sample_1.fastq sample_1.fastq &gt; sample_1b.tsv
</code></pre>

<p>Try it, see what ends up in sample_1b.tsv (maybe use <em>less</em>)</p>
<details>
  <summary>Hint</summary>

Use *less* to examine it.

</details>

<details>
  <summary>Answer</summary>

**Answer**: No, in the second instance we get 4 copies of each line.

**Why**: In the first command *paste* will use the input file (standard input) 4 times since the *cat*
command will only give one copy of the file to *paste*, where as, in the second command *paste* will open
the file 4 times.  Note: this is quite confusing and is not necessory to remember; its just an interesting
side point.

</details>

<hr />
<h4 id="58-check-that-sample_1tsv-has-the-correct-number-of-lines-use-the-head-command-to-view-the-first-20-lines-of-the-file">5.8) Check that <em>sample_1.tsv</em> has the correct number of lines. Use the <em>head</em> command to view the first <em>20</em> lines of the file.</h4>
<details>
  <summary>Hint</summary>

Remember the *wc* command.

</details>

<details>
  <summary>Answer</summary>

We can count the number of lines in *sample_1.tsv* using *wc*:


<pre><code class="sh">$ wc -l sample_1.tsv
</code></pre>


The output should be *750000* as expected (1/4 of the number of lines in sample_1.fastq).

To view the first *20* lines of *sample_1.tsv* use the *head* command:


<pre><code class="sh">$ head -20 sample_1.tsv
</code></pre>


</details>

<hr />
<h4 id="59-use-the-cut-command-to-print-out-the-second-column-of-sample_1tsv-redirect-the-output-to-a-file-called-sample_1dnatxt">5.9) Use the <em>cut</em> command to print out the second column of <em>sample_1.tsv</em>. Redirect the output to a file called <em>sample_1.dna.txt</em>.</h4>
<details>
  <summary>Hint</summary>

See exercise 5.3 (for cut) and 5.5 (redirection)

</details>

<details>
  <summary>Answer</summary>

The file sample_1.tsv is in column format. The cut command can be used to select certain columns
from the file. The DNA sequences appear in column 2, we select that column using the -f 2 flag
(the f stands for "field").


<pre><code class="sh">cut -f 2 sample_1.tsv &gt; sample_1.dna.txt
</code></pre>


Check that the output file looks reasonable using *head* or *less*.

</details>

<hr />
<h4 id="510-use-the-sort-command-to-sort-the-lines-of-sample_1dnatxt-and-redirect-the-output-to-sample_1dnasortedtxt-use-head-to-look-at-the-first-few-lines-of-the-output-file-you-should-see-a-lot-of-repeated-sequences-of-as">5.10) Use the <em>sort</em> command to sort the lines of <em>sample_1.dna.txt</em> and redirect the output to <em>sample_1.dna.sorted.txt</em>. Use <em>head</em> to look at the first few lines of the output file. You should see a lot of repeated sequences of As.</h4>
<details>
  <summary>Hint</summary>

Use *man* (sort) and see exercise 5.5 (redirection)

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ sort sample_1.dna.txt &gt; sample_1.dna.sorted.txt
</code></pre>


Running *head* on the output file reveals that there are duplicate DNA sequences in the input FASTQ
file.

</details>

<hr />
<h4 id="511-use-the-uniq-command-to-remove-duplicate-consecutive-lines-from-sample_1dnasortedtxt-redirect-the-result-to-sample_1dnauniqtxt-compare-the-number-of-lines-in-sample1_dnatxt-to-the-number-of-lines-in-sample_1dnauniqtxt">5.11) Use the <em>uniq</em> command to remove duplicate consecutive lines from <em>sample_1.dna.sorted.txt</em>, redirect the result to <em>sample_1.dna.uniq.txt</em>. Compare the number of lines in sample1_dna.txt to the number of lines in <em>sample_1.dna.uniq.txt</em>.</h4>
<details>
  <summary>Hint</summary>

I am pretty sure you have already used *man* (or just guessed how to use *uniq*).  You're also a gun at
redirection now.

</details>

<details>
  <summary>Answer</summary>

<pre><code class="sh">$ uniq sample_1.dna.sorted.txt &gt; sample_1.dna.uniq.txt
</code></pre>


Compare the outputs of:


<pre><code class="sh">$ wc -l sample_1.dna.sorted.txt
750000
$ wc -l sample_1.dna.uniq.txt
614490
</code></pre>


View the contents of *sample_1.dna.uniq.txt* to check that the duplicate DNA sequences have been
removed.

</details>

<hr />
<h4 id="512-can-you-modify-the-command-from-above-to-produce-only-those-sequences-of-dna-which-were-duplicated-in-sample_1dnasortedtxt">5.12) Can you modify the command from above to produce <em>only</em> those sequences of DNA which were duplicated in <em>sample_1.dna.sorted.txt</em>?</h4>
<details>
  <summary>Hint</summary>

Checkout the *uniq* manpage

</details>

<details>
  <summary>Additional Hint</summary>

Look at the man page for uniq.

</details>

<details>
  <summary>Answer</summary>

Use the *-d* flag to *uniq* to print out only the duplicated lines from the file:


<pre><code class="sh">$ uniq -d sample_1.dna.sorted.txt &gt; sample_1.dna.dup.txt
</code></pre>


</details>

<hr />
<h4 id="513-write-a-shell-pipeline-which-will-print-the-number-of-duplicated-dna-sequences-in-sample_1fastq">5.13) Write a <em>shell pipeline</em> which will print the number of duplicated DNA sequences in sample_1.fastq.</h4>
<details>
  <summary>Hint</summary>

That is, *piping* most of the commands you used above instead of redirecting to file

</details>

<details>
  <summary>Additional Hint</summary>

i.e. 6 commands (*cat*, *paste*, *cut*, *sort*, *uniq*, *wc*)

</details>

<details>
  <summary>Answer</summary>

Finally we can 'pipe' all the pieces together into a sophisticated pipeline which starts with a
FASTQ file and ends with a list of duplicated DNA sequences:

**Answer**:

<pre><code class="sh">$ cat sample_1.fastq | paste - - - - | cut -f 2 | sort | uniq -d | wc -l
56079
</code></pre>


The output file should have *56079* lines.

</details>

<hr />
<h4 id="514-advanced-write-a-shell-script-which-will-print-the-number-of-duplicated-dna-sequences-in-sample_1fastq">5.14) (Advanced) Write a shell script which will print the number of duplicated DNA sequences in sample_1.fastq.</h4>
<details>
  <summary>Hint</summary>

Check out the *sleepy* file (with *cat* or *nano*); there is a bit of magic on the first line that you will need.

You also need to tell bash that this file can be executed (check out *chmod* command).

</details>

<details>
  <summary>Answer</summary>

Put the answer to *5.13* into a file called *sample_1_dups.sh* (or whatever you want). Use *nano* to
create the file.

**Answer**: the contents of the file will look like this:


<pre><code class="sh">#!/bin/bash

cat sample_1.fastq | paste - - - - | cut -f 2 | sort | uniq -d | wc -l
</code></pre>


<div class="info"><b>Note</b>: the first line has special meaning.  If it starts with '<em>#!</em>' (Hash
then exclamation mark) then it tells bash this file is a script that can be interpreted.  The command
(including full path) used to intepret the script is placed right after the magic code.</div>

Give everyone execute permissions on the file with chmod:


<pre><code class="sh">$ chmod +x sample_1_dups.sh
</code></pre>


You can run the script like so:


<pre><code class="sh">$ ./sample_1_dups.sh
</code></pre>


If all goes well the script should behave in exactly the same way as the answer to 5.13.

</details>

<hr />
<h4 id="515-advanced-modify-your-shell-script-so-that-it-accepts-the-name-of-the-input-fastq-file-as-a-command-line-parameter">5.15) (Advanced) Modify your shell script so that it accepts the name of the input FASTQ file as a command line parameter.</h4>
<details>
  <summary>Hint</summary>

Shell scripts can refer to command line arguments by their position using special variables called
*$0*, *$1*, *$2* and so on.

</details>

<details>
  <summary>Additional Hint</summary>

*$0* refers to the name of the script as it was called on the command line.
*$1* refers to the first command line argument, and so on.

</details>

<details>
  <summary>Answer</summary>

Copy the shell script from *5.14* into a new file:


<pre><code class="sh">$ cp sample_1_dups.sh fastq_dups.sh
</code></pre>


Edit the new shell script file and change it to use the command line parameters:


<pre><code class="sh">#!/bin/bash

cat $1 | paste - - - - | cut -f 2 | sort | uniq -d | wc -l
</code></pre>


You can run the new script like so:


<pre><code class="sh">$ ./fastq_dups.sh sample_1.fastq
</code></pre>


In the above example the script takes *sample_1.fastq* as input and prints the number of duplicated
sequences as output.

**A better Answer**:

Ideally we would write our shell script to be more robust. At the moment it just assumes there
will be at least one command line argument. However, it would be better to check and produce an
error message if insufficient arguments were given:


<pre><code class="sh">#!/bin/bash
if [ $# -eq 1 ]; then
    cat $1 | paste - - - - | cut -f 2 | sort | uniq -d | wc -l
else
    echo &quot;Usage: $0 &lt;fastq_filename&gt;&quot;
    exit 1
fi
</code></pre>


The '*if ...; then*' line means: do the following line(s) ONLY if the *...* (called condition) bit is true.

The '*else*' line means: otherwise do the following line(s) instead.  Note: it is optional.

The '*fi*' line means: this marks the end of the current *if* or *else* section.

The '*[ $# -eq 1 ]*' part is the condition:

* *$#*: is a special shell variable that indicates how many command line arguments were given.
* *-eq*: checks if the numbers on either side of it are equal.
* *1*: is a number one

<div class="warning"><b>Spaces in conditions</b>:
Bash is VERY picky about the spaces within the conditions; if you get it wrong it will just behave strangely
(without warning).  You MUST put a space near the share brackets and between each part of the condition!</div>

So in words our script is saying "if user provided 1 filename, then count the duplicates, otherwise print an error".

<div class="info"><b>Exit-status</b>:
It is a Unix standard that when the user provides incorrect commandline arguments we print a usage message
and return a *non-zero* exit status.  The *exit status* is a standard way for other programs to know if
our program ran correctly; 0 means everything went as expected, any other number is an error.  If you don't
provide an *exit ..* line then it automatically returns a 0 for you.</div>

</details>

<hr />
<h4 id="516-advanced-modify-your-shell-script-so-that-it-accepts-zero-or-more-fastq-files-on-the-command-line-argument-and-outputs-the-number-of-duplicated-dna-sequences-in-each-file">5.16) (Advanced) Modify your shell script so that it accepts zero or more FASTQ files on the command line argument and outputs the number of duplicated DNA sequences in each file.</h4>
<details>
  <summary>Answer</summary>

We can add a loop to our script to accept multiple input FASTQ files:


<pre><code class="sh">#!/bin/bash
for file in $@; do
    dups=$(cat $file | paste - - - - | cut -f 2 | sort | uniq -d | wc -l)
    echo &quot;$file $dups&quot;
done
</code></pre>


There's a lot going on in this script.

The *$@* is a sequence of all command line arguments.

The '*for ...; do*' (a.k.a. for loop) iterates over that sequence one argument at a time, assigning the current argument in
the sequence to the variable called *file*.

The *$(...)* allow us to capture the output of another command (in-place of the *...*).  In this
case we capture the output of the pipeline and save it to the variable called *dups*.

If you had multiple FASTQ files available you could run the script like so:


<pre><code class="sh">./fastq_dups.sh sample_1.fastq sample_2.fastq sample_3.fastq
</code></pre>


And it would produce output like:


<pre><code class="text">sample_1.fastq 56079
sample_2.fastq XXXXX
sample_3.fastq YYYYY
</code></pre>


</details>

<h2 id="finished">Finished</h2>
<p>Well done, you learnt a lot over the last 5 topics and you should be proud of your achievement; it
was a lot to take in.</p>
<p>From here you should be comfortable around the Unix command line and ready to take on the HPC
Workshop.</p>
<p>You will no-doubt forget a lot of what you learnt here so I encourage you to save a link to this
workshop for later reference.</p>
<p>Thank you for your attendance, please don't forget to complete the training survey and give it
back to the workshop facilitators.</p></article></div></div></main><footer class="md-footer"><div class="md-footer-nav"><nav class="md-footer-nav__inner md-grid"><a href="../../hpc/hpc/" title="Introduction to HPC" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev"><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-back md-footer-nav__button"></i></div><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Previous</span>Introduction to HPC</span></div></a><a href="../../gvl_launch/gvl_launch/" title="Launching a Personal GVL Server" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next"><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Next</span>Launching a Personal GVL Server</span></div><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i></div></a></nav></div><div class="md-footer-meta md-typeset"><div class="md-footer-meta__inner md-grid"><div class="md-footer-copyright"><div class="md-footer-copyright__highlight"><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a></div>powered by <a href="https://www.mkdocs.org">MkDocs</a> and <a href="https://squidfunk.github.io/mkdocs-material/">Material for MkDocs</a></div></div></div></footer></div><script src="../../../assets/javascripts/application.b260a35d.js"></script><script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script></body></html>