<!doctype html><html lang="en" class="no-js"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1"><meta http-equiv="x-ua-compatible" content="ie=edge"><meta name="lang:clipboard.copy" content="Copy to clipboard"><meta name="lang:clipboard.copied" content="Copied to clipboard"><meta name="lang:search.language" content="en"><meta name="lang:search.pipeline.stopwords" content="True"><meta name="lang:search.pipeline.trimmer" content="True"><meta name="lang:search.result.none" content="No matching documents"><meta name="lang:search.result.one" content="1 matching document"><meta name="lang:search.result.other" content="# matching documents"><meta name="lang:search.tokenizer" content="[\s\-]+"><link rel="shortcut icon" href="../../../assets/images/favicon.png"><meta name="generator" content="mkdocs-1.0.4, mkdocs-material-4.3.1"><title>Tutorial - Bioinformatics Documentation</title><link rel="stylesheet" href="../../../assets/stylesheets/application.4031d38b.css"><script src="../../../assets/javascripts/modernizr.74668098.js"></script><link href="https://fonts.gstatic.com" rel="preconnect" crossorigin><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono&display=swap"><style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style><link rel="stylesheet" href="../../../assets/fonts/material-icons.css"><link rel="stylesheet" href="../../../extra.css"><script>window.ga = window.ga || function() {
    (ga.q = ga.q || []).push(arguments)
  }
  ga.l = +new Date
  /* Setup integration and send page view */
  ga("create", "UA-21794901-5", "www.melbournebioinformatics.org.au/tutorials/")
  ga("set", "anonymizeIp", true)
  ga("send", "pageview")
  /* Register handler to log search on blur */
  document.addEventListener("DOMContentLoaded", () => {
    if (document.forms.search) {
      var query = document.forms.search.query
      query.addEventListener("blur", function() {
        if (this.value) {
          var path = document.location.pathname;
          ga("send", "pageview", path + "?q=" + this.value)
        }
      })
    }
  })</script><script async src="https://www.google-analytics.com/analytics.js"></script></head><body dir="ltr"><svg class="md-svg"><defs><svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="__github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg></defs></svg> <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off"> <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off"> <label class="md-overlay" data-md-component="overlay" for="__drawer"></label><a href="#galaxy-workflows" tabindex="1" class="md-skip">Skip to content </a><header class="md-header" data-md-component="header"><nav class="md-header-nav md-grid"><div class="md-flex"><div class="md-flex__cell md-flex__cell--shrink"><a href="../../.." title="Bioinformatics Documentation" class="md-header-nav__button md-logo"><i class="md-icon"></i></a></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--menu md-header-nav__button" for="__drawer"></label></div><div class="md-flex__cell md-flex__cell--stretch"><div class="md-flex__ellipsis md-header-nav__title" data-md-component="title"><span class="md-header-nav__topic">Bioinformatics Documentation</span><span class="md-header-nav__topic">Tutorial</span></div></div><div class="md-flex__cell md-flex__cell--shrink"><label class="md-icon md-icon--search md-header-nav__button" for="__search"></label><div class="md-search" data-md-component="search" role="dialog"><label class="md-search__overlay" for="__search"></label><div class="md-search__inner" role="search"><form class="md-search__form" name="search"><input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active"> <label class="md-icon md-search__icon" for="__search"></label> <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">&#xE5CD;</button></form><div class="md-search__output"><div class="md-search__scrollwrap" data-md-scrollfix><div class="md-search-result" data-md-component="result"><div class="md-search-result__meta">Type to start searching</div><ol class="md-search-result__list"></ol></div></div></div></div></div></div><div class="md-flex__cell md-flex__cell--shrink"><div class="md-header-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div></div></div></nav></header><div class="md-container"><main class="md-main"><div class="md-main__inner md-grid" data-md-component="container"><div class="md-sidebar md-sidebar--primary" data-md-component="navigation"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--primary" data-md-level="0"><label class="md-nav__title md-nav__title--site" for="__drawer"><a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo"><i class="md-icon"></i></a>Bioinformatics Documentation</label><div class="md-nav__source"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-source="github"><div class="md-source__icon"><svg viewBox="0 0 24 24" width="24" height="24"><use xlink:href="#__github" width="24" height="24"></use></svg></div><div class="md-source__repository">melbournebioinformatics/MelBioInf_docs</div></a></div><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../.." title="Home" class="md-nav__link">Home</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2"><label class="md-nav__link" for="nav-2">Guides</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-2">Guides</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../../guides/bioinfo/" title="What is bioinformatics?" class="md-nav__link">What is bioinformatics?</a></li><li class="md-nav__item"><a href="../../../guides/galaxy/" title="The Galaxy platform" class="md-nav__link">The Galaxy platform</a></li></ul></nav></li><li class="md-nav__item md-nav__item--active md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked><label class="md-nav__link" for="nav-3">General Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-3">General Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../python_overview/python_overview/" title="Python Overview" class="md-nav__link">Python Overview</a></li><li class="md-nav__item"><a href="../../python_pandas_tidy_data/python_pandas_tidy_data/" title="Data tidying with Python and Pandas" class="md-nav__link">Data tidying with Python and Pandas</a></li><li class="md-nav__item"><a href="../../python_viz/python_viz/" title="Data visualisation with Python" class="md-nav__link">Data visualisation with Python</a></li><li class="md-nav__item"><a href="../../using_git/Using_Git/" title="Using Git and Github for revision control" class="md-nav__link">Using Git and Github for revision control</a></li><li class="md-nav__item"><a href="../../hpc/hpc/" title="Introduction to HPC" class="md-nav__link">Introduction to HPC</a></li><li class="md-nav__item"><a href="../../unix/unix/" title="Introduction to Unix" class="md-nav__link">Introduction to Unix</a></li><li class="md-nav__item"><a href="../../gvl_launch/gvl_launch/" title="Launching a Personal GVL Server" class="md-nav__link">Launching a Personal GVL Server</a></li><li class="md-nav__item"><a href="../../galaxy_101/galaxy_101/" title="Introduction to Galaxy" class="md-nav__link">Introduction to Galaxy</a></li><li class="md-nav__item md-nav__item--active md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-9" type="checkbox" id="nav-3-9" checked><label class="md-nav__link" for="nav-3-9">Galaxy Workflows</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-3-9">Galaxy Workflows</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--active"><input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="__toc"><label class="md-nav__link md-nav__link--active" for="__toc">Tutorial</label><a href="./" title="Tutorial" class="md-nav__link md-nav__link--active">Tutorial</a><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a></li><li class="md-nav__item"><a href="#section-1-preparation" title="Section 1: Preparation." class="md-nav__link">Section 1: Preparation.</a></li><li class="md-nav__item"><a href="#section-2-create-and-run-a-workflow" title="Section 2: Create and run a workflow." class="md-nav__link">Section 2: Create and run a workflow.</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#import-the-workflow-history" title="Import the workflow history" class="md-nav__link">Import the workflow history</a></li></ul></nav></li><li class="md-nav__item"><a href="#workflow-creation-method-1" title="Workflow creation: Method 1" class="md-nav__link">Workflow creation: Method 1</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#some-discussion" title="Some discussion" class="md-nav__link">Some discussion</a></li></ul></nav></li><li class="md-nav__item"><a href="#workflow-creation-method-2" title="Workflow Creation: Method 2" class="md-nav__link">Workflow Creation: Method 2</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#step-1-create-a-workflow-name-and-edit-space" title="Step 1: Create a workflow name and edit space." class="md-nav__link">Step 1: Create a workflow name and edit space.</a></li><li class="md-nav__item"><a href="#step-2-open-the-editor-and-place-component-tools" title="Step 2: Open the editor and place component tools" class="md-nav__link">Step 2: Open the editor and place component tools</a></li><li class="md-nav__item"><a href="#step-3-save-it" title="Step 3: Save it!" class="md-nav__link">Step 3: Save it!</a></li></ul></nav></li><li class="md-nav__item"><a href="#running-the-workflow" title="Running the workflow" class="md-nav__link">Running the workflow</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#create-the-new-history" title="Create the new history" class="md-nav__link">Create the new history</a></li><li class="md-nav__item"><a href="#run-the-workflow" title="Run the workflow" class="md-nav__link">Run the workflow</a></li></ul></nav></li><li class="md-nav__item"><a href="#what-now" title="What now?" class="md-nav__link">What now?</a></li></ul></nav></li><li class="md-nav__item"><a href="../history_creation/" title="History Creation" class="md-nav__link">History Creation</a></li></ul></nav></li><li class="md-nav__item"><a href="../../docker/docker/" title="Containerized Bioinformatics" class="md-nav__link">Containerized Bioinformatics</a></li><li class="md-nav__item"><a href="../../cwl/cwl/" title="Common Workflow Language for Bioinformatics" class="md-nav__link">Common Workflow Language for Bioinformatics</a></li><li class="md-nav__item"><a href="../../bionitio/bionitio/" title="Command-line software development" class="md-nav__link">Command-line software development</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4"><label class="md-nav__link" for="nav-4">Genomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-4">Genomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" title="Introduction to Variant Calling" class="md-nav__link">Introduction to Variant Calling</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2"><label class="md-nav__link" for="nav-4-2">Variant Calling Advanced</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-2">Variant Calling Advanced</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../var_detect_advanced/var_detect_advanced_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3"><label class="md-nav__link" for="nav-4-3">RNA-seq DGE Basic</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-3">RNA-seq DGE Basic</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" title="Tuxedo Protocol Tutorial" class="md-nav__link">Tuxedo Protocol Tutorial</a></li><li class="md-nav__item"><a href="../../rna_seq_dge_basic/rna_seq_basic_background/" title="Background" class="md-nav__link">Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" title="RNA-seq DGE Advanced" class="md-nav__link">RNA-seq DGE Advanced</a></li><li class="md-nav__item"><a href="../../rna_seq_exp_design/rna_seq_experimental_design/" title="RNA-seq DGE Experimental Design" class="md-nav__link">RNA-seq DGE Experimental Design</a></li><li class="md-nav__item"><a href="../../singlecell/" title="10X single-cell RNA-seq analysis in R" class="md-nav__link">10X single-cell RNA-seq analysis in R</a></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-7" type="checkbox" id="nav-4-7"><label class="md-nav__link" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-4-7">Microbial de novo Assembly for Illumina Data: Velvet</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../assembly/assembly-background/" title="Method Background" class="md-nav__link">Method Background</a></li></ul></nav></li><li class="md-nav__item"><a href="../../assembly/spades/" title="Microbial assembly: Spades" class="md-nav__link">Microbial assembly: Spades</a></li><li class="md-nav__item"><a href="../../pacbio/" title="Long read assembly" class="md-nav__link">Long read assembly</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5"><label class="md-nav__link" for="nav-5">Proteomics Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-5">Proteomics Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-1" type="checkbox" id="nav-5-1"><label class="md-nav__link" for="nav-5-1">Identifying proteins from mass spectrometry data</label><nav class="md-nav" data-md-component="collapsible" data-md-level="2"><label class="md-nav__title" for="nav-5-1">Identifying proteins from mass spectrometry data</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../proteomics_basic/proteomics_basic/" title="Tutorial" class="md-nav__link">Tutorial</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_galaxy/" title="Galaxy Background" class="md-nav__link">Galaxy Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_data_formats/" title="Data Formats Background" class="md-nav__link">Data Formats Background</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_findcountitems_workflow/" title="Create find count items workflow" class="md-nav__link">Create find count items workflow</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_databases/" title="Protein Databases" class="md-nav__link">Protein Databases</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_search_engines/" title="Search Engines" class="md-nav__link">Search Engines</a></li><li class="md-nav__item"><a href="../../proteomics_basic/background_protein_prophet/" title="Protein Prophet" class="md-nav__link">Protein Prophet</a></li></ul></nav></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6"><label class="md-nav__link" for="nav-6">Molecular Modelling Tutorials</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-6">Molecular Modelling Tutorials</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../molecular_dynamics_101/molecular_dynamics_101/" title="Molecular Dynamics - Introduction to cluster computing" class="md-nav__link">Molecular Dynamics - Introduction to cluster computing</a></li><li class="md-nav__item"><a href="../../molecular_dynamics_201/molecular_dynamics_201/" title="Molecular Dynamics - Building input files, visualising the trajectory" class="md-nav__link">Molecular Dynamics - Building input files, visualising the trajectory</a></li></ul></nav></li><li class="md-nav__item md-nav__item--nested"><input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7"><label class="md-nav__link" for="nav-7">Protocols</label><nav class="md-nav" data-md-component="collapsible" data-md-level="1"><label class="md-nav__title" for="nav-7">Protocols</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="../../assembly/assembly-protocol/" title="De Novo Genome Assembly for Illumina Data" class="md-nav__link">De Novo Genome Assembly for Illumina Data</a></li></ul></nav></li></ul></nav></div></div></div><div class="md-sidebar md-sidebar--secondary" data-md-component="toc"><div class="md-sidebar__scrollwrap"><div class="md-sidebar__inner"><nav class="md-nav md-nav--secondary"><label class="md-nav__title" for="__toc">Table of contents</label><ul class="md-nav__list" data-md-scrollfix><li class="md-nav__item"><a href="#background" title="Background" class="md-nav__link">Background</a></li><li class="md-nav__item"><a href="#section-1-preparation" title="Section 1: Preparation." class="md-nav__link">Section 1: Preparation.</a></li><li class="md-nav__item"><a href="#section-2-create-and-run-a-workflow" title="Section 2: Create and run a workflow." class="md-nav__link">Section 2: Create and run a workflow.</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#import-the-workflow-history" title="Import the workflow history" class="md-nav__link">Import the workflow history</a></li></ul></nav></li><li class="md-nav__item"><a href="#workflow-creation-method-1" title="Workflow creation: Method 1" class="md-nav__link">Workflow creation: Method 1</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#some-discussion" title="Some discussion" class="md-nav__link">Some discussion</a></li></ul></nav></li><li class="md-nav__item"><a href="#workflow-creation-method-2" title="Workflow Creation: Method 2" class="md-nav__link">Workflow Creation: Method 2</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#step-1-create-a-workflow-name-and-edit-space" title="Step 1: Create a workflow name and edit space." class="md-nav__link">Step 1: Create a workflow name and edit space.</a></li><li class="md-nav__item"><a href="#step-2-open-the-editor-and-place-component-tools" title="Step 2: Open the editor and place component tools" class="md-nav__link">Step 2: Open the editor and place component tools</a></li><li class="md-nav__item"><a href="#step-3-save-it" title="Step 3: Save it!" class="md-nav__link">Step 3: Save it!</a></li></ul></nav></li><li class="md-nav__item"><a href="#running-the-workflow" title="Running the workflow" class="md-nav__link">Running the workflow</a><nav class="md-nav"><ul class="md-nav__list"><li class="md-nav__item"><a href="#create-the-new-history" title="Create the new history" class="md-nav__link">Create the new history</a></li><li class="md-nav__item"><a href="#run-the-workflow" title="Run the workflow" class="md-nav__link">Run the workflow</a></li></ul></nav></li><li class="md-nav__item"><a href="#what-now" title="What now?" class="md-nav__link">What now?</a></li></ul></nav></div></div></div><div class="md-content"><article class="md-content__inner md-typeset"><a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/galaxy-workflows/galaxy-workflows.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a><style>
  .image-header-gvl {
    -webkit-column-count: 3;
    -moz-column-count: 3;
    column-count: 3;

    -webkit-column-gap: 140px;
    -moz-column-gap: 140px;
    column-gap: 140px;
  }
</style>

<div class="image-header-gvl">
    <a href=http://genome.edu.au><img src="../media/gvl_logo.jpg" alt="GVL logo" width="112"/></a>
    <a href="http://galaxyproject.org"><img src="../media/GalaxyLogoHighRes.png" width="164" /></a>
    <a href="https://www.melbournebioinformatics.org.au/"><img src="../../../img/melbioinf_logo.png" alt="Melbourne Bioinformatics logo" width="164"/></a>
</div>

<h1 id="galaxy-workflows">Galaxy Workflows</h1>
<p>Written and maintained by <a href="mailto:simon.gladman@unimelb.edu.au">Simon Gladman</a> - Melbourne Bioinformatics (formerly VLSCI)</p>
<h2 id="background">Background</h2>
<p>This workshop/tutorial will familiarise you with the Galaxy workflow engine. It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>How to construct and use a workflow by various methods</li>
<li>How to share a workflow</li>
</ul>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.</h2>
<p>The purpose of this section is to get you to log in to the server.. For this workshop, you can use a Galaxy server that you have created by following the steps outlined in: <a href="">Launch a GVL Galaxy Instance</a> or you can use the public <a href="https://usegalaxy.org.au/">Galaxy Australia</a></p>
<p><strong>Go to Galaxy URL of your server in Firefox or Chrome (your choice, please don't use IE or Safari.)</strong></p>
<ul>
<li>If you have previously registered on this server just log in:<ul>
<li>On the top menu select: <strong>User -&gt; Login</strong></li>
<li>Enter your password</li>
<li>Click the <strong>Submit</strong> button</li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>If you haven’t registered on this server, you’ll need to now.<ul>
<li>On the top menu select: <strong>User -&gt; Register</strong></li>
<li>Enter your email, choose a password, repeat it and add a (all lower case) one word name</li>
<li>Click on the <strong>Submit</strong> button.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-2-create-and-run-a-workflow">Section 2: Create and run a workflow.</h2>
<p>This section will show you two different methods to create a workflow and then how to run one.</p>
<h3 id="import-the-workflow-history">Import the workflow history</h3>
<p>In this step we will import a shared history to our workspace so we can extract a workflow from it. This will only work on Galaxy servers which have the history available on it. If yours doesn't have the appropriate history, there are instructions to create it <a href="../history_creation/">here</a>. Alternatively, you can extract a workflow from any history you have in your "Saved Histories."</p>
<ul>
<li>From the menu at the top of the Galaxy window, click <strong>Shared Data -&gt; Histories</strong></li>
<li>Find the history called "<em>workflow_finished</em>" and click on it.</li>
<li>Then click on <strong>Import history</strong> at the top right of the screen.</li>
<li>Change the name if you wish and then click <strong>Import</strong></li>
</ul>
<p>This history should now be in your history pane on the right.</p>
<!--

This step will show you how to import a history from a remote source into your own workspace. We will be using this history to build a workflow.

* From the histories menu <img src="../media/Galaxy-menu.png" width=20 />, click **Import from File**.
* Enter *https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Workflow-finished-history.tar.gz* in the URL box.
* Click **Submit**

You might have to wait for a bit, then:

* From the history menu <img src="../media/Galaxy-menu.png" width=20 />, click **Saved Histories**
* Select the history: *Imported: Workflow-finished*


-->

<h2 id="workflow-creation-method-1">Workflow creation: Method 1</h2>
<p>We will create a workflow from an existing history. You can use this method to make a re-useable analysis from one you’ve already done. i.e. You can perform the analysis once and then create a workflow out of it to re-use it on more/new data. We will create a workflow from the history you imported in step 1 above. The footnote below explains the steps that created this history. These are the steps we will mimic in the workflow.</p>
<p>Make sure your current history is the one you imported in Section 2 - Step 1 above (<em>imported: workflow_finished.</em>) If not, switch to it. If you couldn't import it, you should be able to complete this step with any suitable history.</p>
<p><strong>Now we will create the workflow.</strong></p>
<ul>
<li>Click on the histories menu button <img src="../media/Galaxy-menu.png" width=20 /> at the top of the history pane.</li>
<li>Click <strong>Extract Workflow</strong></li>
</ul>
<p>You will now be shown a page which contains the steps used to create the history you are extracting from. We use this page to say what to include in the workflow. We want everything here so we’ll just accept the defaults and:</p>
<ul>
<li>Change the Workflow name to something sensible like “Basic Variant Calling Workflow”</li>
<li>Click <strong>Create Workflow</strong></li>
</ul>
<p>The workflow is now accessible via the bottom of the tool pane by clicking on All Workflows.</p>
<h3 id="some-discussion">Some discussion</h3>
<p>Have a look at your workflow. Click on its button in the workflow list. It’s tool interface will appear. You can now run this workflow any time you like with different input datasets. NOTE: The input data sets must be of the same types as the original ones. i.e. In our case, two fastq reads files and one fasta reference sequence.</p>
<p>More interesting though is to:</p>
<ul>
<li>Click on the  <strong>Workflows</strong> link in the top menu</li>
<li>Click on the down arrow on your workflow’s button.</li>
<li>Click <strong>Edit</strong></li>
</ul>
<p>A visualisation of your workflow will appear. Note the connections and the steps.</p>
<p>Next we’ll go through how to create this workflow using the editor..</p>
<h2 id="workflow-creation-method-2">Workflow Creation: Method 2</h2>
<p>We will now create the same read mapping/variant calling workflow using the editor directly. The workflow needs to take in some reads and a reference, map the reads to the reference using BWA, run Freebayes on the BAM output from BWA to call the variants, and finally filter the resulting vcf file.</p>
<h3 id="step-1-create-a-workflow-name-and-edit-space">Step 1: Create a workflow name and edit space.</h3>
<ul>
<li>Click on <strong>Workflow</strong> in Galaxy’s menu.</li>
<li>Click on the <strong>Create New Workflow</strong> button.</li>
<li>In the "Workflow Name" text box type: <em>Variants from scratch</em></li>
<li>Click the <strong>Create</strong> button.</li>
</ul>
<p>You should now be presented with a blank workflow editing grid.</p>
<h3 id="step-2-open-the-editor-and-place-component-tools">Step 2: Open the editor and place component tools</h3>
<p><strong>Add three input datafiles.</strong></p>
<ul>
<li>In the Workflow control section of the tool pane, click on <strong>Inputs -&gt; Input dataset</strong> three times.</li>
<li>Spread them out towards the left hand side of the workflow grid by clicking and dragging them around.</li>
<li>For each one, change their name.<ul>
<li>Click on each input box in turn</li>
<li>In the right hand pane (where the history usually is), change the name to:<ul>
<li>Reference data</li>
<li>Reads 1</li>
<li>Reads 2 - respectively.</li>
</ul>
</li>
<li>If you click on <strong>Input dataset</strong> in the tan box at the top of the right hand panel on the screen, you can change the name of the box as it appears on the editing layout. You need to give each one a more sensible name and press <em>Enter</em> to make the change.</li>
</ul>
</li>
</ul>
<p><strong>Add in the BWA mapping step.</strong></p>
<ul>
<li>Click on <strong>NGS: Mapping -&gt; Map with BWA</strong> in the tool pane. BWA will be added to the workflow grid.</li>
<li>In the right hand pane (where the history usually is), change the following parameters.<ul>
<li>Change “Will you select a reference genome from your history or use a built-in index?:” to <em>Use a genome from history.</em></li>
<li>Note that the BWA box on the grid changes to match these settings.</li>
</ul>
</li>
</ul>
<p><strong>Connect the tools together.</strong></p>
<ul>
<li>Click and drag on the output of one of the input dataset tools to each of the input spots in the BWA tool. (Make connections.)<ul>
<li>"Reference data" output to reference to "Use the following dataset as the reference sequence"</li>
<li>"Reads 1" output to "Select first set of reads"</li>
<li>"Reads 2" output to "Select second set of reads"</li>
</ul>
</li>
</ul>
<p><strong>Add in the Freebayes (Variant Calling step.)</strong></p>
<ul>
<li>Click on <strong>NGS: Variant Calling -&gt; Freebayes</strong></li>
<li>In the right hand pane, change the following:<ul>
<li>"Choose the source for the reference list:"  to <em>History</em></li>
<li>Connect "Map with BWA" bam output to "Freebayes’" bam input.</li>
<li>Connect the "Reference data" output to "Freebayes’" <em>Use the following dataset as the reference sequence</em> input.</li>
</ul>
</li>
</ul>
<p>If you’re keen - <strong>Note: this is optional.</strong> Also change the following parameters the right hand pane for freebayes to make it a bit more sensible for variant calling in bacterial genomes.</p>
<ul>
<li>"Choose parameter selection level": <em>Complete list of all options</em></li>
<li>"Population model options": <em>Set population model options</em></li>
<li>"Set ploidy for the analysis": <em>1</em></li>
<li>"Input filters": <em>Set input filters</em></li>
<li>"Exclude alignments from analysis if they have a mapping quality less than": <em>20</em></li>
<li>"Exclude alleles from analysis if their supporting base quality is less than": <em>20</em></li>
<li>"Require at least this fraction of observations … to evaluate the position": <em>0.9</em></li>
<li>"Require at least this count of observations .. to evaluate the position": <em>10</em></li>
<li>"Population and mappability priors": <em>Set population and mappability priors</em></li>
<li>"Disable incorporation of prior expectations about observations": <em>Yes</em></li>
</ul>
<p><strong>Add in the Filter step.</strong></p>
<ul>
<li>Click on <strong>Filter and Sort - &gt; Filter</strong></li>
<li>Connect "Freebayes’" output_vcf to the "filter" input.</li>
<li>In the right hand pane, change the following:<ul>
<li>"With the following condition": <em>c6 &gt; 500</em></li>
<li>"Number of header lines to skip": <em>56</em></li>
</ul>
</li>
</ul>
<p>Phew! We’re nearly done! The only thing left is to select which workflow outputs we want to keep in our history. Next to each output for every tool is a star. Clicking on the stars will select those files as workflow outputs, everything else will be hidden in the history. In this case we only really want the BAM file and the final variants (vcf file.) Therefore:</p>
<p><strong>Select workflow outputs.</strong></p>
<ul>
<li>Click on the star next to "Map with BWA’s" bam file output.</li>
<li>Click on the star next to "Filter’s" output vcf.</li>
</ul>
<h3 id="step-3-save-it">Step 3: Save it!</h3>
<p>Click on the <img src="../media/Galaxy-menu.png" width=20 /> at the top of the workflow grid and select <strong>Save</strong>.</p>
<p>Congratulations. You’ve just created a Galaxy workflow.</p>
<p>Now to run it!</p>
<h2 id="running-the-workflow">Running the workflow</h2>
<p>We will now make a new history called "Test" and run the workflow on it’s data.</p>
<h3 id="create-the-new-history">Create the new history</h3>
<ul>
<li>From the Histories Menu, select <strong>Copy Datasets</strong></li>
<li>Select the 2 x fastq files and the Ecoli .fna file.</li>
<li>Under destination history, enter <em>Test</em> into "New History Named:"</li>
<li>Click <strong>Copy History Items</strong> button</li>
<li>Click on the link to the new history in the green bar at the top of the screen</li>
</ul>
<h3 id="run-the-workflow">Run the workflow</h3>
<ul>
<li>On the tools pane, click <strong>All Workflows</strong></li>
<li>Select the <em>Variants from scratch</em> workflow (or whatever you called it.)</li>
<li>Give it the correct files.</li>
<li><em>Ecoli ... .fna</em> for Reference data</li>
<li><em>bacterial_std_err_1.fastq</em> for Reads 1</li>
<li><em>bacterial_std_err_2.fastq</em> for Reads 2</li>
<li>Click <strong>Run Workflow</strong></li>
</ul>
<p>Your workflow will now run. It will send the right files to the right tools at the right time to the cluster (compute engine on your machine) and wait for them to finish. Watch as they turn yellow then green in turn.</p>
<h2 id="what-now">What now?</h2>
<p>Where to start? There’s so much you can do with Workflows. You can even run them on multiple file sets from the one setup.</p></article></div></div></main><footer class="md-footer"><div class="md-footer-nav"><nav class="md-footer-nav__inner md-grid"><a href="../../galaxy_101/galaxy_101/" title="Introduction to Galaxy" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev"><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-back md-footer-nav__button"></i></div><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Previous</span>Introduction to Galaxy</span></div></a><a href="../history_creation/" title="History Creation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next"><div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title"><span class="md-flex__ellipsis"><span class="md-footer-nav__direction">Next</span>History Creation</span></div><div class="md-flex__cell md-flex__cell--shrink"><i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i></div></a></nav></div><div class="md-footer-meta md-typeset"><div class="md-footer-meta__inner md-grid"><div class="md-footer-copyright"><div class="md-footer-copyright__highlight"><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a></div>powered by <a href="https://www.mkdocs.org">MkDocs</a> and <a href="https://squidfunk.github.io/mkdocs-material/">Material for MkDocs</a></div></div></div></footer></div><script src="../../../assets/javascripts/application.b260a35d.js"></script><script>app.initialize({version:"1.0.4",url:{base:"../../.."}})</script></body></html>